{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import pandas as pd\n",
    "\n",
    "sys.path.append(\"../src\")\n",
    "from pharmgkb import RawData\n",
    "from utils import CsvCleaner\n",
    "\n",
    "\n",
    "data_path = os.path.join(\"..\", \"data\")\n",
    "pharmgkb_path = os.path.join(data_path, \"pharmgkb\")\n",
    "processed_path = os.path.join(data_path, \"pharmgkb_processed\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quick and dirty exploration of files for curation"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Chemicals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(os.path.join(pharmgkb_path, \"chemicals\", \"chemicals.csv\"))\n",
    "for i,c in enumerate(df.columns):\n",
    "    print(i, c)\n",
    "\n",
    "df[\"Top FDA Label Testing Level\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[df[\"Name\"]==\"apremilast\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Retrieve SMILES for PubChem compounds\n",
    "import requests\n",
    "\n",
    "data = requests.get(\"https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/cid/59272813/property/CanonicalSMILES/TXT\")\n",
    "smi = data.text.strip()\n",
    "smi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sys.path.append(os.path.join(\"../src\"))\n",
    "\n",
    "from utils import CsvCleaner\n",
    "import requests\n",
    "\n",
    "c = CsvCleaner()\n",
    "\n",
    "data_dict = {}\n",
    "for r in df.values:\n",
    "    chemical = c.stringify(r[1])\n",
    "    crossr = c.inline_comma_splitter(r[6])\n",
    "    if crossr is not None:\n",
    "        for i in crossr:\n",
    "            print(crossr)\n",
    "            if len(crossr) == 1:\n",
    "                if \"PubChem Compound\" in i:\n",
    "                    cpd = (i.split(\":\")[-1])\n",
    "                    data = requests.get(\"https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/cid/{}/property/CanonicalSMILES/TXT\".format(cpd))\n",
    "                    smi = data.text.strip()\n",
    "                    print(smi)\n",
    "            else:\n",
    "                if \"PubChem Compound\" in i:\n",
    "                    cpd = (i.split(\":\")[-1][:-1])\n",
    "                    data = requests.get(\"https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/cid/{}/property/CanonicalSMILES/TXT\".format(cpd))\n",
    "                    smi = data.text.strip()\n",
    "                    print(smi)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(4532, 6)\n",
      "1756\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv(os.path.join(processed_path, \"chemical.csv\"))\n",
    "print(df.shape)\n",
    "print(len(df[df[\"smiles\"].isna()]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cid</th>\n",
       "      <th>chemical</th>\n",
       "      <th>chemical_type</th>\n",
       "      <th>smiles</th>\n",
       "      <th>dosing_guideline</th>\n",
       "      <th>drug_label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>PA151958637</td>\n",
       "      <td>4-methylthioamphetamine</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>PA154081778</td>\n",
       "      <td>4-methylumbelliferone</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>PA151958362</td>\n",
       "      <td>6,7-dihydroxybergamottin</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270</th>\n",
       "      <td>PA152530740</td>\n",
       "      <td>8-cyclopentyl-1,3-dipropylxanthine</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>PA164747080</td>\n",
       "      <td>abatacept</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>PA448006</td>\n",
       "      <td>abciximab</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>298</th>\n",
       "      <td>PA166104276</td>\n",
       "      <td>ABT-751</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313</th>\n",
       "      <td>PA448014</td>\n",
       "      <td>acetamide mea</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315</th>\n",
       "      <td>PA166246281</td>\n",
       "      <td>acetaminophen / caffeine / dihydrocodeine</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320</th>\n",
       "      <td>PA165290927</td>\n",
       "      <td>acetaminophen/propoxyphene napsylate</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>322</th>\n",
       "      <td>PA166184496</td>\n",
       "      <td>acetaminophen / tramadol</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>323</th>\n",
       "      <td>PA448016</td>\n",
       "      <td>acetanilide</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>330</th>\n",
       "      <td>PA448025</td>\n",
       "      <td>acetone</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>354</th>\n",
       "      <td>PA10004</td>\n",
       "      <td>adalimumab</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369</th>\n",
       "      <td>PA165981139</td>\n",
       "      <td>afatinib dimaleate</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>370</th>\n",
       "      <td>PA452609</td>\n",
       "      <td>aflatoxin b1</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>371</th>\n",
       "      <td>PA166118342</td>\n",
       "      <td>aflibercept</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373</th>\n",
       "      <td>PA166122586</td>\n",
       "      <td>afutuzumab</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>374</th>\n",
       "      <td>PA164746527</td>\n",
       "      <td>agalsidase beta</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>388</th>\n",
       "      <td>PA448081</td>\n",
       "      <td>aldesleukin</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             cid                                   chemical chemical_type  \\\n",
       "179  PA151958637                    4-methylthioamphetamine          Drug   \n",
       "180  PA154081778                      4-methylumbelliferone          Drug   \n",
       "222  PA151958362                   6,7-dihydroxybergamottin          Drug   \n",
       "270  PA152530740         8-cyclopentyl-1,3-dipropylxanthine          Drug   \n",
       "293  PA164747080                                  abatacept          Drug   \n",
       "294     PA448006                                  abciximab          Drug   \n",
       "298  PA166104276                                    ABT-751          Drug   \n",
       "313     PA448014                              acetamide mea          Drug   \n",
       "315  PA166246281  acetaminophen / caffeine / dihydrocodeine          Drug   \n",
       "320  PA165290927       acetaminophen/propoxyphene napsylate          Drug   \n",
       "322  PA166184496                   acetaminophen / tramadol          Drug   \n",
       "323     PA448016                                acetanilide          Drug   \n",
       "330     PA448025                                    acetone          Drug   \n",
       "354      PA10004                                 adalimumab          Drug   \n",
       "369  PA165981139                         afatinib dimaleate          Drug   \n",
       "370     PA452609                               aflatoxin b1          Drug   \n",
       "371  PA166118342                                aflibercept          Drug   \n",
       "373  PA166122586                                 afutuzumab          Drug   \n",
       "374  PA164746527                            agalsidase beta          Drug   \n",
       "388     PA448081                                aldesleukin          Drug   \n",
       "\n",
       "    smiles  dosing_guideline  drug_label  \n",
       "179    NaN                -1           0  \n",
       "180    NaN                -1           0  \n",
       "222    NaN                -1           0  \n",
       "270    NaN                -1           0  \n",
       "293    NaN                -1           0  \n",
       "294    NaN                -1           0  \n",
       "298    NaN                -1           0  \n",
       "313    NaN                -1           0  \n",
       "315    NaN                -1           3  \n",
       "320    NaN                -1           0  \n",
       "322    NaN                -1           3  \n",
       "323    NaN                -1           0  \n",
       "330    NaN                -1           0  \n",
       "354    NaN                -1           0  \n",
       "369    NaN                -1           0  \n",
       "370    NaN                -1           0  \n",
       "371    NaN                -1           0  \n",
       "373    NaN                -1           4  \n",
       "374    NaN                -1           0  \n",
       "388    NaN                -1           0  "
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[(df[\"smiles\"].isna())&(df[\"chemical_type\"]==\"Drug\")].head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: 404\n",
      "Code: PUGREST.NotFound\n",
      "Message: No CID found\n",
      "Detail: No CID found that matches the given name\n"
     ]
    }
   ],
   "source": [
    "data = requests.get(\"https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/name/adalimumab/property/CanonicalSMILES/TXT\")\n",
    "smi = data.text.strip()\n",
    "print(smi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "yes\n"
     ]
    }
   ],
   "source": [
    "if \"Status: 404\" in smi:\n",
    "    print(\"yes\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 PharmGKB Accession Id\n",
      "1 NCBI Gene ID\n",
      "2 HGNC ID\n",
      "3 Ensembl Id\n",
      "4 Name\n",
      "5 Symbol\n",
      "6 Alternate Names\n",
      "7 Alternate Symbols\n",
      "8 Is VIP\n",
      "9 Has Variant Annotation\n",
      "10 Cross-references\n",
      "11 Has CPIC Dosing Guideline\n",
      "12 Chromosome\n",
      "13 Chromosomal Start - GRCh37\n",
      "14 Chromosomal Stop - GRCh37\n",
      "15 Chromosomal Start - GRCh38\n",
      "16 Chromosomal Stop - GRCh38\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv(os.path.join(pharmgkb_path, \"genes\", \"genes.csv\"),encoding= 'unicode_escape')\n",
    "for i,c in enumerate(df.columns):\n",
    "    print(i, c)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Disease"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 PharmGKB Accession Id\n",
      "1 Name\n",
      "2 Alternate Names\n",
      "3 Cross-references\n",
      "4 External Vocabulary\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv(os.path.join(pharmgkb_path, \"phenotypes\", \"phenotypes.csv\"),encoding= 'unicode_escape')\n",
    "for i,c in enumerate(df.columns):\n",
    "    print(i, c)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 Variant ID\n",
      "1 Variant Name\n",
      "2 Gene IDs\n",
      "3 Gene Symbols\n",
      "4 Location\n",
      "5 Variant Annotation count\n",
      "6 Clinical Annotation count\n",
      "7 Level 1/2 Clinical Annotation count\n",
      "8 Guideline Annotation count\n",
      "9 Label Annotation count\n",
      "10 Synonyms\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv(os.path.join(pharmgkb_path, \"variants\", \"variants.csv\"),encoding= 'unicode_escape')\n",
    "for i,c in enumerate(df.columns):\n",
    "    print(i, c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CYP3A4*36\n"
     ]
    }
   ],
   "source": [
    "sys.path.append(\"../src\")\n",
    "from utils import CsvCleaner\n",
    "c = CsvCleaner()\n",
    "\n",
    "# references\n",
    "vars = pd.read_csv(os.path.join(processed_path, \"variant.csv\"))\n",
    "haps = pd.read_csv(os.path.join(processed_path, \"haplotype.csv\"))\n",
    "refs = haps[\"haplotype\"].tolist() + vars[\"variant\"].tolist()\n",
    "\n",
    "df1 = pd.read_csv(os.path.join(pharmgkb_path, \"variantAnnotations\", \"var_pheno_ann.csv\"),encoding= 'unicode_escape')\n",
    "df2 = pd.read_csv(os.path.join(pharmgkb_path, \"variantAnnotations\", \"var_drug_ann.csv\"),encoding= 'unicode_escape')\n",
    "df3 = pd.read_csv(os.path.join(pharmgkb_path, \"variantAnnotations\", \"var_fa_ann.csv\"),encoding= 'unicode_escape')\n",
    "newvars = []\n",
    "for r in df1.values:\n",
    "    v = c.inline_comma_splitter_space(r[1])\n",
    "    for v_ in v:\n",
    "        newvars += [v_]\n",
    "for r in df2.values:\n",
    "    v = c.inline_comma_splitter_space(r[1])\n",
    "    for v_ in v:\n",
    "        newvars += [v_]\n",
    "for r in df3.values:\n",
    "    v = c.inline_comma_splitter_space(r[1])\n",
    "    for v_ in v:\n",
    "        newvars += [v_]\n",
    "\n",
    "#corrections known\n",
    "known_haps = [\"G6PD Canton\", \"Taiwan-Hakka\", \"Gifu-like\", \"Agrigento-like\", \"G6PD Mediterranean\", \"Dallas\", \"Panama\", \"Sassari\", \"Cagliari\", \"Birmingham\"]\n",
    "newvars = [\":\".join(value.split(\":\")[:2]) if value.startswith(\"HLA-\") else value for value in newvars]\n",
    "newvars = [value for value in newvars if value not in known_haps]\n",
    "newvars = [\"G6PD B (reference)\" if value == \"G6PD B (wildtype)\" else value for value in newvars]\n",
    "uniquevars = list(set(newvars)-set(refs))\n",
    "for var in uniquevars:\n",
    "    if \"*\" in var: #this is not a variant but a haplotype, will be added manually to haplotypes.csv\n",
    "        uniquevars.remove(var)\n",
    "        print(var)\n",
    "\n",
    "df = pd.DataFrame(uniquevars, columns=[\"variant\"])\n",
    "df.to_csv(os.path.join(processed_path, \"orphan_variant_list.csv\"), index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Haplotypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 Entity1_id\n",
      "1 Entity1_name\n",
      "2 Entity1_type\n",
      "3 Entity2_id\n",
      "4 Entity2_name\n",
      "5 Entity2_type\n",
      "6 Evidence\n",
      "7 Association\n",
      "8 PK\n",
      "9 PD\n",
      "10 PMIDs\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv(os.path.join(pharmgkb_path, \"relationships\", \"relationships.csv\"),encoding= 'unicode_escape')\n",
    "for i,c in enumerate(df.columns):\n",
    "    print(i, c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#We keep all the haplotypes that appear, either in entity 1 or in entity 2, and then get the gene from the Haplotype name\n",
    "df1 = df[\"Entity1_name\"][df[\"Entity1_type\"]==\"Haplotype\"].tolist()\n",
    "print(len(df1))\n",
    "df2 = df[\"Entity2_name\"][df[\"Entity2_type\"]==\"Haplotype\"].tolist()\n",
    "print(len(df2))\n",
    "print(len(set(df1+df2)))\n",
    "\n",
    "#get unique hids for either entity 1 or entity2 -- is the same\n",
    "haps = list(set(df1+df2))\n",
    "hids = []\n",
    "for r in df.values:\n",
    "    hap = r[1]\n",
    "    hid = r[0]\n",
    "    if hap in haps:\n",
    "        hids += [hid]\n",
    "len(hids)\n",
    "len(set(hids))\n",
    "\n",
    "haps = list(set(df1+df2))\n",
    "hids = []\n",
    "for r in df.values:\n",
    "    hap = r[4]\n",
    "    hid = r[3]\n",
    "    if hap in haps:\n",
    "        hids += [hid]\n",
    "len(hids)\n",
    "len(set(hids))\n",
    "hids = list(set(hids))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(11788, 11)\n",
      "(1028, 11)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_10338/925833839.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df1.drop_duplicates(subset= [\"Entity1_name\"], keep=\"first\", inplace=True)\n"
     ]
    }
   ],
   "source": [
    "#We keep all the haplotypes that appear, either in entity 1 or in entity 2, and then get the gene from the Haplotype name\n",
    "df1 = df[df[\"Entity1_type\"]==\"Haplotype\"]\n",
    "print(df1.shape)\n",
    "df1.drop_duplicates(subset= [\"Entity1_name\"], keep=\"first\", inplace=True)\n",
    "print(df1.shape)\n",
    "df1 = df1[[\"Entity1_id\", \"Entity1_name\"]]\n",
    "df1.rename(columns={\"Entity1_id\": \"hid\", \"Entity1_name\":\"haplotype\"}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = []\n",
    "for h in df1[\"haplotype\"]:\n",
    "    if \" \" in h:\n",
    "        g = h.split(\" \")[0]\n",
    "    elif \"*\" in h:\n",
    "        g = h.split(\"*\")[0]\n",
    "    else:\n",
    "        g = h\n",
    "    gene += [g]\n",
    "df1[\"gene\"]= gene\n",
    "gl = list(set(df1[\"gene\"].tolist()))\n",
    "gl"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Clincal variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(os.path.join(pharmgkb_path, \"clinicalVariants\", \"clinicalVariants.csv\"),encoding= 'unicode_escape')\n",
    "for i,c in enumerate(df.columns):\n",
    "    print(i, c)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Clinical Annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alleles\n",
      "0 Clinical Annotation ID\n",
      "1 Genotype/Allele\n",
      "2 Annotation Text\n",
      "3 Allele Function\n",
      "4 test\n",
      "Evidence\n",
      "0 Clinical Annotation ID\n",
      "1 Evidence ID\n",
      "2 Evidence Type\n",
      "3 Evidence URL\n",
      "4 PMID\n",
      "5 Summary\n",
      "6 Score\n",
      "History\n",
      "0 Clinical Annotation ID\n",
      "1 Date (YYYY-MM-DD)\n",
      "2 Type\n",
      "3 Comment\n",
      "ClinicalAnn\n",
      "0 Clinical Annotation ID\n",
      "1 Variant/Haplotypes\n",
      "2 Gene\n",
      "3 Level of Evidence\n",
      "4 Level Override\n",
      "5 Level Modifiers\n",
      "6 Score\n",
      "7 Phenotype Category\n",
      "8 PMID Count\n",
      "9 Evidence Count\n",
      "10 Drug(s)\n",
      "11 Phenotype(s)\n",
      "12 Latest History Date (YYYY-MM-DD)\n",
      "13 URL\n",
      "14 Specialty Population\n"
     ]
    }
   ],
   "source": [
    "#clinical annotations are identified by a single ID but can be  referenced in different files\n",
    "al = pd.read_csv(os.path.join(pharmgkb_path, \"clinicalAnnotations\", \"clinical_ann_alleles.csv\"))\n",
    "ev = pd.read_csv(os.path.join(pharmgkb_path, \"clinicalAnnotations\", \"clinical_ann_evidence.csv\"),encoding= 'unicode_escape')\n",
    "his = pd.read_csv(os.path.join(pharmgkb_path, \"clinicalAnnotations\", \"clinical_ann_history.csv\"),encoding= 'unicode_escape')\n",
    "cl = pd.read_csv(os.path.join(pharmgkb_path, \"clinicalAnnotations\", \"clinical_annotations.csv\"),encoding= 'unicode_escape')\n",
    "\n",
    "\n",
    "print(\"Alleles\")\n",
    "for i,c in enumerate(al.columns):\n",
    "    print(i, c)\n",
    "print(\"Evidence\")\n",
    "for i,c in enumerate(ev.columns):\n",
    "    print(i, c)\n",
    "print(\"History\")\n",
    "for i,c in enumerate(his.columns):\n",
    "    print(i, c)\n",
    "print(\"ClinicalAnn\")\n",
    "for i,c in enumerate(cl.columns):\n",
    "    print(i, c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Clinical Annotation ID</th>\n",
       "      <th>Genotype/Allele</th>\n",
       "      <th>Annotation Text</th>\n",
       "      <th>Allele Function</th>\n",
       "      <th>test</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2809</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>*1</td>\n",
       "      <td>Patients carrying the CYP2D6*1 allele in combi...</td>\n",
       "      <td>Normal function</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2810</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>*2</td>\n",
       "      <td>Patients carrying the CYP2D6*2 allele in combi...</td>\n",
       "      <td>Normal function</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2811</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>*3</td>\n",
       "      <td>Patients carrying the CYP2D6*3 allele in combi...</td>\n",
       "      <td>No function</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2812</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>*4</td>\n",
       "      <td>Patients carrying the CYP2D6*4 allele in combi...</td>\n",
       "      <td>No function</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2813</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>*5</td>\n",
       "      <td>Patients carrying the CYP2D6*5 allele in combi...</td>\n",
       "      <td>No function</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2814</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>*6</td>\n",
       "      <td>Patients carrying the CYP2D6*6 allele in combi...</td>\n",
       "      <td>No function</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2815</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>*10</td>\n",
       "      <td>Patients carrying the CYP2D6*10 allele in comb...</td>\n",
       "      <td>Decreased function (AV 0.25)</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Clinical Annotation ID Genotype/Allele  \\\n",
       "2809              1183615480              *1   \n",
       "2810              1183615480              *2   \n",
       "2811              1183615480              *3   \n",
       "2812              1183615480              *4   \n",
       "2813              1183615480              *5   \n",
       "2814              1183615480              *6   \n",
       "2815              1183615480             *10   \n",
       "\n",
       "                                        Annotation Text  \\\n",
       "2809  Patients carrying the CYP2D6*1 allele in combi...   \n",
       "2810  Patients carrying the CYP2D6*2 allele in combi...   \n",
       "2811  Patients carrying the CYP2D6*3 allele in combi...   \n",
       "2812  Patients carrying the CYP2D6*4 allele in combi...   \n",
       "2813  Patients carrying the CYP2D6*5 allele in combi...   \n",
       "2814  Patients carrying the CYP2D6*6 allele in combi...   \n",
       "2815  Patients carrying the CYP2D6*10 allele in comb...   \n",
       "\n",
       "                   Allele Function  test  \n",
       "2809               Normal function   NaN  \n",
       "2810               Normal function   NaN  \n",
       "2811                   No function   NaN  \n",
       "2812                   No function   NaN  \n",
       "2813                   No function   NaN  \n",
       "2814                   No function   NaN  \n",
       "2815  Decreased function (AV 0.25)   NaN  "
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Use a CAID (Clinical Ann ID) as an example see the information across files\n",
    "caid = 1183615480\n",
    "al[al[\"Clinical Annotation ID\"]==caid]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Clinical Annotation ID</th>\n",
       "      <th>Evidence ID</th>\n",
       "      <th>Evidence Type</th>\n",
       "      <th>Evidence URL</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Summary</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2765</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>827807126</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/827...</td>\n",
       "      <td>21599570.0</td>\n",
       "      <td>CYP2D6 *1/*1 is associated with increased clea...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2766</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>827807118</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/827...</td>\n",
       "      <td>21599570.0</td>\n",
       "      <td>CYP2D6 *1/*1 is associated with increased clea...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2767</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>827807123</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/827...</td>\n",
       "      <td>21599570.0</td>\n",
       "      <td>CYP2D6 *1/*1 is associated with increased clea...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2768</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>827807099</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/827...</td>\n",
       "      <td>21599570.0</td>\n",
       "      <td>CYP2D6 *1/*1 is associated with increased clea...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2769</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>982036882</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/982...</td>\n",
       "      <td>16849011.0</td>\n",
       "      <td>CYP2D6 *10 is associated with decreased metabo...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2770</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>982036911</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/982...</td>\n",
       "      <td>16849011.0</td>\n",
       "      <td>CYP2D6 *4 is not associated with metabolism of...</td>\n",
       "      <td>-0.125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2771</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>982036924</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/982...</td>\n",
       "      <td>16849011.0</td>\n",
       "      <td>CYP2D6 *5 is not associated with metabolism of...</td>\n",
       "      <td>-0.125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2772</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>982043452</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/982...</td>\n",
       "      <td>16595916.0</td>\n",
       "      <td>CYP2D6 *10/*10 is associated with decreased cl...</td>\n",
       "      <td>2.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2773</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>982044314</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/982...</td>\n",
       "      <td>17329852.0</td>\n",
       "      <td>CYP2D6 *1/*4 is associated with decreased clea...</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2774</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>982044321</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/982...</td>\n",
       "      <td>17329852.0</td>\n",
       "      <td>CYP2D6 *1/*5 is associated with decreased clea...</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2775</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>982044327</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/982...</td>\n",
       "      <td>17329852.0</td>\n",
       "      <td>CYP2D6 *10/*10 is associated with decreased cl...</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Clinical Annotation ID Evidence ID            Evidence Type  \\\n",
       "2765              1183615480   827807126  Variant Drug Annotation   \n",
       "2766              1183615480   827807118  Variant Drug Annotation   \n",
       "2767              1183615480   827807123  Variant Drug Annotation   \n",
       "2768              1183615480   827807099  Variant Drug Annotation   \n",
       "2769              1183615480   982036882  Variant Drug Annotation   \n",
       "2770              1183615480   982036911  Variant Drug Annotation   \n",
       "2771              1183615480   982036924  Variant Drug Annotation   \n",
       "2772              1183615480   982043452  Variant Drug Annotation   \n",
       "2773              1183615480   982044314  Variant Drug Annotation   \n",
       "2774              1183615480   982044321  Variant Drug Annotation   \n",
       "2775              1183615480   982044327  Variant Drug Annotation   \n",
       "\n",
       "                                           Evidence URL        PMID  \\\n",
       "2765  https://www.pharmgkb.org/variantAnnotation/827...  21599570.0   \n",
       "2766  https://www.pharmgkb.org/variantAnnotation/827...  21599570.0   \n",
       "2767  https://www.pharmgkb.org/variantAnnotation/827...  21599570.0   \n",
       "2768  https://www.pharmgkb.org/variantAnnotation/827...  21599570.0   \n",
       "2769  https://www.pharmgkb.org/variantAnnotation/982...  16849011.0   \n",
       "2770  https://www.pharmgkb.org/variantAnnotation/982...  16849011.0   \n",
       "2771  https://www.pharmgkb.org/variantAnnotation/982...  16849011.0   \n",
       "2772  https://www.pharmgkb.org/variantAnnotation/982...  16595916.0   \n",
       "2773  https://www.pharmgkb.org/variantAnnotation/982...  17329852.0   \n",
       "2774  https://www.pharmgkb.org/variantAnnotation/982...  17329852.0   \n",
       "2775  https://www.pharmgkb.org/variantAnnotation/982...  17329852.0   \n",
       "\n",
       "                                                Summary   Score  \n",
       "2765  CYP2D6 *1/*1 is associated with increased clea...       0  \n",
       "2766  CYP2D6 *1/*1 is associated with increased clea...       0  \n",
       "2767  CYP2D6 *1/*1 is associated with increased clea...       0  \n",
       "2768  CYP2D6 *1/*1 is associated with increased clea...       0  \n",
       "2769  CYP2D6 *10 is associated with decreased metabo...       1  \n",
       "2770  CYP2D6 *4 is not associated with metabolism of...  -0.125  \n",
       "2771  CYP2D6 *5 is not associated with metabolism of...  -0.125  \n",
       "2772  CYP2D6 *10/*10 is associated with decreased cl...    2.25  \n",
       "2773  CYP2D6 *1/*4 is associated with decreased clea...     1.5  \n",
       "2774  CYP2D6 *1/*5 is associated with decreased clea...     1.5  \n",
       "2775  CYP2D6 *10/*10 is associated with decreased cl...     1.5  "
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ev[ev[\"Clinical Annotation ID\"]==caid]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Clinical Annotation ID</th>\n",
       "      <th>Variant/Haplotypes</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Level of Evidence</th>\n",
       "      <th>Level Override</th>\n",
       "      <th>Level Modifiers</th>\n",
       "      <th>Score</th>\n",
       "      <th>Phenotype Category</th>\n",
       "      <th>PMID Count</th>\n",
       "      <th>Evidence Count</th>\n",
       "      <th>Drug(s)</th>\n",
       "      <th>Phenotype(s)</th>\n",
       "      <th>Latest History Date (YYYY-MM-DD)</th>\n",
       "      <th>URL</th>\n",
       "      <th>Specialty Population</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>875</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>CYP2D6*1, CYP2D6*2, CYP2D6*3, CYP2D6*4, CYP2D6...</td>\n",
       "      <td>CYP2D6</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tier 1 VIP</td>\n",
       "      <td>7.5</td>\n",
       "      <td>Metabolism/PK</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>carvedilol</td>\n",
       "      <td>Heart Diseases</td>\n",
       "      <td>22/04/2021</td>\n",
       "      <td>https://www.pharmgkb.org/clinicalAnnotation/11...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Clinical Annotation ID  \\\n",
       "875              1183615480   \n",
       "\n",
       "                                    Variant/Haplotypes    Gene  \\\n",
       "875  CYP2D6*1, CYP2D6*2, CYP2D6*3, CYP2D6*4, CYP2D6...  CYP2D6   \n",
       "\n",
       "    Level of Evidence Level Override Level Modifiers  Score  \\\n",
       "875                 3            NaN      Tier 1 VIP    7.5   \n",
       "\n",
       "    Phenotype Category  PMID Count  Evidence Count     Drug(s)  \\\n",
       "875      Metabolism/PK           4              11  carvedilol   \n",
       "\n",
       "       Phenotype(s) Latest History Date (YYYY-MM-DD)  \\\n",
       "875  Heart Diseases                       22/04/2021   \n",
       "\n",
       "                                                   URL Specialty Population  \n",
       "875  https://www.pharmgkb.org/clinicalAnnotation/11...                  NaN  "
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cl[cl[\"Clinical Annotation ID\"]==caid]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Clinical Annotation ID</th>\n",
       "      <th>Date (YYYY-MM-DD)</th>\n",
       "      <th>Type</th>\n",
       "      <th>Comment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1989</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>18/02/2017</td>\n",
       "      <td>Update</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1990</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>12/09/2018</td>\n",
       "      <td>Update</td>\n",
       "      <td>Removed inaccurate variant annotation on PMID ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1991</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>05/11/2018</td>\n",
       "      <td>Update</td>\n",
       "      <td>Updated OMB race to appropriate biogeographica...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1992</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>05/02/2021</td>\n",
       "      <td>Update</td>\n",
       "      <td>Added PMIDs 16849011 and 21599570 to evidence....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1993</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>24/03/2021</td>\n",
       "      <td>Update</td>\n",
       "      <td>CA score added as part of scoring system relea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1994</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>22/04/2021</td>\n",
       "      <td>Update</td>\n",
       "      <td>Added sentence about DPWG 'no recommendation' ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Clinical Annotation ID Date (YYYY-MM-DD)    Type  \\\n",
       "1989              1183615480        18/02/2017  Update   \n",
       "1990              1183615480        12/09/2018  Update   \n",
       "1991              1183615480        05/11/2018  Update   \n",
       "1992              1183615480        05/02/2021  Update   \n",
       "1993              1183615480        24/03/2021  Update   \n",
       "1994              1183615480        22/04/2021  Update   \n",
       "\n",
       "                                                Comment  \n",
       "1989                                                NaN  \n",
       "1990  Removed inaccurate variant annotation on PMID ...  \n",
       "1991  Updated OMB race to appropriate biogeographica...  \n",
       "1992  Added PMIDs 16849011 and 21599570 to evidence....  \n",
       "1993  CA score added as part of scoring system relea...  \n",
       "1994  Added sentence about DPWG 'no recommendation' ...  "
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "his[his[\"Clinical Annotation ID\"]==caid]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "#create temporal smaller files to work on:\n",
    "\n",
    "df = pd.read_csv(os.path.join(processed_path, \"clinical_annotation.csv\"))\n",
    "df = df.head(1000)\n",
    "df.to_csv(os.path.join(processed_path, \"clinann_tmp.csv\"), index=False)\n",
    "\n",
    "df = pd.read_csv(os.path.join(pharmgkb_path, \"clinicalAnnotations\", \"clinical_ann_alleles.csv\"))\n",
    "df = df.head(1000)\n",
    "df.to_csv(os.path.join(processed_path, \"clinann_all_tmp.csv\"), index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Clinical variant vs clinical Annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['caid', 'vid', 'variant', 'hid', 'haplotype', 'gene', 'evidence',\n",
      "       'score', 'phenotype', 'chemical', 'disease'],\n",
      "      dtype='object')\n",
      "Index(['vid', 'variant', 'hid', 'haplotype', 'gene', 'phenotype', 'evidence',\n",
      "       'chemical', 'disease'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "df1 = pd.read_csv(os.path.join(processed_path, \"clinical_annotation.csv\"))\n",
    "df2 = pd.read_csv(os.path.join(processed_path, \"clinical_variant.csv\"))\n",
    "\n",
    "print(df1.columns)\n",
    "print(df2.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11533\n"
     ]
    }
   ],
   "source": [
    "# Select the columns to compare for duplication\n",
    "columns_to_compare = ['vid', 'variant', 'hid', 'haplotype', 'gene', 'chemical', \"evidence\"]\n",
    "\n",
    "# Find duplicated rows in df1 based on the selected columns\n",
    "duplicated_rows_df1 = df1[df1.duplicated(subset=columns_to_compare, keep=False)]\n",
    "\n",
    "# Find duplicated rows in df2 based on the selected columns\n",
    "duplicated_rows_df2 = df2[df2.duplicated(subset=columns_to_compare, keep=False)]\n",
    "\n",
    "# Combine the duplicated rows from both dataframes\n",
    "duplicated_rows = pd.concat([duplicated_rows_df1, duplicated_rows_df2], ignore_index=True)\n",
    "\n",
    "# Print the duplicated rows\n",
    "print(len(duplicated_rows))\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Variant Annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Study Parameters ID', 'Variant Annotation ID', 'Study Type',\n",
      "       'Study Cases', 'Study Controls', 'Characteristics',\n",
      "       'Characteristics Type', 'Frequency In Cases',\n",
      "       'Allele Of Frequency In Cases', 'Frequency In Controls',\n",
      "       'Allele Of Frequency In Controls', 'P Value', 'Ratio Stat Type',\n",
      "       'Ratio Stat', 'Confidence Interval Start', 'Confidence Interval Stop',\n",
      "       'Biogeographical Groups'],\n",
      "      dtype='object')\n",
      "Index(['Variant Annotation ID', 'Variant/Haplotypes', 'Gene', 'Drug(s)',\n",
      "       'PMID', 'Phenotype Category', 'Significance', 'Notes', 'Sentence',\n",
      "       'Alleles', 'Specialty Population'],\n",
      "      dtype='object')\n",
      "Index(['Variant Annotation ID', 'Variant/Haplotypes', 'Gene', 'Drug(s)',\n",
      "       'PMID', 'Phenotype Category', 'Significance', 'Notes', 'Sentence',\n",
      "       'Alleles', 'Specialty Population'],\n",
      "      dtype='object')\n",
      "Index(['Variant Annotation ID', 'Variant/Haplotypes', 'Gene', 'Drug(s)',\n",
      "       'PMID', 'Phenotype Category', 'Significance', 'Notes', 'Sentence',\n",
      "       'Alleles', 'Specialty Population'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "df1 = pd.read_csv(os.path.join(pharmgkb_path, \"variantAnnotations\", \"study_parameters.csv\"))\n",
    "df2 = pd.read_csv(os.path.join(pharmgkb_path, \"variantAnnotations\", \"var_drug_ann.csv\"))\n",
    "df3 = pd.read_csv(os.path.join(pharmgkb_path, \"variantAnnotations\", \"var_fa_ann.csv\"))\n",
    "df4 = pd.read_csv(os.path.join(pharmgkb_path, \"variantAnnotations\", \"var_pheno_ann.csv\"), encoding= 'unicode_escape')\n",
    "\n",
    "print(df1.columns)\n",
    "print(df2.columns)\n",
    "print(df3.columns)\n",
    "print(df4.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32076 11222 1810 12844\n",
      "24993\n",
      "11104\n",
      "1157\n",
      "12732\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "#How do the different variant IDs relate to each other\n",
    "col = \"Variant Annotation ID\"\n",
    "\n",
    "print(len(df1), len(df2), len(df3), len(df4))\n",
    "print(len(set(df1[col])))\n",
    "print(len(set(df1[col]) & set(df2[col])))\n",
    "print(len(set(df1[col]) & set(df3[col])))\n",
    "print(len(set(df1[col]) & set(df4[col])))\n",
    "print(len(set(df2[col]) & set(df4[col])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "#Are Variant Annotation IDs different to Variant IDs? YES\n",
    "var = pd.read_csv(os.path.join(processed_path, \"variant.csv\"))\n",
    "print(len(set(df1[col]) & set(var[\"vid\"])))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "#Are Variant Annotation IDs different to Clinical Annotation IDs? YES\n",
    "clan = pd.read_csv(os.path.join(processed_path, \"clinical_annotation_allele.csv\"))\n",
    "print(len(set(df1[col]) & set(clan[\"caid\"])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10471\n",
      "10359\n"
     ]
    }
   ],
   "source": [
    "#Can we append the bioG group and study id to the var_drug_ann and var_pheno_ann files using vaid?\n",
    "df_ = pd.read_csv(os.path.join(processed_path, \"var_drug_ann.csv\"))\n",
    "print(len(set(df_[\"vaid\"])))\n",
    "print(len(list(set(df1[\"Variant Annotation ID\"])&set(df_[\"vaid\"]))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>caid</th>\n",
       "      <th>vid</th>\n",
       "      <th>variant</th>\n",
       "      <th>hid</th>\n",
       "      <th>haplotype</th>\n",
       "      <th>gene</th>\n",
       "      <th>evidence</th>\n",
       "      <th>score</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>chemical</th>\n",
       "      <th>disease</th>\n",
       "      <th>allele</th>\n",
       "      <th>anntext</th>\n",
       "      <th>function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7527</th>\n",
       "      <td>655385701</td>\n",
       "      <td>PA166157284</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>3</td>\n",
       "      <td>5.750</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>HIV Infections</td>\n",
       "      <td>AA</td>\n",
       "      <td>Patients with the AA genotype and HIV who are ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7528</th>\n",
       "      <td>655385701</td>\n",
       "      <td>PA166157284</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>3</td>\n",
       "      <td>5.750</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>HIV Infections</td>\n",
       "      <td>AG</td>\n",
       "      <td>Patients with the GA genotype and HIV who are ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7529</th>\n",
       "      <td>655385701</td>\n",
       "      <td>PA166157284</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>3</td>\n",
       "      <td>5.750</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>HIV Infections</td>\n",
       "      <td>GG</td>\n",
       "      <td>Patients with the GG genotype and HIV who are ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7530</th>\n",
       "      <td>655385701</td>\n",
       "      <td>PA166157284</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>3</td>\n",
       "      <td>5.750</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>HIV Infections</td>\n",
       "      <td>AA</td>\n",
       "      <td>Patients with the AA genotype and HIV who are ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7531</th>\n",
       "      <td>655385701</td>\n",
       "      <td>PA166157284</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>3</td>\n",
       "      <td>5.750</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>HIV Infections</td>\n",
       "      <td>AG</td>\n",
       "      <td>Patients with the GA genotype and HIV who are ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7532</th>\n",
       "      <td>655385701</td>\n",
       "      <td>PA166157284</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>3</td>\n",
       "      <td>5.750</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>HIV Infections</td>\n",
       "      <td>GG</td>\n",
       "      <td>Patients with the GG genotype and HIV who are ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10161</th>\n",
       "      <td>981238288</td>\n",
       "      <td>PA166157316</td>\n",
       "      <td>rs2032582</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>4</td>\n",
       "      <td>-0.250</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>HIV Infections</td>\n",
       "      <td>AA</td>\n",
       "      <td>Patients with the AA genotype may have decreas...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10162</th>\n",
       "      <td>981238288</td>\n",
       "      <td>PA166157316</td>\n",
       "      <td>rs2032582</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>4</td>\n",
       "      <td>-0.250</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>HIV Infections</td>\n",
       "      <td>AC</td>\n",
       "      <td>Patients with the AC genotype may have decreas...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10163</th>\n",
       "      <td>981238288</td>\n",
       "      <td>PA166157316</td>\n",
       "      <td>rs2032582</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>4</td>\n",
       "      <td>-0.250</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>HIV Infections</td>\n",
       "      <td>CC</td>\n",
       "      <td>Patients with the CC genotype may have increas...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13455</th>\n",
       "      <td>1184233711</td>\n",
       "      <td>PA166157284</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>3</td>\n",
       "      <td>0.500</td>\n",
       "      <td>Other</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>HIV Infections</td>\n",
       "      <td>AA</td>\n",
       "      <td>Patients with the AA genotype and HIV infectio...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13456</th>\n",
       "      <td>1184233711</td>\n",
       "      <td>PA166157284</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>3</td>\n",
       "      <td>0.500</td>\n",
       "      <td>Other</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>HIV Infections</td>\n",
       "      <td>AG</td>\n",
       "      <td>Patients with the AG genotype and HIV infectio...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13457</th>\n",
       "      <td>1184233711</td>\n",
       "      <td>PA166157284</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>3</td>\n",
       "      <td>0.500</td>\n",
       "      <td>Other</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>HIV Infections</td>\n",
       "      <td>GG</td>\n",
       "      <td>Patients with the GG genotype and HIV infectio...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13458</th>\n",
       "      <td>1184133889</td>\n",
       "      <td>PA166157239</td>\n",
       "      <td>rs3842</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>3</td>\n",
       "      <td>4.125</td>\n",
       "      <td>Other</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>HIV Infections</td>\n",
       "      <td>CC</td>\n",
       "      <td>Patients with CC genotype, and HIV infection, ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13459</th>\n",
       "      <td>1184133889</td>\n",
       "      <td>PA166157239</td>\n",
       "      <td>rs3842</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>3</td>\n",
       "      <td>4.125</td>\n",
       "      <td>Other</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>HIV Infections</td>\n",
       "      <td>CT</td>\n",
       "      <td>Patients with the CT genotype, and HIV infecti...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13460</th>\n",
       "      <td>1184133889</td>\n",
       "      <td>PA166157239</td>\n",
       "      <td>rs3842</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>3</td>\n",
       "      <td>4.125</td>\n",
       "      <td>Other</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>HIV Infections</td>\n",
       "      <td>TT</td>\n",
       "      <td>Patients with the TT genotype, and HIV infecti...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16658</th>\n",
       "      <td>1447943739</td>\n",
       "      <td>PA166157288</td>\n",
       "      <td>rs1128503</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>4</td>\n",
       "      <td>-0.625</td>\n",
       "      <td>Metabolism/PK</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>HIV Infections</td>\n",
       "      <td>AA</td>\n",
       "      <td>Patients with AA genotype and HIV may have dec...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16659</th>\n",
       "      <td>1447943739</td>\n",
       "      <td>PA166157288</td>\n",
       "      <td>rs1128503</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>4</td>\n",
       "      <td>-0.625</td>\n",
       "      <td>Metabolism/PK</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>HIV Infections</td>\n",
       "      <td>AG</td>\n",
       "      <td>Patients with AG genotype and HIV may have inc...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16660</th>\n",
       "      <td>1447943739</td>\n",
       "      <td>PA166157288</td>\n",
       "      <td>rs1128503</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>4</td>\n",
       "      <td>-0.625</td>\n",
       "      <td>Metabolism/PK</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>HIV Infections</td>\n",
       "      <td>GG</td>\n",
       "      <td>Patients with GG genotype and HIV may have inc...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             caid          vid    variant  hid haplotype   gene evidence  \\\n",
       "7527    655385701  PA166157284  rs1045642  NaN       NaN  ABCB1        3   \n",
       "7528    655385701  PA166157284  rs1045642  NaN       NaN  ABCB1        3   \n",
       "7529    655385701  PA166157284  rs1045642  NaN       NaN  ABCB1        3   \n",
       "7530    655385701  PA166157284  rs1045642  NaN       NaN  ABCB1        3   \n",
       "7531    655385701  PA166157284  rs1045642  NaN       NaN  ABCB1        3   \n",
       "7532    655385701  PA166157284  rs1045642  NaN       NaN  ABCB1        3   \n",
       "10161   981238288  PA166157316  rs2032582  NaN       NaN  ABCB1        4   \n",
       "10162   981238288  PA166157316  rs2032582  NaN       NaN  ABCB1        4   \n",
       "10163   981238288  PA166157316  rs2032582  NaN       NaN  ABCB1        4   \n",
       "13455  1184233711  PA166157284  rs1045642  NaN       NaN  ABCB1        3   \n",
       "13456  1184233711  PA166157284  rs1045642  NaN       NaN  ABCB1        3   \n",
       "13457  1184233711  PA166157284  rs1045642  NaN       NaN  ABCB1        3   \n",
       "13458  1184133889  PA166157239     rs3842  NaN       NaN  ABCB1        3   \n",
       "13459  1184133889  PA166157239     rs3842  NaN       NaN  ABCB1        3   \n",
       "13460  1184133889  PA166157239     rs3842  NaN       NaN  ABCB1        3   \n",
       "16658  1447943739  PA166157288  rs1128503  NaN       NaN  ABCB1        4   \n",
       "16659  1447943739  PA166157288  rs1128503  NaN       NaN  ABCB1        4   \n",
       "16660  1447943739  PA166157288  rs1128503  NaN       NaN  ABCB1        4   \n",
       "\n",
       "       score      phenotype   chemical         disease allele  \\\n",
       "7527   5.750       Efficacy  efavirenz  HIV Infections     AA   \n",
       "7528   5.750       Efficacy  efavirenz  HIV Infections     AG   \n",
       "7529   5.750       Efficacy  efavirenz  HIV Infections     GG   \n",
       "7530   5.750       Toxicity  efavirenz  HIV Infections     AA   \n",
       "7531   5.750       Toxicity  efavirenz  HIV Infections     AG   \n",
       "7532   5.750       Toxicity  efavirenz  HIV Infections     GG   \n",
       "10161 -0.250       Efficacy  efavirenz  HIV Infections     AA   \n",
       "10162 -0.250       Efficacy  efavirenz  HIV Infections     AC   \n",
       "10163 -0.250       Efficacy  efavirenz  HIV Infections     CC   \n",
       "13455  0.500          Other  efavirenz  HIV Infections     AA   \n",
       "13456  0.500          Other  efavirenz  HIV Infections     AG   \n",
       "13457  0.500          Other  efavirenz  HIV Infections     GG   \n",
       "13458  4.125          Other  efavirenz  HIV Infections     CC   \n",
       "13459  4.125          Other  efavirenz  HIV Infections     CT   \n",
       "13460  4.125          Other  efavirenz  HIV Infections     TT   \n",
       "16658 -0.625  Metabolism/PK  efavirenz  HIV Infections     AA   \n",
       "16659 -0.625  Metabolism/PK  efavirenz  HIV Infections     AG   \n",
       "16660 -0.625  Metabolism/PK  efavirenz  HIV Infections     GG   \n",
       "\n",
       "                                                 anntext function  \n",
       "7527   Patients with the AA genotype and HIV who are ...      NaN  \n",
       "7528   Patients with the GA genotype and HIV who are ...      NaN  \n",
       "7529   Patients with the GG genotype and HIV who are ...      NaN  \n",
       "7530   Patients with the AA genotype and HIV who are ...      NaN  \n",
       "7531   Patients with the GA genotype and HIV who are ...      NaN  \n",
       "7532   Patients with the GG genotype and HIV who are ...      NaN  \n",
       "10161  Patients with the AA genotype may have decreas...      NaN  \n",
       "10162  Patients with the AC genotype may have decreas...      NaN  \n",
       "10163  Patients with the CC genotype may have increas...      NaN  \n",
       "13455  Patients with the AA genotype and HIV infectio...      NaN  \n",
       "13456  Patients with the AG genotype and HIV infectio...      NaN  \n",
       "13457  Patients with the GG genotype and HIV infectio...      NaN  \n",
       "13458  Patients with CC genotype, and HIV infection, ...      NaN  \n",
       "13459  Patients with the CT genotype, and HIV infecti...      NaN  \n",
       "13460  Patients with the TT genotype, and HIV infecti...      NaN  \n",
       "16658  Patients with AA genotype and HIV may have dec...      NaN  \n",
       "16659  Patients with AG genotype and HIV may have inc...      NaN  \n",
       "16660  Patients with GG genotype and HIV may have inc...      NaN  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Compare Variant Annotations and Clinical Annotations\n",
    "clan[(clan[\"gene\"]==\"ABCB1\")&(clan[\"chemical\"]==\"efavirenz\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant Annotation ID</th>\n",
       "      <th>Variant/Haplotypes</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Drug(s)</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Phenotype Category</th>\n",
       "      <th>Significance</th>\n",
       "      <th>Notes</th>\n",
       "      <th>Sentence</th>\n",
       "      <th>Alleles</th>\n",
       "      <th>Specialty Population</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>655385693</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>16267764</td>\n",
       "      <td>Other</td>\n",
       "      <td>no</td>\n",
       "      <td>Although not significant, a trend for increasi...</td>\n",
       "      <td>Genotype AA is associated with increased CD4 T...</td>\n",
       "      <td>AA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>655385696</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>16267764</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>yes</td>\n",
       "      <td>Please note; 3435C &gt; T alleles were given in t...</td>\n",
       "      <td>Genotype AA is associated with increased likel...</td>\n",
       "      <td>AA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1407</th>\n",
       "      <td>1184168552</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>23734829</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>no</td>\n",
       "      <td>No association was seen between the alleles of...</td>\n",
       "      <td>Allele G is not associated with Neurotoxicity ...</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4032</th>\n",
       "      <td>827695550</td>\n",
       "      <td>rs2032582</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>16267764</td>\n",
       "      <td>Metabolism/PK</td>\n",
       "      <td>no</td>\n",
       "      <td>Genotypes GG, GT, TT, AG, AT (minus strand) we...</td>\n",
       "      <td>Genotypes AT + CT are not associated with incr...</td>\n",
       "      <td>AT + CT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5973</th>\n",
       "      <td>827695530</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>16267764</td>\n",
       "      <td>Metabolism/PK</td>\n",
       "      <td>no</td>\n",
       "      <td>Please note; 3435C &gt; T alleles were given in t...</td>\n",
       "      <td>Genotype AA is not associated with increased l...</td>\n",
       "      <td>AA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6061</th>\n",
       "      <td>827695457</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>16267764</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>yes</td>\n",
       "      <td>Please note; 3435C &gt; T alleles were given in t...</td>\n",
       "      <td>Allele A is associated with increased likeliho...</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8752</th>\n",
       "      <td>1448993831</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>26715213</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Genotype AA are associated with increased risk...</td>\n",
       "      <td>AA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10657</th>\n",
       "      <td>769163058</td>\n",
       "      <td>rs3842</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>20860463</td>\n",
       "      <td>Metabolism/PK</td>\n",
       "      <td>yes</td>\n",
       "      <td>PBMC accumulation was calculated by: (cell con...</td>\n",
       "      <td>Genotypes CC + CT are associated with increase...</td>\n",
       "      <td>CC + CT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10659</th>\n",
       "      <td>769163062</td>\n",
       "      <td>rs1128503</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>20860463</td>\n",
       "      <td>Metabolism/PK</td>\n",
       "      <td>no</td>\n",
       "      <td>No association with accumulation of efavirenz ...</td>\n",
       "      <td>Genotype AA is not associated with increased m...</td>\n",
       "      <td>AA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10662</th>\n",
       "      <td>769163063</td>\n",
       "      <td>rs2032582</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>20860463</td>\n",
       "      <td>Metabolism/PK</td>\n",
       "      <td>no</td>\n",
       "      <td>No association was found with efavirenz accumu...</td>\n",
       "      <td>Genotype AA is not associated with increased m...</td>\n",
       "      <td>AA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10664</th>\n",
       "      <td>769163065</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>20860463</td>\n",
       "      <td>Metabolism/PK</td>\n",
       "      <td>no</td>\n",
       "      <td>No significant association with drug accumulat...</td>\n",
       "      <td>Genotypes AA + AG are not associated with incr...</td>\n",
       "      <td>AA + AG</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11484</th>\n",
       "      <td>827695468</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>16267764</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>yes</td>\n",
       "      <td>Please note; 3435C &gt; T alleles were given in t...</td>\n",
       "      <td>Allele A is associated with decreased likeliho...</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11485</th>\n",
       "      <td>827695494</td>\n",
       "      <td>rs2032582</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>16267764</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>yes</td>\n",
       "      <td>The journal gave the alleles G &gt; T (minus stra...</td>\n",
       "      <td>Allele A is associated with decreased likeliho...</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12413</th>\n",
       "      <td>1448995896</td>\n",
       "      <td>rs3842</td>\n",
       "      <td>ABCB1</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>19916993</td>\n",
       "      <td>Metabolism/PK</td>\n",
       "      <td>yes</td>\n",
       "      <td>Efavirenz relative bioavailability was 26% hig...</td>\n",
       "      <td>Genotype CC is associated with increased bioav...</td>\n",
       "      <td>CC</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Variant Annotation ID Variant/Haplotypes   Gene    Drug(s)      PMID  \\\n",
       "13                 655385693          rs1045642  ABCB1  efavirenz  16267764   \n",
       "14                 655385696          rs1045642  ABCB1  efavirenz  16267764   \n",
       "1407              1184168552          rs1045642  ABCB1  efavirenz  23734829   \n",
       "4032               827695550          rs2032582  ABCB1  efavirenz  16267764   \n",
       "5973               827695530          rs1045642  ABCB1  efavirenz  16267764   \n",
       "6061               827695457          rs1045642  ABCB1  efavirenz  16267764   \n",
       "8752              1448993831          rs1045642  ABCB1  efavirenz  26715213   \n",
       "10657              769163058             rs3842  ABCB1  efavirenz  20860463   \n",
       "10659              769163062          rs1128503  ABCB1  efavirenz  20860463   \n",
       "10662              769163063          rs2032582  ABCB1  efavirenz  20860463   \n",
       "10664              769163065          rs1045642  ABCB1  efavirenz  20860463   \n",
       "11484              827695468          rs1045642  ABCB1  efavirenz  16267764   \n",
       "11485              827695494          rs2032582  ABCB1  efavirenz  16267764   \n",
       "12413             1448995896             rs3842  ABCB1  efavirenz  19916993   \n",
       "\n",
       "      Phenotype Category Significance  \\\n",
       "13                 Other           no   \n",
       "14              Efficacy          yes   \n",
       "1407            Toxicity           no   \n",
       "4032       Metabolism/PK           no   \n",
       "5973       Metabolism/PK           no   \n",
       "6061            Toxicity          yes   \n",
       "8752            Toxicity          yes   \n",
       "10657      Metabolism/PK          yes   \n",
       "10659      Metabolism/PK           no   \n",
       "10662      Metabolism/PK           no   \n",
       "10664      Metabolism/PK           no   \n",
       "11484           Efficacy          yes   \n",
       "11485           Efficacy          yes   \n",
       "12413      Metabolism/PK          yes   \n",
       "\n",
       "                                                   Notes  \\\n",
       "13     Although not significant, a trend for increasi...   \n",
       "14     Please note; 3435C > T alleles were given in t...   \n",
       "1407   No association was seen between the alleles of...   \n",
       "4032   Genotypes GG, GT, TT, AG, AT (minus strand) we...   \n",
       "5973   Please note; 3435C > T alleles were given in t...   \n",
       "6061   Please note; 3435C > T alleles were given in t...   \n",
       "8752                                                 NaN   \n",
       "10657  PBMC accumulation was calculated by: (cell con...   \n",
       "10659  No association with accumulation of efavirenz ...   \n",
       "10662  No association was found with efavirenz accumu...   \n",
       "10664  No significant association with drug accumulat...   \n",
       "11484  Please note; 3435C > T alleles were given in t...   \n",
       "11485  The journal gave the alleles G > T (minus stra...   \n",
       "12413  Efavirenz relative bioavailability was 26% hig...   \n",
       "\n",
       "                                                Sentence  Alleles  \\\n",
       "13     Genotype AA is associated with increased CD4 T...       AA   \n",
       "14     Genotype AA is associated with increased likel...       AA   \n",
       "1407   Allele G is not associated with Neurotoxicity ...        G   \n",
       "4032   Genotypes AT + CT are not associated with incr...  AT + CT   \n",
       "5973   Genotype AA is not associated with increased l...       AA   \n",
       "6061   Allele A is associated with increased likeliho...        A   \n",
       "8752   Genotype AA are associated with increased risk...       AA   \n",
       "10657  Genotypes CC + CT are associated with increase...  CC + CT   \n",
       "10659  Genotype AA is not associated with increased m...       AA   \n",
       "10662  Genotype AA is not associated with increased m...       AA   \n",
       "10664  Genotypes AA + AG are not associated with incr...  AA + AG   \n",
       "11484  Allele A is associated with decreased likeliho...        A   \n",
       "11485  Allele A is associated with decreased likeliho...        A   \n",
       "12413  Genotype CC is associated with increased bioav...       CC   \n",
       "\n",
       "      Specialty Population  \n",
       "13                     NaN  \n",
       "14                     NaN  \n",
       "1407                   NaN  \n",
       "4032                   NaN  \n",
       "5973                   NaN  \n",
       "6061                   NaN  \n",
       "8752                   NaN  \n",
       "10657                  NaN  \n",
       "10659                  NaN  \n",
       "10662                  NaN  \n",
       "10664                  NaN  \n",
       "11484                  NaN  \n",
       "11485                  NaN  \n",
       "12413                  NaN  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df4[(df4[\"Gene\"]==\"ABCB1\")&(df4[\"Drug(s)\"]==\"efavirenz\")]"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Haplotype Deconvolution\n",
    "For which genes is there information about the haplotype? Consider all that appear in:\n",
    "- Clinical Annotations\n",
    "- Variant Annotations\n",
    "- Haplotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df1 = pd.read_csv(os.path.join(processed_path, \"clinical_annotation.csv\"))\n",
    "df2 = pd.read_csv(os.path.join(processed_path, \"clinical_variant.csv\"))\n",
    "df3 = pd.read_csv(os.path.join(processed_path, \"var_drug_ann.csv\"))\n",
    "df4 = pd.read_csv(os.path.join(processed_path, \"var_pheno_ann.csv\"))\n",
    "df5 = pd.read_csv(os.path.join(processed_path, \"haplotype.csv\"))\n",
    "genes = (df1[\"gene\"].tolist()+df2[\"gene\"].tolist()+df3[\"gene\"].tolist()+df4[\"gene\"].tolist()+df5[\"gene\"].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_names = list(\n",
    "    set(\n",
    "        df1[\"gene\"][~df1[\"haplotype\"].isna()].tolist()\n",
    "        + df2[\"gene\"][~df2[\"haplotype\"].isna()].tolist()\n",
    "        + df3[\"gene\"][~df3[\"haplotype\"].isna()].tolist()\n",
    "        + df4[\"gene\"][~df4[\"haplotype\"].isna()].tolist()\n",
    "        + df5[\"gene\"].tolist()\n",
    "    )\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD B (reference)',\n",
       " 'G6PD Cairo',\n",
       " 'G6PD Dagua',\n",
       " 'G6PD Farroupilha',\n",
       " 'G6PD Gidra',\n",
       " 'G6PD Gond',\n",
       " 'G6PD Laibin',\n",
       " 'G6PD No name',\n",
       " 'G6PD Papua',\n",
       " 'G6PD Quing Yan',\n",
       " 'G6PD Sierra Leone',\n",
       " 'G6PD Sierra Leone',\n",
       " 'G6PD Tokyo, Fukushima',\n",
       " 'G6PD Yunan']"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(os.path.join(processed_path, \"haplotypes_gene\", \"G6PD_haplotypes.csv\"))\n",
    "df[\"haplotype\"][df[\"hid\"].isna()].tolist()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Drug Labels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['PharmGKB ID', 'Name', 'Source', 'Biomarker Flag', 'Testing Level',\n",
       "       'Has Prescribing Info', 'Has Dosing Info', 'Has Alternate Drug',\n",
       "       'Cancer Genome', 'Prescribing', 'Chemicals', 'Genes',\n",
       "       'Variants/Haplotypes', 'Latest History Date (YYYY-MM-DD)'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(os.path.join(pharmgkb_path, \"drugLabels\", \"drugLabels.csv\"))\n",
    "df.columns"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Automated annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Chemical ID', 'Chemical Name', 'Chemical in Text', 'Variation ID',\n",
       "       'Variation Name', 'Variation Type', 'Variation in Text', 'Gene IDs',\n",
       "       'Gene Symbols', 'Gene in Text', 'Literature ID', 'PMID',\n",
       "       'Literature Title', 'Publication Year', 'Journal', 'Sentence',\n",
       "       'Source'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(os.path.join(pharmgkb_path, \"automated_annotations\", \"automated_annotations.csv\"))\n",
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(20223, 17)\n",
      "(12865, 17)\n"
     ]
    }
   ],
   "source": [
    "#check how many do not have gene info\n",
    "print(df.shape)\n",
    "print(df[~df[\"Gene Symbols\"].isna()].shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chemical ID</th>\n",
       "      <th>Chemical Name</th>\n",
       "      <th>Chemical in Text</th>\n",
       "      <th>Variation ID</th>\n",
       "      <th>Variation Name</th>\n",
       "      <th>Variation Type</th>\n",
       "      <th>Variation in Text</th>\n",
       "      <th>Gene IDs</th>\n",
       "      <th>Gene Symbols</th>\n",
       "      <th>Gene in Text</th>\n",
       "      <th>Literature ID</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Literature Title</th>\n",
       "      <th>Publication Year</th>\n",
       "      <th>Journal</th>\n",
       "      <th>Sentence</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PA449028</td>\n",
       "      <td>clarithromycin</td>\n",
       "      <td>clarithromycin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Variant</td>\n",
       "      <td>A2146G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31528091</td>\n",
       "      <td>Review of current diagnostic methods and advan...</td>\n",
       "      <td>2019</td>\n",
       "      <td>World Journal of Gastroenterology</td>\n",
       "      <td>Moreover, they can distinguish the three most ...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PA448515</td>\n",
       "      <td>azathioprine</td>\n",
       "      <td>azathioprine</td>\n",
       "      <td>rs10889677</td>\n",
       "      <td>rs10889677</td>\n",
       "      <td>Variant</td>\n",
       "      <td>rs10889677</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28210080</td>\n",
       "      <td>IL23R  single nucleotide polymorphisms could b...</td>\n",
       "      <td>2017</td>\n",
       "      <td>World Journal of Gastroenterology</td>\n",
       "      <td>For patients with the rs10889677, a therapy wi...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PA448515</td>\n",
       "      <td>azathioprine</td>\n",
       "      <td>azathioprine</td>\n",
       "      <td>rs10889677</td>\n",
       "      <td>rs10889677</td>\n",
       "      <td>Variant</td>\n",
       "      <td>rs10889677</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28210080</td>\n",
       "      <td>IL23R  single nucleotide polymorphisms could b...</td>\n",
       "      <td>2017</td>\n",
       "      <td>World Journal of Gastroenterology</td>\n",
       "      <td>The rs10889677 was associated with azathioprin...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PA448515</td>\n",
       "      <td>azathioprine</td>\n",
       "      <td>azathioprine</td>\n",
       "      <td>rs10889677</td>\n",
       "      <td>rs10889677</td>\n",
       "      <td>Variant</td>\n",
       "      <td>rs10889677</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28210080</td>\n",
       "      <td>IL23R  single nucleotide polymorphisms could b...</td>\n",
       "      <td>2017</td>\n",
       "      <td>World Journal of Gastroenterology</td>\n",
       "      <td>The rs10889677 was associated with azathioprin...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PA452639</td>\n",
       "      <td>infliximab</td>\n",
       "      <td>infliximab</td>\n",
       "      <td>rs2522057</td>\n",
       "      <td>rs2522057</td>\n",
       "      <td>Variant</td>\n",
       "      <td>rs2522057</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28785150</td>\n",
       "      <td>Genetic polymorphisms predict response to anti...</td>\n",
       "      <td>2017</td>\n",
       "      <td>World Journal of Gastroenterology</td>\n",
       "      <td>A SNP in this cluster (rs2522057) may be assoc...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PA166122590</td>\n",
       "      <td>simeprevir</td>\n",
       "      <td>simeprevir</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Variant</td>\n",
       "      <td>Q80K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27920481</td>\n",
       "      <td>New real-time-PCR method to identify single po...</td>\n",
       "      <td>2016</td>\n",
       "      <td>World Journal of Gastroenterology</td>\n",
       "      <td>One such naturally occurring mutation is the Q...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PA166122590</td>\n",
       "      <td>simeprevir</td>\n",
       "      <td>simeprevir</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Variant</td>\n",
       "      <td>Q80K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27920481</td>\n",
       "      <td>New real-time-PCR method to identify single po...</td>\n",
       "      <td>2016</td>\n",
       "      <td>World Journal of Gastroenterology</td>\n",
       "      <td>One well-documented example is the Q80K polymo...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PA166128167</td>\n",
       "      <td>daclatasvir</td>\n",
       "      <td>daclatasvir</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Variant</td>\n",
       "      <td>Y93H</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27920481</td>\n",
       "      <td>New real-time-PCR method to identify single po...</td>\n",
       "      <td>2016</td>\n",
       "      <td>World Journal of Gastroenterology</td>\n",
       "      <td>We are now working on adapting this method for...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PA166122590</td>\n",
       "      <td>simeprevir</td>\n",
       "      <td>simeprevir</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Variant</td>\n",
       "      <td>Q80K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27920481</td>\n",
       "      <td>New real-time-PCR method to identify single po...</td>\n",
       "      <td>2016</td>\n",
       "      <td>World Journal of Gastroenterology</td>\n",
       "      <td>In addition, Q80K seems to be a natural varian...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PA449028</td>\n",
       "      <td>clarithromycin</td>\n",
       "      <td>clarithromycin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Variant</td>\n",
       "      <td>A2142G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28465629</td>\n",
       "      <td>Noninvasive molecular analysis of Helicobacter...</td>\n",
       "      <td>2017</td>\n",
       "      <td>World Journal of Gastroenterology</td>\n",
       "      <td>A first noninvasive investigation using stool ...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>PA449009</td>\n",
       "      <td>ciprofloxacin</td>\n",
       "      <td>ciprofloxacin</td>\n",
       "      <td>rs778523219</td>\n",
       "      <td>rs778523219</td>\n",
       "      <td>Variant</td>\n",
       "      <td>Ala-70-Thr</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23406779</td>\n",
       "      <td>Resistance mechanisms in Campylobacter jejuni</td>\n",
       "      <td>2013</td>\n",
       "      <td>Virulence</td>\n",
       "      <td>Perhaps since only a single point mutation is ...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PA449009</td>\n",
       "      <td>ciprofloxacin</td>\n",
       "      <td>ciprofloxacin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Variant</td>\n",
       "      <td>Asp-90-Asn</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23406779</td>\n",
       "      <td>Resistance mechanisms in Campylobacter jejuni</td>\n",
       "      <td>2013</td>\n",
       "      <td>Virulence</td>\n",
       "      <td>Perhaps since only a single point mutation is ...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>PA451241</td>\n",
       "      <td>ribavirin</td>\n",
       "      <td>ribavirin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Variant</td>\n",
       "      <td>G1634R</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28727933</td>\n",
       "      <td>Genotype-specific acquisition, evolution and a...</td>\n",
       "      <td>2017</td>\n",
       "      <td>Virulence</td>\n",
       "      <td>Among these mutations, G1634R was identified i...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>PA450721</td>\n",
       "      <td>oseltamivir</td>\n",
       "      <td>oseltamivir</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Variant</td>\n",
       "      <td>H274Y</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28694997</td>\n",
       "      <td>Antiviral drug resistance as an adaptive process</td>\n",
       "      <td>2016</td>\n",
       "      <td>Virus Evolution</td>\n",
       "      <td>The popular NA inhibitor oseltamivir has been ...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PA450721</td>\n",
       "      <td>oseltamivir</td>\n",
       "      <td>oseltamivir</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Variant</td>\n",
       "      <td>H274Y</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28694997</td>\n",
       "      <td>Antiviral drug resistance as an adaptive process</td>\n",
       "      <td>2016</td>\n",
       "      <td>Virus Evolution</td>\n",
       "      <td>For example, the H274Y mutation in influenza (...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>PA166122593</td>\n",
       "      <td>sofosbuvir</td>\n",
       "      <td>sofosbuvir</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Variant</td>\n",
       "      <td>S282T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29362671</td>\n",
       "      <td>The global origins of resistance-associated va...</td>\n",
       "      <td>2018</td>\n",
       "      <td>Virus Evolution</td>\n",
       "      <td>The S282T RAV is the only variant to confer re...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>PA166122593</td>\n",
       "      <td>sofosbuvir</td>\n",
       "      <td>sofosbuvir</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Variant</td>\n",
       "      <td>S282T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29362671</td>\n",
       "      <td>The global origins of resistance-associated va...</td>\n",
       "      <td>2018</td>\n",
       "      <td>Virus Evolution</td>\n",
       "      <td>Interestingly, the S282T RAV, the only known R...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>PA166163411</td>\n",
       "      <td>dasabuvir</td>\n",
       "      <td>dasabuvir</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Variant</td>\n",
       "      <td>C316N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29362671</td>\n",
       "      <td>The global origins of resistance-associated va...</td>\n",
       "      <td>2018</td>\n",
       "      <td>Virus Evolution</td>\n",
       "      <td>This relationship could confer a higher probab...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PA450721</td>\n",
       "      <td>oseltamivir</td>\n",
       "      <td>oseltamivir</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Variant</td>\n",
       "      <td>H275Y</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24078044</td>\n",
       "      <td>Molecular evolution of the hemagglutinin and n...</td>\n",
       "      <td>2013</td>\n",
       "      <td>Virus Genes</td>\n",
       "      <td>None of the isolates was positive for the know...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PA164740891</td>\n",
       "      <td>zanamivir</td>\n",
       "      <td>zanamivir</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Variant</td>\n",
       "      <td>R152K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24078044</td>\n",
       "      <td>Molecular evolution of the hemagglutinin and n...</td>\n",
       "      <td>2013</td>\n",
       "      <td>Virus Genes</td>\n",
       "      <td>The zanamivir resistance mutation R152K was al...</td>\n",
       "      <td>PGxMine</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Chemical ID   Chemical Name Chemical in Text Variation ID Variation Name  \\\n",
       "0      PA449028  clarithromycin   clarithromycin          NaN            NaN   \n",
       "5      PA448515    azathioprine     azathioprine   rs10889677     rs10889677   \n",
       "6      PA448515    azathioprine     azathioprine   rs10889677     rs10889677   \n",
       "7      PA448515    azathioprine     azathioprine   rs10889677     rs10889677   \n",
       "8      PA452639      infliximab       infliximab    rs2522057      rs2522057   \n",
       "10  PA166122590      simeprevir       simeprevir          NaN            NaN   \n",
       "11  PA166122590      simeprevir       simeprevir          NaN            NaN   \n",
       "12  PA166128167     daclatasvir      daclatasvir          NaN            NaN   \n",
       "13  PA166122590      simeprevir       simeprevir          NaN            NaN   \n",
       "14     PA449028  clarithromycin   clarithromycin          NaN            NaN   \n",
       "16     PA449009   ciprofloxacin    ciprofloxacin  rs778523219    rs778523219   \n",
       "17     PA449009   ciprofloxacin    ciprofloxacin          NaN            NaN   \n",
       "18     PA451241       ribavirin        ribavirin          NaN            NaN   \n",
       "19     PA450721     oseltamivir      oseltamivir          NaN            NaN   \n",
       "20     PA450721     oseltamivir      oseltamivir          NaN            NaN   \n",
       "21  PA166122593      sofosbuvir       sofosbuvir          NaN            NaN   \n",
       "22  PA166122593      sofosbuvir       sofosbuvir          NaN            NaN   \n",
       "23  PA166163411       dasabuvir        dasabuvir          NaN            NaN   \n",
       "24     PA450721     oseltamivir      oseltamivir          NaN            NaN   \n",
       "25  PA164740891       zanamivir        zanamivir          NaN            NaN   \n",
       "\n",
       "   Variation Type Variation in Text Gene IDs Gene Symbols  Gene in Text  \\\n",
       "0         Variant            A2146G      NaN          NaN           NaN   \n",
       "5         Variant        rs10889677      NaN          NaN           NaN   \n",
       "6         Variant        rs10889677      NaN          NaN           NaN   \n",
       "7         Variant        rs10889677      NaN          NaN           NaN   \n",
       "8         Variant         rs2522057      NaN          NaN           NaN   \n",
       "10        Variant              Q80K      NaN          NaN           NaN   \n",
       "11        Variant              Q80K      NaN          NaN           NaN   \n",
       "12        Variant              Y93H      NaN          NaN           NaN   \n",
       "13        Variant              Q80K      NaN          NaN           NaN   \n",
       "14        Variant            A2142G      NaN          NaN           NaN   \n",
       "16        Variant        Ala-70-Thr      NaN          NaN           NaN   \n",
       "17        Variant        Asp-90-Asn      NaN          NaN           NaN   \n",
       "18        Variant            G1634R      NaN          NaN           NaN   \n",
       "19        Variant             H274Y      NaN          NaN           NaN   \n",
       "20        Variant             H274Y      NaN          NaN           NaN   \n",
       "21        Variant             S282T      NaN          NaN           NaN   \n",
       "22        Variant             S282T      NaN          NaN           NaN   \n",
       "23        Variant             C316N      NaN          NaN           NaN   \n",
       "24        Variant             H275Y      NaN          NaN           NaN   \n",
       "25        Variant             R152K      NaN          NaN           NaN   \n",
       "\n",
       "    Literature ID      PMID  \\\n",
       "0             NaN  31528091   \n",
       "5             NaN  28210080   \n",
       "6             NaN  28210080   \n",
       "7             NaN  28210080   \n",
       "8             NaN  28785150   \n",
       "10            NaN  27920481   \n",
       "11            NaN  27920481   \n",
       "12            NaN  27920481   \n",
       "13            NaN  27920481   \n",
       "14            NaN  28465629   \n",
       "16            NaN  23406779   \n",
       "17            NaN  23406779   \n",
       "18            NaN  28727933   \n",
       "19            NaN  28694997   \n",
       "20            NaN  28694997   \n",
       "21            NaN  29362671   \n",
       "22            NaN  29362671   \n",
       "23            NaN  29362671   \n",
       "24            NaN  24078044   \n",
       "25            NaN  24078044   \n",
       "\n",
       "                                     Literature Title  Publication Year  \\\n",
       "0   Review of current diagnostic methods and advan...              2019   \n",
       "5   IL23R  single nucleotide polymorphisms could b...              2017   \n",
       "6   IL23R  single nucleotide polymorphisms could b...              2017   \n",
       "7   IL23R  single nucleotide polymorphisms could b...              2017   \n",
       "8   Genetic polymorphisms predict response to anti...              2017   \n",
       "10  New real-time-PCR method to identify single po...              2016   \n",
       "11  New real-time-PCR method to identify single po...              2016   \n",
       "12  New real-time-PCR method to identify single po...              2016   \n",
       "13  New real-time-PCR method to identify single po...              2016   \n",
       "14  Noninvasive molecular analysis of Helicobacter...              2017   \n",
       "16      Resistance mechanisms in Campylobacter jejuni              2013   \n",
       "17      Resistance mechanisms in Campylobacter jejuni              2013   \n",
       "18  Genotype-specific acquisition, evolution and a...              2017   \n",
       "19   Antiviral drug resistance as an adaptive process              2016   \n",
       "20   Antiviral drug resistance as an adaptive process              2016   \n",
       "21  The global origins of resistance-associated va...              2018   \n",
       "22  The global origins of resistance-associated va...              2018   \n",
       "23  The global origins of resistance-associated va...              2018   \n",
       "24  Molecular evolution of the hemagglutinin and n...              2013   \n",
       "25  Molecular evolution of the hemagglutinin and n...              2013   \n",
       "\n",
       "                              Journal  \\\n",
       "0   World Journal of Gastroenterology   \n",
       "5   World Journal of Gastroenterology   \n",
       "6   World Journal of Gastroenterology   \n",
       "7   World Journal of Gastroenterology   \n",
       "8   World Journal of Gastroenterology   \n",
       "10  World Journal of Gastroenterology   \n",
       "11  World Journal of Gastroenterology   \n",
       "12  World Journal of Gastroenterology   \n",
       "13  World Journal of Gastroenterology   \n",
       "14  World Journal of Gastroenterology   \n",
       "16                          Virulence   \n",
       "17                          Virulence   \n",
       "18                          Virulence   \n",
       "19                    Virus Evolution   \n",
       "20                    Virus Evolution   \n",
       "21                    Virus Evolution   \n",
       "22                    Virus Evolution   \n",
       "23                    Virus Evolution   \n",
       "24                        Virus Genes   \n",
       "25                        Virus Genes   \n",
       "\n",
       "                                             Sentence   Source  \n",
       "0   Moreover, they can distinguish the three most ...  PGxMine  \n",
       "5   For patients with the rs10889677, a therapy wi...  PGxMine  \n",
       "6   The rs10889677 was associated with azathioprin...  PGxMine  \n",
       "7   The rs10889677 was associated with azathioprin...  PGxMine  \n",
       "8   A SNP in this cluster (rs2522057) may be assoc...  PGxMine  \n",
       "10  One such naturally occurring mutation is the Q...  PGxMine  \n",
       "11  One well-documented example is the Q80K polymo...  PGxMine  \n",
       "12  We are now working on adapting this method for...  PGxMine  \n",
       "13  In addition, Q80K seems to be a natural varian...  PGxMine  \n",
       "14  A first noninvasive investigation using stool ...  PGxMine  \n",
       "16  Perhaps since only a single point mutation is ...  PGxMine  \n",
       "17  Perhaps since only a single point mutation is ...  PGxMine  \n",
       "18  Among these mutations, G1634R was identified i...  PGxMine  \n",
       "19  The popular NA inhibitor oseltamivir has been ...  PGxMine  \n",
       "20  For example, the H274Y mutation in influenza (...  PGxMine  \n",
       "21  The S282T RAV is the only variant to confer re...  PGxMine  \n",
       "22  Interestingly, the S282T RAV, the only known R...  PGxMine  \n",
       "23  This relationship could confer a higher probab...  PGxMine  \n",
       "24  None of the isolates was positive for the know...  PGxMine  \n",
       "25  The zanamivir resistance mutation R152K was al...  PGxMine  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df[\"Gene Symbols\"].isna()].head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pgx",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.0"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
