{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import pandas as pd\n",
    "\n",
    "sys.path.append(\"../src\")\n",
    "from pharmgkb import RawData\n",
    "\n",
    "\n",
    "data_path = os.path.join(\"..\", \"data\")\n",
    "pharmgkb_path = os.path.join(data_path, \"pharmgkb\")\n",
    "processed_path = os.path.join(data_path, \"pharmgkb_processed\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quick and dirty exploration of files for curation"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Chemicals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(os.path.join(pharmgkb_path, \"chemicals\", \"chemicals.csv\"))\n",
    "for i,c in enumerate(df.columns):\n",
    "    print(i, c)\n",
    "\n",
    "df[\"Top FDA Label Testing Level\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[df[\"Name\"]==\"apremilast\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Retrieve SMILES for PubChem compounds\n",
    "import requests\n",
    "\n",
    "data = requests.get(\"https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/cid/59272813/property/CanonicalSMILES/TXT\")\n",
    "smi = data.text.strip()\n",
    "smi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sys.path.append(os.path.join(\"../src\"))\n",
    "\n",
    "from utils import CsvCleaner\n",
    "import requests\n",
    "\n",
    "c = CsvCleaner()\n",
    "\n",
    "data_dict = {}\n",
    "for r in df.values:\n",
    "    chemical = c.stringify(r[1])\n",
    "    crossr = c.inline_comma_splitter(r[6])\n",
    "    if crossr is not None:\n",
    "        for i in crossr:\n",
    "            print(crossr)\n",
    "            if len(crossr) == 1:\n",
    "                if \"PubChem Compound\" in i:\n",
    "                    cpd = (i.split(\":\")[-1])\n",
    "                    data = requests.get(\"https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/cid/{}/property/CanonicalSMILES/TXT\".format(cpd))\n",
    "                    smi = data.text.strip()\n",
    "                    print(smi)\n",
    "            else:\n",
    "                if \"PubChem Compound\" in i:\n",
    "                    cpd = (i.split(\":\")[-1][:-1])\n",
    "                    data = requests.get(\"https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/cid/{}/property/CanonicalSMILES/TXT\".format(cpd))\n",
    "                    smi = data.text.strip()\n",
    "                    print(smi)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(4532, 6)\n",
      "1756\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv(os.path.join(processed_path, \"chemical.csv\"))\n",
    "print(df.shape)\n",
    "print(len(df[df[\"smiles\"].isna()]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cid</th>\n",
       "      <th>chemical</th>\n",
       "      <th>chemical_type</th>\n",
       "      <th>smiles</th>\n",
       "      <th>dosing_guideline</th>\n",
       "      <th>drug_label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>PA151958637</td>\n",
       "      <td>4-methylthioamphetamine</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>PA154081778</td>\n",
       "      <td>4-methylumbelliferone</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>PA151958362</td>\n",
       "      <td>6,7-dihydroxybergamottin</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270</th>\n",
       "      <td>PA152530740</td>\n",
       "      <td>8-cyclopentyl-1,3-dipropylxanthine</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>PA164747080</td>\n",
       "      <td>abatacept</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>PA448006</td>\n",
       "      <td>abciximab</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>298</th>\n",
       "      <td>PA166104276</td>\n",
       "      <td>ABT-751</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313</th>\n",
       "      <td>PA448014</td>\n",
       "      <td>acetamide mea</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315</th>\n",
       "      <td>PA166246281</td>\n",
       "      <td>acetaminophen / caffeine / dihydrocodeine</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320</th>\n",
       "      <td>PA165290927</td>\n",
       "      <td>acetaminophen/propoxyphene napsylate</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>322</th>\n",
       "      <td>PA166184496</td>\n",
       "      <td>acetaminophen / tramadol</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>323</th>\n",
       "      <td>PA448016</td>\n",
       "      <td>acetanilide</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>330</th>\n",
       "      <td>PA448025</td>\n",
       "      <td>acetone</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>354</th>\n",
       "      <td>PA10004</td>\n",
       "      <td>adalimumab</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369</th>\n",
       "      <td>PA165981139</td>\n",
       "      <td>afatinib dimaleate</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>370</th>\n",
       "      <td>PA452609</td>\n",
       "      <td>aflatoxin b1</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>371</th>\n",
       "      <td>PA166118342</td>\n",
       "      <td>aflibercept</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373</th>\n",
       "      <td>PA166122586</td>\n",
       "      <td>afutuzumab</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>374</th>\n",
       "      <td>PA164746527</td>\n",
       "      <td>agalsidase beta</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>388</th>\n",
       "      <td>PA448081</td>\n",
       "      <td>aldesleukin</td>\n",
       "      <td>Drug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             cid                                   chemical chemical_type  \\\n",
       "179  PA151958637                    4-methylthioamphetamine          Drug   \n",
       "180  PA154081778                      4-methylumbelliferone          Drug   \n",
       "222  PA151958362                   6,7-dihydroxybergamottin          Drug   \n",
       "270  PA152530740         8-cyclopentyl-1,3-dipropylxanthine          Drug   \n",
       "293  PA164747080                                  abatacept          Drug   \n",
       "294     PA448006                                  abciximab          Drug   \n",
       "298  PA166104276                                    ABT-751          Drug   \n",
       "313     PA448014                              acetamide mea          Drug   \n",
       "315  PA166246281  acetaminophen / caffeine / dihydrocodeine          Drug   \n",
       "320  PA165290927       acetaminophen/propoxyphene napsylate          Drug   \n",
       "322  PA166184496                   acetaminophen / tramadol          Drug   \n",
       "323     PA448016                                acetanilide          Drug   \n",
       "330     PA448025                                    acetone          Drug   \n",
       "354      PA10004                                 adalimumab          Drug   \n",
       "369  PA165981139                         afatinib dimaleate          Drug   \n",
       "370     PA452609                               aflatoxin b1          Drug   \n",
       "371  PA166118342                                aflibercept          Drug   \n",
       "373  PA166122586                                 afutuzumab          Drug   \n",
       "374  PA164746527                            agalsidase beta          Drug   \n",
       "388     PA448081                                aldesleukin          Drug   \n",
       "\n",
       "    smiles  dosing_guideline  drug_label  \n",
       "179    NaN                -1           0  \n",
       "180    NaN                -1           0  \n",
       "222    NaN                -1           0  \n",
       "270    NaN                -1           0  \n",
       "293    NaN                -1           0  \n",
       "294    NaN                -1           0  \n",
       "298    NaN                -1           0  \n",
       "313    NaN                -1           0  \n",
       "315    NaN                -1           3  \n",
       "320    NaN                -1           0  \n",
       "322    NaN                -1           3  \n",
       "323    NaN                -1           0  \n",
       "330    NaN                -1           0  \n",
       "354    NaN                -1           0  \n",
       "369    NaN                -1           0  \n",
       "370    NaN                -1           0  \n",
       "371    NaN                -1           0  \n",
       "373    NaN                -1           4  \n",
       "374    NaN                -1           0  \n",
       "388    NaN                -1           0  "
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[(df[\"smiles\"].isna())&(df[\"chemical_type\"]==\"Drug\")].head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: 404\n",
      "Code: PUGREST.NotFound\n",
      "Message: No CID found\n",
      "Detail: No CID found that matches the given name\n"
     ]
    }
   ],
   "source": [
    "data = requests.get(\"https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/name/adalimumab/property/CanonicalSMILES/TXT\")\n",
    "smi = data.text.strip()\n",
    "print(smi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "yes\n"
     ]
    }
   ],
   "source": [
    "if \"Status: 404\" in smi:\n",
    "    print(\"yes\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 PharmGKB Accession Id\n",
      "1 NCBI Gene ID\n",
      "2 HGNC ID\n",
      "3 Ensembl Id\n",
      "4 Name\n",
      "5 Symbol\n",
      "6 Alternate Names\n",
      "7 Alternate Symbols\n",
      "8 Is VIP\n",
      "9 Has Variant Annotation\n",
      "10 Cross-references\n",
      "11 Has CPIC Dosing Guideline\n",
      "12 Chromosome\n",
      "13 Chromosomal Start - GRCh37\n",
      "14 Chromosomal Stop - GRCh37\n",
      "15 Chromosomal Start - GRCh38\n",
      "16 Chromosomal Stop - GRCh38\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv(os.path.join(pharmgkb_path, \"genes\", \"genes.csv\"),encoding= 'unicode_escape')\n",
    "for i,c in enumerate(df.columns):\n",
    "    print(i, c)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Disease"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 PharmGKB Accession Id\n",
      "1 Name\n",
      "2 Alternate Names\n",
      "3 Cross-references\n",
      "4 External Vocabulary\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv(os.path.join(pharmgkb_path, \"phenotypes\", \"phenotypes.csv\"),encoding= 'unicode_escape')\n",
    "for i,c in enumerate(df.columns):\n",
    "    print(i, c)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 Variant ID\n",
      "1 Variant Name\n",
      "2 Gene IDs\n",
      "3 Gene Symbols\n",
      "4 Location\n",
      "5 Variant Annotation count\n",
      "6 Clinical Annotation count\n",
      "7 Level 1/2 Clinical Annotation count\n",
      "8 Guideline Annotation count\n",
      "9 Label Annotation count\n",
      "10 Synonyms\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv(os.path.join(pharmgkb_path, \"variants\", \"variants.csv\"),encoding= 'unicode_escape')\n",
    "for i,c in enumerate(df.columns):\n",
    "    print(i, c)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Haplotypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 Entity1_id\n",
      "1 Entity1_name\n",
      "2 Entity1_type\n",
      "3 Entity2_id\n",
      "4 Entity2_name\n",
      "5 Entity2_type\n",
      "6 Evidence\n",
      "7 Association\n",
      "8 PK\n",
      "9 PD\n",
      "10 PMIDs\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv(os.path.join(pharmgkb_path, \"relationships\", \"relationships.csv\"),encoding= 'unicode_escape')\n",
    "for i,c in enumerate(df.columns):\n",
    "    print(i, c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#We keep all the haplotypes that appear, either in entity 1 or in entity 2, and then get the gene from the Haplotype name\n",
    "df1 = df[\"Entity1_name\"][df[\"Entity1_type\"]==\"Haplotype\"].tolist()\n",
    "print(len(df1))\n",
    "df2 = df[\"Entity2_name\"][df[\"Entity2_type\"]==\"Haplotype\"].tolist()\n",
    "print(len(df2))\n",
    "print(len(set(df1+df2)))\n",
    "\n",
    "#get unique hids for either entity 1 or entity2 -- is the same\n",
    "haps = list(set(df1+df2))\n",
    "hids = []\n",
    "for r in df.values:\n",
    "    hap = r[1]\n",
    "    hid = r[0]\n",
    "    if hap in haps:\n",
    "        hids += [hid]\n",
    "len(hids)\n",
    "len(set(hids))\n",
    "\n",
    "haps = list(set(df1+df2))\n",
    "hids = []\n",
    "for r in df.values:\n",
    "    hap = r[4]\n",
    "    hid = r[3]\n",
    "    if hap in haps:\n",
    "        hids += [hid]\n",
    "len(hids)\n",
    "len(set(hids))\n",
    "hids = list(set(hids))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(11788, 11)\n",
      "(1028, 11)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_10338/925833839.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df1.drop_duplicates(subset= [\"Entity1_name\"], keep=\"first\", inplace=True)\n"
     ]
    }
   ],
   "source": [
    "#We keep all the haplotypes that appear, either in entity 1 or in entity 2, and then get the gene from the Haplotype name\n",
    "df1 = df[df[\"Entity1_type\"]==\"Haplotype\"]\n",
    "print(df1.shape)\n",
    "df1.drop_duplicates(subset= [\"Entity1_name\"], keep=\"first\", inplace=True)\n",
    "print(df1.shape)\n",
    "df1 = df1[[\"Entity1_id\", \"Entity1_name\"]]\n",
    "df1.rename(columns={\"Entity1_id\": \"hid\", \"Entity1_name\":\"haplotype\"}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = []\n",
    "for h in df1[\"haplotype\"]:\n",
    "    if \" \" in h:\n",
    "        g = h.split(\" \")[0]\n",
    "    elif \"*\" in h:\n",
    "        g = h.split(\"*\")[0]\n",
    "    else:\n",
    "        g = h\n",
    "    gene += [g]\n",
    "df1[\"gene\"]= gene\n",
    "gl = list(set(df1[\"gene\"].tolist()))\n",
    "gl"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Clincal variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(os.path.join(pharmgkb_path, \"clinicalVariants\", \"clinicalVariants.csv\"),encoding= 'unicode_escape')\n",
    "for i,c in enumerate(df.columns):\n",
    "    print(i, c)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Clinical Annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alleles\n",
      "0 Clinical Annotation ID\n",
      "1 Genotype/Allele\n",
      "2 Annotation Text\n",
      "3 Allele Function\n",
      "4 test\n",
      "Evidence\n",
      "0 Clinical Annotation ID\n",
      "1 Evidence ID\n",
      "2 Evidence Type\n",
      "3 Evidence URL\n",
      "4 PMID\n",
      "5 Summary\n",
      "6 Score\n",
      "History\n",
      "0 Clinical Annotation ID\n",
      "1 Date (YYYY-MM-DD)\n",
      "2 Type\n",
      "3 Comment\n",
      "ClinicalAnn\n",
      "0 Clinical Annotation ID\n",
      "1 Variant/Haplotypes\n",
      "2 Gene\n",
      "3 Level of Evidence\n",
      "4 Level Override\n",
      "5 Level Modifiers\n",
      "6 Score\n",
      "7 Phenotype Category\n",
      "8 PMID Count\n",
      "9 Evidence Count\n",
      "10 Drug(s)\n",
      "11 Phenotype(s)\n",
      "12 Latest History Date (YYYY-MM-DD)\n",
      "13 URL\n",
      "14 Specialty Population\n"
     ]
    }
   ],
   "source": [
    "#clinical annotations are identified by a single ID but can be  referenced in different files\n",
    "al = pd.read_csv(os.path.join(pharmgkb_path, \"clinicalAnnotations\", \"clinical_ann_alleles.csv\"))\n",
    "ev = pd.read_csv(os.path.join(pharmgkb_path, \"clinicalAnnotations\", \"clinical_ann_evidence.csv\"),encoding= 'unicode_escape')\n",
    "his = pd.read_csv(os.path.join(pharmgkb_path, \"clinicalAnnotations\", \"clinical_ann_history.csv\"),encoding= 'unicode_escape')\n",
    "cl = pd.read_csv(os.path.join(pharmgkb_path, \"clinicalAnnotations\", \"clinical_annotations.csv\"),encoding= 'unicode_escape')\n",
    "\n",
    "\n",
    "print(\"Alleles\")\n",
    "for i,c in enumerate(al.columns):\n",
    "    print(i, c)\n",
    "print(\"Evidence\")\n",
    "for i,c in enumerate(ev.columns):\n",
    "    print(i, c)\n",
    "print(\"History\")\n",
    "for i,c in enumerate(his.columns):\n",
    "    print(i, c)\n",
    "print(\"ClinicalAnn\")\n",
    "for i,c in enumerate(cl.columns):\n",
    "    print(i, c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Clinical Annotation ID</th>\n",
       "      <th>Genotype/Allele</th>\n",
       "      <th>Annotation Text</th>\n",
       "      <th>Allele Function</th>\n",
       "      <th>test</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2809</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>*1</td>\n",
       "      <td>Patients carrying the CYP2D6*1 allele in combi...</td>\n",
       "      <td>Normal function</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2810</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>*2</td>\n",
       "      <td>Patients carrying the CYP2D6*2 allele in combi...</td>\n",
       "      <td>Normal function</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2811</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>*3</td>\n",
       "      <td>Patients carrying the CYP2D6*3 allele in combi...</td>\n",
       "      <td>No function</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2812</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>*4</td>\n",
       "      <td>Patients carrying the CYP2D6*4 allele in combi...</td>\n",
       "      <td>No function</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2813</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>*5</td>\n",
       "      <td>Patients carrying the CYP2D6*5 allele in combi...</td>\n",
       "      <td>No function</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2814</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>*6</td>\n",
       "      <td>Patients carrying the CYP2D6*6 allele in combi...</td>\n",
       "      <td>No function</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2815</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>*10</td>\n",
       "      <td>Patients carrying the CYP2D6*10 allele in comb...</td>\n",
       "      <td>Decreased function (AV 0.25)</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Clinical Annotation ID Genotype/Allele  \\\n",
       "2809              1183615480              *1   \n",
       "2810              1183615480              *2   \n",
       "2811              1183615480              *3   \n",
       "2812              1183615480              *4   \n",
       "2813              1183615480              *5   \n",
       "2814              1183615480              *6   \n",
       "2815              1183615480             *10   \n",
       "\n",
       "                                        Annotation Text  \\\n",
       "2809  Patients carrying the CYP2D6*1 allele in combi...   \n",
       "2810  Patients carrying the CYP2D6*2 allele in combi...   \n",
       "2811  Patients carrying the CYP2D6*3 allele in combi...   \n",
       "2812  Patients carrying the CYP2D6*4 allele in combi...   \n",
       "2813  Patients carrying the CYP2D6*5 allele in combi...   \n",
       "2814  Patients carrying the CYP2D6*6 allele in combi...   \n",
       "2815  Patients carrying the CYP2D6*10 allele in comb...   \n",
       "\n",
       "                   Allele Function  test  \n",
       "2809               Normal function   NaN  \n",
       "2810               Normal function   NaN  \n",
       "2811                   No function   NaN  \n",
       "2812                   No function   NaN  \n",
       "2813                   No function   NaN  \n",
       "2814                   No function   NaN  \n",
       "2815  Decreased function (AV 0.25)   NaN  "
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Use a CAID (Clinical Ann ID) as an example see the information across files\n",
    "caid = 1183615480\n",
    "al[al[\"Clinical Annotation ID\"]==caid]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Clinical Annotation ID</th>\n",
       "      <th>Evidence ID</th>\n",
       "      <th>Evidence Type</th>\n",
       "      <th>Evidence URL</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Summary</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2765</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>827807126</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/827...</td>\n",
       "      <td>21599570.0</td>\n",
       "      <td>CYP2D6 *1/*1 is associated with increased clea...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2766</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>827807118</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/827...</td>\n",
       "      <td>21599570.0</td>\n",
       "      <td>CYP2D6 *1/*1 is associated with increased clea...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2767</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>827807123</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/827...</td>\n",
       "      <td>21599570.0</td>\n",
       "      <td>CYP2D6 *1/*1 is associated with increased clea...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2768</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>827807099</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/827...</td>\n",
       "      <td>21599570.0</td>\n",
       "      <td>CYP2D6 *1/*1 is associated with increased clea...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2769</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>982036882</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/982...</td>\n",
       "      <td>16849011.0</td>\n",
       "      <td>CYP2D6 *10 is associated with decreased metabo...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2770</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>982036911</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/982...</td>\n",
       "      <td>16849011.0</td>\n",
       "      <td>CYP2D6 *4 is not associated with metabolism of...</td>\n",
       "      <td>-0.125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2771</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>982036924</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/982...</td>\n",
       "      <td>16849011.0</td>\n",
       "      <td>CYP2D6 *5 is not associated with metabolism of...</td>\n",
       "      <td>-0.125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2772</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>982043452</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/982...</td>\n",
       "      <td>16595916.0</td>\n",
       "      <td>CYP2D6 *10/*10 is associated with decreased cl...</td>\n",
       "      <td>2.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2773</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>982044314</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/982...</td>\n",
       "      <td>17329852.0</td>\n",
       "      <td>CYP2D6 *1/*4 is associated with decreased clea...</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2774</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>982044321</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/982...</td>\n",
       "      <td>17329852.0</td>\n",
       "      <td>CYP2D6 *1/*5 is associated with decreased clea...</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2775</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>982044327</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/982...</td>\n",
       "      <td>17329852.0</td>\n",
       "      <td>CYP2D6 *10/*10 is associated with decreased cl...</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Clinical Annotation ID Evidence ID            Evidence Type  \\\n",
       "2765              1183615480   827807126  Variant Drug Annotation   \n",
       "2766              1183615480   827807118  Variant Drug Annotation   \n",
       "2767              1183615480   827807123  Variant Drug Annotation   \n",
       "2768              1183615480   827807099  Variant Drug Annotation   \n",
       "2769              1183615480   982036882  Variant Drug Annotation   \n",
       "2770              1183615480   982036911  Variant Drug Annotation   \n",
       "2771              1183615480   982036924  Variant Drug Annotation   \n",
       "2772              1183615480   982043452  Variant Drug Annotation   \n",
       "2773              1183615480   982044314  Variant Drug Annotation   \n",
       "2774              1183615480   982044321  Variant Drug Annotation   \n",
       "2775              1183615480   982044327  Variant Drug Annotation   \n",
       "\n",
       "                                           Evidence URL        PMID  \\\n",
       "2765  https://www.pharmgkb.org/variantAnnotation/827...  21599570.0   \n",
       "2766  https://www.pharmgkb.org/variantAnnotation/827...  21599570.0   \n",
       "2767  https://www.pharmgkb.org/variantAnnotation/827...  21599570.0   \n",
       "2768  https://www.pharmgkb.org/variantAnnotation/827...  21599570.0   \n",
       "2769  https://www.pharmgkb.org/variantAnnotation/982...  16849011.0   \n",
       "2770  https://www.pharmgkb.org/variantAnnotation/982...  16849011.0   \n",
       "2771  https://www.pharmgkb.org/variantAnnotation/982...  16849011.0   \n",
       "2772  https://www.pharmgkb.org/variantAnnotation/982...  16595916.0   \n",
       "2773  https://www.pharmgkb.org/variantAnnotation/982...  17329852.0   \n",
       "2774  https://www.pharmgkb.org/variantAnnotation/982...  17329852.0   \n",
       "2775  https://www.pharmgkb.org/variantAnnotation/982...  17329852.0   \n",
       "\n",
       "                                                Summary   Score  \n",
       "2765  CYP2D6 *1/*1 is associated with increased clea...       0  \n",
       "2766  CYP2D6 *1/*1 is associated with increased clea...       0  \n",
       "2767  CYP2D6 *1/*1 is associated with increased clea...       0  \n",
       "2768  CYP2D6 *1/*1 is associated with increased clea...       0  \n",
       "2769  CYP2D6 *10 is associated with decreased metabo...       1  \n",
       "2770  CYP2D6 *4 is not associated with metabolism of...  -0.125  \n",
       "2771  CYP2D6 *5 is not associated with metabolism of...  -0.125  \n",
       "2772  CYP2D6 *10/*10 is associated with decreased cl...    2.25  \n",
       "2773  CYP2D6 *1/*4 is associated with decreased clea...     1.5  \n",
       "2774  CYP2D6 *1/*5 is associated with decreased clea...     1.5  \n",
       "2775  CYP2D6 *10/*10 is associated with decreased cl...     1.5  "
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ev[ev[\"Clinical Annotation ID\"]==caid]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Clinical Annotation ID</th>\n",
       "      <th>Variant/Haplotypes</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Level of Evidence</th>\n",
       "      <th>Level Override</th>\n",
       "      <th>Level Modifiers</th>\n",
       "      <th>Score</th>\n",
       "      <th>Phenotype Category</th>\n",
       "      <th>PMID Count</th>\n",
       "      <th>Evidence Count</th>\n",
       "      <th>Drug(s)</th>\n",
       "      <th>Phenotype(s)</th>\n",
       "      <th>Latest History Date (YYYY-MM-DD)</th>\n",
       "      <th>URL</th>\n",
       "      <th>Specialty Population</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>875</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>CYP2D6*1, CYP2D6*2, CYP2D6*3, CYP2D6*4, CYP2D6...</td>\n",
       "      <td>CYP2D6</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tier 1 VIP</td>\n",
       "      <td>7.5</td>\n",
       "      <td>Metabolism/PK</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>carvedilol</td>\n",
       "      <td>Heart Diseases</td>\n",
       "      <td>22/04/2021</td>\n",
       "      <td>https://www.pharmgkb.org/clinicalAnnotation/11...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Clinical Annotation ID  \\\n",
       "875              1183615480   \n",
       "\n",
       "                                    Variant/Haplotypes    Gene  \\\n",
       "875  CYP2D6*1, CYP2D6*2, CYP2D6*3, CYP2D6*4, CYP2D6...  CYP2D6   \n",
       "\n",
       "    Level of Evidence Level Override Level Modifiers  Score  \\\n",
       "875                 3            NaN      Tier 1 VIP    7.5   \n",
       "\n",
       "    Phenotype Category  PMID Count  Evidence Count     Drug(s)  \\\n",
       "875      Metabolism/PK           4              11  carvedilol   \n",
       "\n",
       "       Phenotype(s) Latest History Date (YYYY-MM-DD)  \\\n",
       "875  Heart Diseases                       22/04/2021   \n",
       "\n",
       "                                                   URL Specialty Population  \n",
       "875  https://www.pharmgkb.org/clinicalAnnotation/11...                  NaN  "
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cl[cl[\"Clinical Annotation ID\"]==caid]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Clinical Annotation ID</th>\n",
       "      <th>Date (YYYY-MM-DD)</th>\n",
       "      <th>Type</th>\n",
       "      <th>Comment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1989</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>18/02/2017</td>\n",
       "      <td>Update</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1990</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>12/09/2018</td>\n",
       "      <td>Update</td>\n",
       "      <td>Removed inaccurate variant annotation on PMID ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1991</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>05/11/2018</td>\n",
       "      <td>Update</td>\n",
       "      <td>Updated OMB race to appropriate biogeographica...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1992</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>05/02/2021</td>\n",
       "      <td>Update</td>\n",
       "      <td>Added PMIDs 16849011 and 21599570 to evidence....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1993</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>24/03/2021</td>\n",
       "      <td>Update</td>\n",
       "      <td>CA score added as part of scoring system relea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1994</th>\n",
       "      <td>1183615480</td>\n",
       "      <td>22/04/2021</td>\n",
       "      <td>Update</td>\n",
       "      <td>Added sentence about DPWG 'no recommendation' ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Clinical Annotation ID Date (YYYY-MM-DD)    Type  \\\n",
       "1989              1183615480        18/02/2017  Update   \n",
       "1990              1183615480        12/09/2018  Update   \n",
       "1991              1183615480        05/11/2018  Update   \n",
       "1992              1183615480        05/02/2021  Update   \n",
       "1993              1183615480        24/03/2021  Update   \n",
       "1994              1183615480        22/04/2021  Update   \n",
       "\n",
       "                                                Comment  \n",
       "1989                                                NaN  \n",
       "1990  Removed inaccurate variant annotation on PMID ...  \n",
       "1991  Updated OMB race to appropriate biogeographica...  \n",
       "1992  Added PMIDs 16849011 and 21599570 to evidence....  \n",
       "1993  CA score added as part of scoring system relea...  \n",
       "1994  Added sentence about DPWG 'no recommendation' ...  "
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "his[his[\"Clinical Annotation ID\"]==caid]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "#create temporal smaller files to work on:\n",
    "\n",
    "df = pd.read_csv(os.path.join(processed_path, \"clinical_annotation.csv\"))\n",
    "df = df.head(1000)\n",
    "df.to_csv(os.path.join(processed_path, \"clinann_tmp.csv\"), index=False)\n",
    "\n",
    "df = pd.read_csv(os.path.join(pharmgkb_path, \"clinicalAnnotations\", \"clinical_ann_alleles.csv\"))\n",
    "df = df.head(1000)\n",
    "df.to_csv(os.path.join(processed_path, \"clinann_all_tmp.csv\"), index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'*1'"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "h = \"CYP2C9*1\"\n",
    "\"*\"+h.split(\"*\")[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## PGX Relation Table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "df = pd.read_csv(\"../data/pharmgkb_processed/pgx_relation.csv\")\n",
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(\"Association IDs: \"+str(len(set(df[\"aid\"]))))\n",
    "print(\"Genomic variation: \"+str(len(set(df[\"genomic_variation\"]))))\n",
    "print(\"Variant IDs: \"+str(len(set(df[\"vid\"]))))\n",
    "print(\"Haplotype IDs: \"+str(len(set(df[\"hid\"]))))\n",
    "print(\"Gene IDs: \"+str(len(set(df[\"gid\"]))))\n",
    "print(\"Chemical IDs: \"+str(len(set(df[\"cid\"]))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "bid = [\"AAC\",\"AME\",\"SAS\",\"EAS\",\"EUR\",\"LAT\",\"NEA\",\"OCE\",\"SSA\",\"MG\",\"UNK\",\"CST\"]\n",
    "counts = []\n",
    "\n",
    "for b in bid:\n",
    "    count = (df['bid'] == b).sum()\n",
    "    counts += [count]\n",
    "\n",
    "colors = ['#5F4690', '#E6F598', '#F46D43', '#FDE08B', '#66C2A5', '#CC503E', '#ABDDA4', '#3288BD', '#D53D4F', \"#808080\", \"#808080\", \"#808080\"]\n",
    "fig, ax = plt.subplots()\n",
    "ax.bar(bid, counts, color=colors)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ev = df[\"evidence\"].unique().tolist()\n",
    "counts = []\n",
    "for e in sorted(ev):\n",
    "    count = (df['evidence'] == e).sum()\n",
    "    counts += [count]\n",
    "\n",
    "colors = ['#5F4690', '#1D6996', '#38A6A5', '#0F8554', '#73AF48', '#EDAD08', '#E17C05', '#CC503E', '#94346E']\n",
    "fig, ax = plt.subplots()\n",
    "ax.bar(sorted(ev), counts, color=colors)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#most abundant genes\n",
    "counts = df[\"gene\"].value_counts()\n",
    "print(counts[:20])\n",
    "\n",
    "#most abundant compounds\n",
    "\n",
    "counts = df[\"chemical\"].value_counts()\n",
    "print(counts[:20])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pgx",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.0"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
