{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Preparing Variants from PharmGKB for variant annotation\n",
    "Only run this when the pipeline for data processing has been completed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = \"/Users/annamontaner/Documents/BSC/scratch/cli79/cli79334/projects/other/pharmGKB\"\n",
    "\n",
    "vars = pd.read_csv(os.path.join(path,\"input_data\", \"variant_assembly.csv\"), header=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "vid                 object\n",
       "variant             object\n",
       "gene                object\n",
       "gid                 object\n",
       "assembly            object\n",
       "begin              float64\n",
       "end_               float64\n",
       "intronic_offset    float64\n",
       "ref_allele          object\n",
       "ref_hgvs            object\n",
       "var_allele          object\n",
       "var_hgvs            object\n",
       "chr                 object\n",
       "dtype: object"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check data types\n",
    "vars.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total rows:  12674\n",
      "Gene missing:  954\n",
      "Assembly missing:  264\n",
      "Begin missing:  264\n",
      "Ref Allele missing:  264\n",
      "Var Allele missing:  264\n"
     ]
    }
   ],
   "source": [
    "# Check how many rows...\n",
    "print(\"Total rows: \", len(vars))\n",
    "print(\"Gene missing: \", len(vars[vars[\"gid\"].isna()]))\n",
    "print(\"Assembly missing: \", len(vars[vars[\"assembly\"].isna()]))\n",
    "print(\"Begin missing: \", len(vars[vars[\"begin\"].isna()]))\n",
    "print(\"Ref Allele missing: \", len(vars[vars[\"ref_allele\"].isna()]))\n",
    "print(\"Var Allele missing: \", len(vars[vars[\"var_allele\"].isna()]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12410\n"
     ]
    }
   ],
   "source": [
    "# Remove rows for which reference or alternative alleles are not available\n",
    "vars = vars[~vars[\"ref_allele\"].isna()]\n",
    "vars = vars[~vars[\"var_allele\"].isna()]\n",
    "print(len(vars))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12319\n"
     ]
    }
   ],
   "source": [
    "# Remove rows with \"-\"\" instead of A/C/G/T\n",
    "vars = vars[~(vars[\"ref_allele\"]==\"-\")]\n",
    "vars = vars[~(vars[\"var_allele\"]==\"-\")]\n",
    "print(len(vars))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "93\n"
     ]
    }
   ],
   "source": [
    "print(len(vars[vars[\"var_hgvs\"].isna()]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\nStill some differences between snp and snvs due to bad hgvs_alt notation. Example of a variant filtered out in snp but present in snvs:\\n9934\\tPA166157504\\trs72552784\\tABCB1\\tPA267\\tGRCh38\\t87516598.0\\t87516598.0\\t0.0\\tC\\tNC_000007.14:g.87516598=\\tA\\tNC_000006.12:g.160139851_160139853del\\t[GRCh38]chr7\\tleft_only\\n'"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Keep only SNVs\n",
    "\n",
    "# Gemma: use var_hgvs notation\n",
    "snp = vars[vars[\"var_hgvs\"].str.contains(\">\", na=False)]\n",
    "\n",
    "# Anna: a) use ref and alt alleles' length and b) remove variants containing characters other than A/C/G/T\n",
    "# a)\n",
    "snvs_tmp = vars[vars['ref_allele'].str.len() == vars['var_allele'].str.len()]\n",
    "\n",
    "# b)\n",
    "allowed_values = ['A', 'C', 'G', 'T']\n",
    "snvs = snvs_tmp[snvs_tmp['ref_allele'].str.match(f'^({\"|\".join(allowed_values)})$') &\n",
    "               snvs_tmp['var_allele'].str.match(f'^({\"|\".join(allowed_values)})$')]\n",
    "\n",
    "# Check differences between snp (GEMMA) and snvs_tmp (Anna)\n",
    "merged = snvs_tmp.merge(snp, how='outer', indicator=True)\n",
    "comparison1 = merged[merged['_merge'] == 'left_only']\n",
    "\n",
    "\"\"\"\n",
    "Some single-nucleotide changes don't have a proper var_hgvs annotation so\n",
    "it's better to use the nucleotide length of reference and alt alleles instead.\n",
    "E.g. \n",
    "12441\tPA166161042\trs78365220\tG6PD\tPA28469\tGRCh38\t154535270.0\t154535270.0\t0.0\tA\tNC_000023.11:g.154535270=\tC\tNaN\t[GRCh38]chrX\n",
    "\"\"\"\n",
    "\n",
    "# Check differences between snp (GEMMA) and snvs (Anna)\n",
    "merged = snvs.merge(snp, how='outer', indicator=True)\n",
    "comparison2 = merged[merged['_merge'] == 'left_only']\n",
    "\n",
    "\"\"\"\n",
    "Still some differences between snp and snvs due to bad hgvs_alt notation. \n",
    "Example of a variant filtered out in snp but present in snvs:\n",
    "9934\tPA166157504\trs72552784\tABCB1\tPA267\tGRCh38\t87516598.0\t87516598.0\t0.0\tC\tNC_000007.14:g.87516598=\tA\tNC_000006.12:g.160139851_160139853del\t[GRCh38]chr7\tleft_only\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rows vars:  12319\n",
      "Rows snp:  11981\n",
      "Rows snvs_tmp:  12164\n",
      "Rows snvs:  12088\n"
     ]
    }
   ],
   "source": [
    "print(\"Rows vars: \", len(vars))\n",
    "print(\"Rows snp: \", len(snp))\n",
    "print(\"Rows snvs_tmp: \", len(snvs_tmp))\n",
    "print(\"Rows snvs: \", len(snvs))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Ref allele types:  ['C' 'T' 'G' 'A']\n",
      "Var allele types:  ['T' 'A' 'C' 'G']\n"
     ]
    }
   ],
   "source": [
    "# Check variant types in snvs dataframe\n",
    "print('Ref allele types: ',snvs['ref_allele'].unique())\n",
    "print('Var allele types: ',snvs['var_allele'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/kt/qf89bw917mg3nhrtrkj4_lmc0000gn/T/ipykernel_3194/3509328262.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  snvs['begin'] = snvs['begin'].astype(\"int\")\n",
      "/var/folders/kt/qf89bw917mg3nhrtrkj4_lmc0000gn/T/ipykernel_3194/3509328262.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  snvs['chr'] = snvs['chr'].astype(\"str\")\n",
      "/var/folders/kt/qf89bw917mg3nhrtrkj4_lmc0000gn/T/ipykernel_3194/3509328262.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  snvs['chr'] = snvs['chr'].str.split(\"]\").str[1].str.replace(\"chr\",\"\")\n"
     ]
    }
   ],
   "source": [
    "#Â Change some column types\n",
    "snvs['begin'] = snvs['begin'].astype(\"int\")\n",
    "snvs['chr'] = snvs['chr'].astype(\"str\")\n",
    "snvs['chr'] = snvs['chr'].str.split(\"]\").str[1].str.replace(\"chr\",\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>vid</th>\n",
       "      <th>variant</th>\n",
       "      <th>gene</th>\n",
       "      <th>gid</th>\n",
       "      <th>assembly</th>\n",
       "      <th>begin</th>\n",
       "      <th>end_</th>\n",
       "      <th>intronic_offset</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>ref_hgvs</th>\n",
       "      <th>var_allele</th>\n",
       "      <th>var_hgvs</th>\n",
       "      <th>chr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PA166156302</td>\n",
       "      <td>rs1000002</td>\n",
       "      <td>ABCC5</td>\n",
       "      <td>PA395</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>183917980</td>\n",
       "      <td>183917980.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C</td>\n",
       "      <td>NC_000003.12:g.183917980=</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000003.12:g.183917980C&gt;T</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PA166156746</td>\n",
       "      <td>rs1000113</td>\n",
       "      <td>IRGM</td>\n",
       "      <td>PA142671652</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>150860514</td>\n",
       "      <td>150860514.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C</td>\n",
       "      <td>NC_000005.10:g.150860514=</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000005.10:g.150860514C&gt;T</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PA166195421</td>\n",
       "      <td>rs10006452</td>\n",
       "      <td>UGT2B7</td>\n",
       "      <td>PA361</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>69112090</td>\n",
       "      <td>69112090.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000004.12:g.69112090=</td>\n",
       "      <td>A</td>\n",
       "      <td>NC_000004.12:g.69112090T&gt;A</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PA166195421</td>\n",
       "      <td>rs10006452</td>\n",
       "      <td>UGT2B7</td>\n",
       "      <td>PA361</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>69112090</td>\n",
       "      <td>69112090.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000004.12:g.69112090=</td>\n",
       "      <td>C</td>\n",
       "      <td>NC_000004.12:g.69112090T&gt;C</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PA166177121</td>\n",
       "      <td>rs10007051</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>129244309</td>\n",
       "      <td>129244309.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C</td>\n",
       "      <td>NC_000004.12:g.129244309=</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000004.12:g.129244309C&gt;T</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12658</th>\n",
       "      <td>PA166155732</td>\n",
       "      <td>rs4663871</td>\n",
       "      <td>UGT1A9</td>\n",
       "      <td>PA419</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>233672941</td>\n",
       "      <td>233672941.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G</td>\n",
       "      <td>NC_000002.12:g.233672941=</td>\n",
       "      <td>A</td>\n",
       "      <td>NC_000002.12:g.233673102C&gt;T</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12659</th>\n",
       "      <td>PA166155732</td>\n",
       "      <td>rs4663871</td>\n",
       "      <td>UGT1A9</td>\n",
       "      <td>PA419</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>233672941</td>\n",
       "      <td>233672941.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G</td>\n",
       "      <td>NC_000002.12:g.233672941=</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000002.12:g.233672941G&gt;A</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12660</th>\n",
       "      <td>PA166155877</td>\n",
       "      <td>rs45554333</td>\n",
       "      <td>UGT1A9</td>\n",
       "      <td>PA419</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>233672932</td>\n",
       "      <td>233672932.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C</td>\n",
       "      <td>NC_000002.12:g.233672932=</td>\n",
       "      <td>A</td>\n",
       "      <td>NC_000002.12:g.233672941G&gt;T</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12661</th>\n",
       "      <td>PA166155877</td>\n",
       "      <td>rs45554333</td>\n",
       "      <td>UGT1A9</td>\n",
       "      <td>PA419</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>233672932</td>\n",
       "      <td>233672932.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C</td>\n",
       "      <td>NC_000002.12:g.233672932=</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000002.12:g.233672932C&gt;A</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12666</th>\n",
       "      <td>PA166155893</td>\n",
       "      <td>rs66915469</td>\n",
       "      <td>UGT1A9</td>\n",
       "      <td>PA419</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>233672660</td>\n",
       "      <td>233672660.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000002.12:g.233672660=</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12088 rows Ã 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               vid     variant    gene          gid assembly      begin  \\\n",
       "0      PA166156302   rs1000002   ABCC5        PA395   GRCh38  183917980   \n",
       "1      PA166156746   rs1000113    IRGM  PA142671652   GRCh38  150860514   \n",
       "2      PA166195421  rs10006452  UGT2B7        PA361   GRCh38   69112090   \n",
       "3      PA166195421  rs10006452  UGT2B7        PA361   GRCh38   69112090   \n",
       "4      PA166177121  rs10007051     NaN          NaN   GRCh38  129244309   \n",
       "...            ...         ...     ...          ...      ...        ...   \n",
       "12658  PA166155732   rs4663871  UGT1A9        PA419   GRCh38  233672941   \n",
       "12659  PA166155732   rs4663871  UGT1A9        PA419   GRCh38  233672941   \n",
       "12660  PA166155877  rs45554333  UGT1A9        PA419   GRCh38  233672932   \n",
       "12661  PA166155877  rs45554333  UGT1A9        PA419   GRCh38  233672932   \n",
       "12666  PA166155893  rs66915469  UGT1A9        PA419   GRCh38  233672660   \n",
       "\n",
       "              end_  intronic_offset ref_allele                   ref_hgvs  \\\n",
       "0      183917980.0              0.0          C  NC_000003.12:g.183917980=   \n",
       "1      150860514.0              0.0          C  NC_000005.10:g.150860514=   \n",
       "2       69112090.0              0.0          T   NC_000004.12:g.69112090=   \n",
       "3       69112090.0              0.0          T   NC_000004.12:g.69112090=   \n",
       "4      129244309.0              0.0          C  NC_000004.12:g.129244309=   \n",
       "...            ...              ...        ...                        ...   \n",
       "12658  233672941.0              0.0          G  NC_000002.12:g.233672941=   \n",
       "12659  233672941.0              0.0          G  NC_000002.12:g.233672941=   \n",
       "12660  233672932.0              0.0          C  NC_000002.12:g.233672932=   \n",
       "12661  233672932.0              0.0          C  NC_000002.12:g.233672932=   \n",
       "12666  233672660.0              0.0          T  NC_000002.12:g.233672660=   \n",
       "\n",
       "      var_allele                     var_hgvs chr  \n",
       "0              T  NC_000003.12:g.183917980C>T   3  \n",
       "1              T  NC_000005.10:g.150860514C>T   5  \n",
       "2              A   NC_000004.12:g.69112090T>A   4  \n",
       "3              C   NC_000004.12:g.69112090T>C   4  \n",
       "4              T  NC_000004.12:g.129244309C>T   4  \n",
       "...          ...                          ...  ..  \n",
       "12658          A  NC_000002.12:g.233673102C>T   2  \n",
       "12659          T  NC_000002.12:g.233672941G>A   2  \n",
       "12660          A  NC_000002.12:g.233672941G>T   2  \n",
       "12661          T  NC_000002.12:g.233672932C>A   2  \n",
       "12666          G                          NaN   2  \n",
       "\n",
       "[12088 rows x 13 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snvs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Build final dataframe\n",
    "df_tmp=snvs.copy()\n",
    "df_tmp = df_tmp.fillna(\".\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Build non-existent columns\n",
    "df_tmp['ID'] = \".\"\n",
    "df_tmp['QUAL'] = \".\"\n",
    "df_tmp['FILTER'] = \"NONE\"\n",
    "\n",
    "#Â Build info column\n",
    "info = \"PGKBVID=\"+df_tmp[\"vid\"]+\";RSID=\"+df_tmp['variant']+\";GENE=\"+df_tmp['gene']+\";PGKBGID=\"+df_tmp['gid']+\";HGVS=\"+df_tmp['var_hgvs']\n",
    "df_tmp['INFO'] = info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_17847/1235631419.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  snp['INFO'] = snp.apply(build_info, axis=1)\n"
     ]
    }
   ],
   "source": [
    "# Build info Gemma\n",
    "\"\"\"\n",
    "def build_info(row):\n",
    "    info_parts = []\n",
    "    if not pd.isna(row['vid']):\n",
    "        info_parts.append('PGKBVID=' + row['vid'])\n",
    "    if not pd.isna(row['variant']):\n",
    "        info_parts.append('RSID=' + row['variant'])\n",
    "    if not pd.isna(row['gene']):\n",
    "        info_parts.append('GENE=' + row['gene'])\n",
    "    if not pd.isna(row['gid']):\n",
    "        info_parts.append('PGKBGID=' + row['gid'])\n",
    "    if not pd.isna(row['var_hgvs']):\n",
    "        info_parts.append('HGVS=' + row['var_hgvs'])\n",
    "\n",
    "    return ';'.join(info_parts)\n",
    "\n",
    "snp['INFO'] = snp.apply(build_info, axis=1)\n",
    "\n",
    "snp[\"ID\"]=\"\"\n",
    "snp[\"FILTER\"]=\"NONE\"\n",
    "snp[\"QUAL\"] = \"\"\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Rename columns \n",
    "df_tmp.rename(columns={\"chr\":\"#CHROM\", \"ref_allele\": \"REF\", \"var_allele\": \"ALT\", \"begin\": \"POS\"}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>vid</th>\n",
       "      <th>variant</th>\n",
       "      <th>gene</th>\n",
       "      <th>gid</th>\n",
       "      <th>assembly</th>\n",
       "      <th>POS</th>\n",
       "      <th>end_</th>\n",
       "      <th>intronic_offset</th>\n",
       "      <th>REF</th>\n",
       "      <th>ref_hgvs</th>\n",
       "      <th>ALT</th>\n",
       "      <th>var_hgvs</th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>ID</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PA166156302</td>\n",
       "      <td>rs1000002</td>\n",
       "      <td>ABCC5</td>\n",
       "      <td>PA395</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>183917980</td>\n",
       "      <td>183917980.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C</td>\n",
       "      <td>NC_000003.12:g.183917980=</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000003.12:g.183917980C&gt;T</td>\n",
       "      <td>3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166156302;RSID=rs1000002;GENE=ABCC5;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PA166156746</td>\n",
       "      <td>rs1000113</td>\n",
       "      <td>IRGM</td>\n",
       "      <td>PA142671652</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>150860514</td>\n",
       "      <td>150860514.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C</td>\n",
       "      <td>NC_000005.10:g.150860514=</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000005.10:g.150860514C&gt;T</td>\n",
       "      <td>5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166156746;RSID=rs1000113;GENE=IRGM;P...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PA166195421</td>\n",
       "      <td>rs10006452</td>\n",
       "      <td>UGT2B7</td>\n",
       "      <td>PA361</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>69112090</td>\n",
       "      <td>69112090.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000004.12:g.69112090=</td>\n",
       "      <td>A</td>\n",
       "      <td>NC_000004.12:g.69112090T&gt;A</td>\n",
       "      <td>4</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166195421;RSID=rs10006452;GENE=UGT2B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PA166195421</td>\n",
       "      <td>rs10006452</td>\n",
       "      <td>UGT2B7</td>\n",
       "      <td>PA361</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>69112090</td>\n",
       "      <td>69112090.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000004.12:g.69112090=</td>\n",
       "      <td>C</td>\n",
       "      <td>NC_000004.12:g.69112090T&gt;C</td>\n",
       "      <td>4</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166195421;RSID=rs10006452;GENE=UGT2B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PA166177121</td>\n",
       "      <td>rs10007051</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>129244309</td>\n",
       "      <td>129244309.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C</td>\n",
       "      <td>NC_000004.12:g.129244309=</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000004.12:g.129244309C&gt;T</td>\n",
       "      <td>4</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166177121;RSID=rs10007051;GENE=.;PGK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12658</th>\n",
       "      <td>PA166155732</td>\n",
       "      <td>rs4663871</td>\n",
       "      <td>UGT1A9</td>\n",
       "      <td>PA419</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>233672941</td>\n",
       "      <td>233672941.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G</td>\n",
       "      <td>NC_000002.12:g.233672941=</td>\n",
       "      <td>A</td>\n",
       "      <td>NC_000002.12:g.233673102C&gt;T</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166155732;RSID=rs4663871;GENE=UGT1A9...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12659</th>\n",
       "      <td>PA166155732</td>\n",
       "      <td>rs4663871</td>\n",
       "      <td>UGT1A9</td>\n",
       "      <td>PA419</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>233672941</td>\n",
       "      <td>233672941.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G</td>\n",
       "      <td>NC_000002.12:g.233672941=</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000002.12:g.233672941G&gt;A</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166155732;RSID=rs4663871;GENE=UGT1A9...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12660</th>\n",
       "      <td>PA166155877</td>\n",
       "      <td>rs45554333</td>\n",
       "      <td>UGT1A9</td>\n",
       "      <td>PA419</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>233672932</td>\n",
       "      <td>233672932.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C</td>\n",
       "      <td>NC_000002.12:g.233672932=</td>\n",
       "      <td>A</td>\n",
       "      <td>NC_000002.12:g.233672941G&gt;T</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166155877;RSID=rs45554333;GENE=UGT1A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12661</th>\n",
       "      <td>PA166155877</td>\n",
       "      <td>rs45554333</td>\n",
       "      <td>UGT1A9</td>\n",
       "      <td>PA419</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>233672932</td>\n",
       "      <td>233672932.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C</td>\n",
       "      <td>NC_000002.12:g.233672932=</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000002.12:g.233672932C&gt;A</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166155877;RSID=rs45554333;GENE=UGT1A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12666</th>\n",
       "      <td>PA166155893</td>\n",
       "      <td>rs66915469</td>\n",
       "      <td>UGT1A9</td>\n",
       "      <td>PA419</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>233672660</td>\n",
       "      <td>233672660.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>T</td>\n",
       "      <td>NC_000002.12:g.233672660=</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166155893;RSID=rs66915469;GENE=UGT1A...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12088 rows Ã 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               vid     variant    gene          gid assembly        POS  \\\n",
       "0      PA166156302   rs1000002   ABCC5        PA395   GRCh38  183917980   \n",
       "1      PA166156746   rs1000113    IRGM  PA142671652   GRCh38  150860514   \n",
       "2      PA166195421  rs10006452  UGT2B7        PA361   GRCh38   69112090   \n",
       "3      PA166195421  rs10006452  UGT2B7        PA361   GRCh38   69112090   \n",
       "4      PA166177121  rs10007051       .            .   GRCh38  129244309   \n",
       "...            ...         ...     ...          ...      ...        ...   \n",
       "12658  PA166155732   rs4663871  UGT1A9        PA419   GRCh38  233672941   \n",
       "12659  PA166155732   rs4663871  UGT1A9        PA419   GRCh38  233672941   \n",
       "12660  PA166155877  rs45554333  UGT1A9        PA419   GRCh38  233672932   \n",
       "12661  PA166155877  rs45554333  UGT1A9        PA419   GRCh38  233672932   \n",
       "12666  PA166155893  rs66915469  UGT1A9        PA419   GRCh38  233672660   \n",
       "\n",
       "              end_  intronic_offset REF                   ref_hgvs ALT  \\\n",
       "0      183917980.0              0.0   C  NC_000003.12:g.183917980=   T   \n",
       "1      150860514.0              0.0   C  NC_000005.10:g.150860514=   T   \n",
       "2       69112090.0              0.0   T   NC_000004.12:g.69112090=   A   \n",
       "3       69112090.0              0.0   T   NC_000004.12:g.69112090=   C   \n",
       "4      129244309.0              0.0   C  NC_000004.12:g.129244309=   T   \n",
       "...            ...              ...  ..                        ...  ..   \n",
       "12658  233672941.0              0.0   G  NC_000002.12:g.233672941=   A   \n",
       "12659  233672941.0              0.0   G  NC_000002.12:g.233672941=   T   \n",
       "12660  233672932.0              0.0   C  NC_000002.12:g.233672932=   A   \n",
       "12661  233672932.0              0.0   C  NC_000002.12:g.233672932=   T   \n",
       "12666  233672660.0              0.0   T  NC_000002.12:g.233672660=   G   \n",
       "\n",
       "                          var_hgvs #CHROM ID QUAL FILTER  \\\n",
       "0      NC_000003.12:g.183917980C>T      3  .    .   NONE   \n",
       "1      NC_000005.10:g.150860514C>T      5  .    .   NONE   \n",
       "2       NC_000004.12:g.69112090T>A      4  .    .   NONE   \n",
       "3       NC_000004.12:g.69112090T>C      4  .    .   NONE   \n",
       "4      NC_000004.12:g.129244309C>T      4  .    .   NONE   \n",
       "...                            ...    ... ..  ...    ...   \n",
       "12658  NC_000002.12:g.233673102C>T      2  .    .   NONE   \n",
       "12659  NC_000002.12:g.233672941G>A      2  .    .   NONE   \n",
       "12660  NC_000002.12:g.233672941G>T      2  .    .   NONE   \n",
       "12661  NC_000002.12:g.233672932C>A      2  .    .   NONE   \n",
       "12666                            .      2  .    .   NONE   \n",
       "\n",
       "                                                    INFO  \n",
       "0      PGKBVID=PA166156302;RSID=rs1000002;GENE=ABCC5;...  \n",
       "1      PGKBVID=PA166156746;RSID=rs1000113;GENE=IRGM;P...  \n",
       "2      PGKBVID=PA166195421;RSID=rs10006452;GENE=UGT2B...  \n",
       "3      PGKBVID=PA166195421;RSID=rs10006452;GENE=UGT2B...  \n",
       "4      PGKBVID=PA166177121;RSID=rs10007051;GENE=.;PGK...  \n",
       "...                                                  ...  \n",
       "12658  PGKBVID=PA166155732;RSID=rs4663871;GENE=UGT1A9...  \n",
       "12659  PGKBVID=PA166155732;RSID=rs4663871;GENE=UGT1A9...  \n",
       "12660  PGKBVID=PA166155877;RSID=rs45554333;GENE=UGT1A...  \n",
       "12661  PGKBVID=PA166155877;RSID=rs45554333;GENE=UGT1A...  \n",
       "12666  PGKBVID=PA166155893;RSID=rs66915469;GENE=UGT1A...  \n",
       "\n",
       "[12088 rows x 17 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df_tmp[[\"#CHROM\", \"POS\", \"ID\", \"REF\", \"ALT\", \"QUAL\", \"FILTER\", \"INFO\"]].sort_values(by=['#CHROM','POS'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10406</th>\n",
       "      <td>1</td>\n",
       "      <td>964594</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166153787;RSID=rs7414551;GENE=KLHL17...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10407</th>\n",
       "      <td>1</td>\n",
       "      <td>964594</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166153787;RSID=rs7414551;GENE=PLEKHN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2689</th>\n",
       "      <td>1</td>\n",
       "      <td>1582436</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166165326;RSID=rs146898897;GENE=.;PG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2344</th>\n",
       "      <td>1</td>\n",
       "      <td>2016961</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166258301;RSID=rs13303344;GENE=.;PGK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2345</th>\n",
       "      <td>1</td>\n",
       "      <td>2016961</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166258301;RSID=rs13303344;GENE=.;PGK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>449</th>\n",
       "      <td>X</td>\n",
       "      <td>154536002</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166157859;RSID=rs1050828;GENE=IKBKG;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12462</th>\n",
       "      <td>X</td>\n",
       "      <td>154536032</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166161038;RSID=rs137852315;GENE=G6PD...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12467</th>\n",
       "      <td>X</td>\n",
       "      <td>154536156</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166161036;RSID=rs76645461;GENE=G6PD;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12468</th>\n",
       "      <td>X</td>\n",
       "      <td>154536168</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166157890;RSID=rs78478128;GENE=G6PD;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12473</th>\n",
       "      <td>X</td>\n",
       "      <td>154546061</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>NONE</td>\n",
       "      <td>PGKBVID=PA166161050;RSID=rs137852340;GENE=G6PD...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12088 rows Ã 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #CHROM        POS ID REF ALT QUAL FILTER  \\\n",
       "10406      1     964594  .   G   A    .   NONE   \n",
       "10407      1     964594  .   G   A    .   NONE   \n",
       "2689       1    1582436  .   C   T    .   NONE   \n",
       "2344       1    2016961  .   C   A    .   NONE   \n",
       "2345       1    2016961  .   C   T    .   NONE   \n",
       "...      ...        ... ..  ..  ..  ...    ...   \n",
       "449        X  154536002  .   C   T    .   NONE   \n",
       "12462      X  154536032  .   C   T    .   NONE   \n",
       "12467      X  154536156  .   A   G    .   NONE   \n",
       "12468      X  154536168  .   G   C    .   NONE   \n",
       "12473      X  154546061  .   T   C    .   NONE   \n",
       "\n",
       "                                                    INFO  \n",
       "10406  PGKBVID=PA166153787;RSID=rs7414551;GENE=KLHL17...  \n",
       "10407  PGKBVID=PA166153787;RSID=rs7414551;GENE=PLEKHN...  \n",
       "2689   PGKBVID=PA166165326;RSID=rs146898897;GENE=.;PG...  \n",
       "2344   PGKBVID=PA166258301;RSID=rs13303344;GENE=.;PGK...  \n",
       "2345   PGKBVID=PA166258301;RSID=rs13303344;GENE=.;PGK...  \n",
       "...                                                  ...  \n",
       "449    PGKBVID=PA166157859;RSID=rs1050828;GENE=IKBKG;...  \n",
       "12462  PGKBVID=PA166161038;RSID=rs137852315;GENE=G6PD...  \n",
       "12467  PGKBVID=PA166161036;RSID=rs76645461;GENE=G6PD;...  \n",
       "12468  PGKBVID=PA166157890;RSID=rs78478128;GENE=G6PD;...  \n",
       "12473  PGKBVID=PA166161050;RSID=rs137852340;GENE=G6PD...  \n",
       "\n",
       "[12088 rows x 8 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Variants in unfiltered PharmGKB dataframe:  6907\n",
      "Variants in filtered PharmGKB dataframe:  6867\n"
     ]
    }
   ],
   "source": [
    "print(\"Variants in unfiltered PharmGKB dataframe: \",len(vars['begin'].unique()))\n",
    "print(\"Variants in filtered PharmGKB dataframe: \",len(df['POS'].unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(os.path.join(path,\"output_data\",\"variant_assembly_auto.tsv\"), sep=\"\\t\", header=True, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "output_VCF = os.path.join(path,\"output_data\",\"variant_assembly_auto.vcf\")\n",
    "input_TSV = os.path.join(path,\"output_data\",\"variant_assembly_auto.tsv\")\n",
    "I = open(input_TSV, \"r\")\n",
    "O = open(output_VCF, \"w\")\n",
    "\n",
    "O.write(\"##fileformat=VCFv4.0\\n\")\n",
    "O.write(\"##FILTER=<ID=NONE,Description=\\\"Not used\\\">\\n\")\n",
    "O.write(\"##INFO=<ID=PGKBVID,Number=1,Type=String,Description=\\\"PharmGKB variant id\\\">\\n\")\n",
    "O.write(\"##INFO=<ID=RSID,Number=1,Type=String,Description=\\\"dbSnp id\\\">\\n\")\n",
    "O.write(\"##INFO=<ID=GENE,Number=1,Type=String,Description=\\\"Gene name\\\">\\n\")\n",
    "O.write(\"##INFO=<ID=PGKBGID,Number=1,Type=String,Description=\\\"PharmGKB gene id\\\">\\n\")\n",
    "O.write(\"##INFO=<ID=HGVS,Number=1,Type=String,Description=\\\"HGVS id\\\">\\n\")\n",
    "O.write(\"##FORMAT=<ID=NO,Number=1,Type=String,Description=\\\"Not used\\\">\\n\")\n",
    "\n",
    "for iLine in I:\n",
    "\tlist = iLine.split(\"\\t\")\n",
    "\tnewLine = '\\t'.join(list)\n",
    "\tO.write(newLine)\n",
    "\n",
    "I.close()\n",
    "O.close()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pgx",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
