{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Variant embeddings\n",
    "\n",
    "In this notebook, we explore strategies to obtain simple vector embeddings from the variant annotation tabular data.\n",
    "\n",
    "These articles are interesting:\n",
    "- https://direct.mit.edu/tacl/article/doi/10.1162/tacl_a_00544/115239/Transformers-for-Tabular-Data-Representation-A\n",
    "\n",
    "```python\n",
    "with open(\"codebook.tsv\", \"w\") as f:\n",
    "    writer = csv.writer(f, delimiter=\"\\t\")\n",
    "    writer.writerow([\"index\", \"column_name\"])\n",
    "    for i, c in enumerate(list(df.columns)):\n",
    "        writer.writerow([i, c])\n",
    "``````"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\n",
    "    \"../data/variants/1000_Genomes/10000_subset_snvs_coding_1kGPhg38.tsv\", sep=\"\\t\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/mduranfrigola/miniconda3/envs/biogpt/lib/python3.10/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    }
   ],
   "source": [
    "import torch\n",
    "from transformers import AutoTokenizer, AutoModelForCausalLM\n",
    "import numpy as np\n",
    "from tqdm import tqdm\n",
    "\n",
    "\n",
    "class BioGPTEmbedder(object):\n",
    "    def __init__(self):\n",
    "        self.tokenizer = AutoTokenizer.from_pretrained(\"microsoft/biogpt\")\n",
    "        self.model = AutoModelForCausalLM.from_pretrained(\"microsoft/biogpt\")\n",
    "\n",
    "    def calculate(self, text_inputs):\n",
    "        X = np.zeros((len(text_inputs), 1024), dtype=np.float32)\n",
    "        for i, text in tqdm(enumerate(text_inputs)):\n",
    "            encoded_input = self.tokenizer(text, return_tensors=\"pt\")\n",
    "            with torch.no_grad():\n",
    "                hidden_states = self.model.base_model(**encoded_input).last_hidden_state\n",
    "            mean_encoding = torch.mean(hidden_states, dim=1)\n",
    "            mean_encoding_np = mean_encoding.numpy()\n",
    "            X[i, :] = mean_encoding_np\n",
    "        return X\n",
    "\n",
    "\n",
    "embedder = BioGPTEmbedder()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Basic information and annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "10it [00:01,  7.68it/s]\n"
     ]
    }
   ],
   "source": [
    "gene_biotypes = {\n",
    "    \"Protein-coding related\": {\n",
    "        \"protein_coding\": \"Genes that code for proteins.\",\n",
    "        \"nonsense_mediated_decay\": \"Genes with transcripts that are subject to Nonsense-Mediated Decay.\",\n",
    "        \"non_stop_decay\": \"Transcripts that have an open reading frame but lack a stop codon.\",\n",
    "        \"ambiguous_orf\": \"Transcripts with an uncertain coding potential.\",\n",
    "        \"disrupted_domain\": \"Transcripts that contain a frameshift within their coding sequence.\",\n",
    "    },\n",
    "    \"Non-coding RNA genes\": {\n",
    "        \"lncRNA\": \"Long non-coding RNA genes.\",\n",
    "        \"miRNA\": \"MicroRNA genes, which produce small RNAs involved in gene silencing.\",\n",
    "        \"snRNA\": \"Small nuclear RNA genes, usually involved in splicing.\",\n",
    "        \"snoRNA\": \"Small nucleolar RNA genes, mainly involved in rRNA modification.\",\n",
    "        \"rRNA\": \"Ribosomal RNA genes.\",\n",
    "        \"tRNA\": \"Transfer RNA genes.\",\n",
    "        \"pseudogene\": \"Non-functional sequences that resemble functional genes.\",\n",
    "        \"Mt_tRNA\": \"Mitochondrial tRNA genes.\",\n",
    "        \"Mt_rRNA\": \"Mitochondrial rRNA genes.\",\n",
    "    },\n",
    "    \"Immunoglobulin/T-cell receptor genes\": {\n",
    "        \"IG_C_gene\": \"Immunoglobulin Constant gene.\",\n",
    "        \"IG_D_gene\": \"Immunoglobulin Diversity gene.\",\n",
    "        \"IG_J_gene\": \"Immunoglobulin Joining gene.\",\n",
    "        \"IG_V_gene\": \"Immunoglobulin Variable gene.\",\n",
    "        \"TR_C_gene\": \"T-cell receptor Constant gene.\",\n",
    "        \"TR_D_gene\": \"T-cell receptor Diversity gene.\",\n",
    "        \"TR_J_gene\": \"T-cell receptor Joining gene.\",\n",
    "        \"TR_V_gene\": \"T-cell receptor Variable gene.\",\n",
    "    },\n",
    "    \"Other types\": {\n",
    "        \"misc_RNA\": \"Miscellaneous RNA genes not classified elsewhere.\",\n",
    "        \"scaRNA\": \"Small Cajal body-specific RNA genes.\",\n",
    "        \"vaultRNA\": \"Vault RNA genes.\",\n",
    "        \"bidirectional_promoter_lncRNA\": \"Non-coding RNA genes that originate from bidirectional promoters.\",\n",
    "        \"retained_intron\": \"Transcripts that retain an intron.\",\n",
    "    },\n",
    "    \"Pseudogenes\": {\n",
    "        \"processed_pseudogene\": \"Pseudogenes derived from retrotransposition.\",\n",
    "        \"unprocessed_pseudogene\": \"Direct duplications of genes.\",\n",
    "        \"polymorphic_pseudogene\": \"Pseudogenes with some functional alleles in the population.\",\n",
    "        \"transcribed_processed_pseudogene\": \"Transcribed processed pseudogenes.\",\n",
    "        \"transcribed_unprocessed_pseudogene\": \"Transcribed unprocessed pseudogenes.\",\n",
    "        \"transcribed_unitary_pseudogene\": \"Transcribed pseudogenes with no functional counterparts.\",\n",
    "        \"unitary_pseudogene\": \"Pseudogenes with no functional counterparts in the genome.\",\n",
    "    },\n",
    "    \"Others\": {\n",
    "        \"TEC\": \"To be Experimentally Confirmed.\",\n",
    "        \"macro_lncRNA\": \"Very long non-coding RNA genes.\",\n",
    "    },\n",
    "}\n",
    "\n",
    "gene_biotypes = dict((l, w) for k, v in gene_biotypes.items() for l, w in v.items())\n",
    "\n",
    "columns = list(df.columns)\n",
    "\n",
    "basic_columns = columns[slice(0, columns.index(\"ANN[*].ERRORS\") + 1)]\n",
    "\n",
    "texts = []\n",
    "\n",
    "for r in df[basic_columns].values:\n",
    "    text = \"\"\n",
    "    c = r[basic_columns.index(\"CHROM\")]\n",
    "    if c == \".\":\n",
    "        text += \"Variant with no known chromosome. \"\n",
    "    else:\n",
    "        text += \"Variant on chromosome {0}. \".format(c)\n",
    "    text += \"Reference allele is {0} and variant allele is {1}. \".format(\n",
    "        r[basic_columns.index(\"REF\")], r[basic_columns.index(\"ALT\")]\n",
    "    )\n",
    "    text += \"Annotated gene is {0}. \".format(r[basic_columns.index(\"ANN[*].GENE\")])\n",
    "    feat = r[basic_columns.index(\"ANN[*].FEATURE\")]\n",
    "    text += \"It is reported on a {0}. \".format(feat)\n",
    "    biot = r[basic_columns.index(\"ANN[*].BIOTYPE\")]\n",
    "    if biot in gene_biotypes:\n",
    "        text += \"The biotype is {0}. \".format(gene_biotypes[biot].rstrip(\".\"))\n",
    "    eff = r[basic_columns.index(\"ANN[*].EFFECT\")].replace(\"_\", \" \")\n",
    "    imp = r[basic_columns.index(\"ANN[*].IMPACT\")].lower()\n",
    "    rnk = r[basic_columns.index(\"ANN[*].RANK\")]\n",
    "    text += \"The effect is {0} with a {1} impact ranked {2}. \".format(eff, imp, rnk)\n",
    "    hgvs = r[basic_columns.index(\"ANN[*].HGVS_P\")]\n",
    "    if hgvs != \".\":\n",
    "        text += \"The point mutation is {0}. \".format(hgvs)\n",
    "    text = text.rstrip(\" \")\n",
    "    texts += [text]\n",
    "\n",
    "\n",
    "X_basic = embedder.calculate(texts[:10])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Frequency"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "population_frequencies_columns = columns[\n",
    "    slice(columns.index(\"AF\"), columns.index(\"dbNSFP_ExAC_SAS_AF\") + 1)\n",
    "]\n",
    "dfreq = pd.DataFrame(df[population_frequencies_columns])\n",
    "\n",
    "X_freq = np.array(dfreq)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Predictions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "prediction_scores_columns = columns[\n",
    "    slice(columns.index(\"dbNSFP_CADD_phred\"), columns.index(\"dbNSFP_VEST4_score\") + 1)\n",
    "]\n",
    "\n",
    "df[\"dbNSFP_Polyphen2_HDIV_pred\"].replace({\"B\": 0, \"P\": 1, \"D\": 2}, inplace=True)\n",
    "df[\"dbNSFP_SIFT_pred\"].replace({\"D\": 1, \"T\": 0}, inplace=True)\n",
    "df[\"dbNSFP_MutationAssessor_pred\"].replace(\n",
    "    {\"H\": 3, \"M\": 2, \"L\": 1, \"N\": 0}, inplace=True\n",
    ")\n",
    "df[\"dbNSFP_MutationTaster_pred\"].replace({\"A\": 2, \"D\": 1, \"N\": 0, \"P\": 0}, inplace=True)\n",
    "\n",
    "\n",
    "def calculate_value(s):\n",
    "    values = s.split(\",\")\n",
    "    values = [v for v in values if v != \".\"]\n",
    "    if not values:\n",
    "        return float(\"NaN\")\n",
    "    mean_value = sum([1 if v == \"D\" else 0 for v in values]) / len(values)\n",
    "    return mean_value\n",
    "\n",
    "\n",
    "df[\"dbNSFP_PROVEAN_pred\"] = df[\"dbNSFP_PROVEAN_pred\"].apply(calculate_value)\n",
    "dpred = pd.DataFrame(df[prediction_scores_columns])\n",
    "dpred.replace(\".\", np.nan, inplace=True)\n",
    "\n",
    "X_pred = np.array(dpred)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Other columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "0it [00:00, ?it/s]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "10it [00:01,  8.06it/s]\n"
     ]
    }
   ],
   "source": [
    "other_columns = columns[\n",
    "    slice(\n",
    "        columns.index(\"dbNSFP_clinvar_id\"), columns.index(\"dbNSFP_Interpro_domain\") + 1\n",
    "    )\n",
    "]\n",
    "texts = []\n",
    "for v in df[[\"dbNSFP_clinvar_clnsig\", \"dbNSFP_clinvar_trait\", \"COSV_ID\"]].values:\n",
    "    t = \"\"\n",
    "    if v[0] != \".\":\n",
    "        t += \"The clinical relevance of the variant is {0}. \".format(v[0])\n",
    "    else:\n",
    "        t += \"The clinical relevance of the variant is not annotated. \"\n",
    "    if v[1] != \".\":\n",
    "        t += \"The traits are {0}. \".format(v[1].replace(\"_\", \" \"))\n",
    "    else:\n",
    "        t += \"No traits are known. \"\n",
    "    if v[2] != \".\":\n",
    "        t += \"The affected domains of the variant in COSMIC are {0}.\".format(v[2])\n",
    "    else:\n",
    "        t += \"No domains in COSMIC.\"\n",
    "    texts += [t]\n",
    "\n",
    "X_other = embedder.calculate(texts[:10])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ANN[*].EFFECT</th>\n",
       "      <th>ANN[*].IMPACT</th>\n",
       "      <th>ANN[*].RANK</th>\n",
       "      <th>ANN[*].HGVS_C</th>\n",
       "      <th>ANN[*].HGVS_P</th>\n",
       "      <th>ANN[*].CDNA_POS</th>\n",
       "      <th>ANN[*].CDNA_LEN</th>\n",
       "      <th>ANN[*].CDS_POS</th>\n",
       "      <th>ANN[*].CDS_LEN</th>\n",
       "      <th>ANN[*].AA_POS</th>\n",
       "      <th>...</th>\n",
       "      <th>dbNSFP_VEST4_score</th>\n",
       "      <th>dbNSFP_clinvar_id</th>\n",
       "      <th>dbNSFP_clinvar_clnsig</th>\n",
       "      <th>dbNSFP_clinvar_trait</th>\n",
       "      <th>dbNSFP_clinvar_review</th>\n",
       "      <th>dbNSFP_clinvar_MedGen_id</th>\n",
       "      <th>dbNSFP_clinvar_OMIM_id</th>\n",
       "      <th>dbNSFP_clinvar_Orphanet_id</th>\n",
       "      <th>COSV_ID</th>\n",
       "      <th>dbNSFP_Interpro_domain</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1</td>\n",
       "      <td>c.119T&gt;C</td>\n",
       "      <td>p.Ile40Thr</td>\n",
       "      <td>367</td>\n",
       "      <td>3144</td>\n",
       "      <td>119</td>\n",
       "      <td>1230</td>\n",
       "      <td>40</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>2</td>\n",
       "      <td>c.124C&gt;A</td>\n",
       "      <td>p.Gln42Lys</td>\n",
       "      <td>339</td>\n",
       "      <td>5548</td>\n",
       "      <td>124</td>\n",
       "      <td>231</td>\n",
       "      <td>42</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>10</td>\n",
       "      <td>c.3283G&gt;A</td>\n",
       "      <td>p.Ala1095Thr</td>\n",
       "      <td>3590</td>\n",
       "      <td>6105</td>\n",
       "      <td>3283</td>\n",
       "      <td>5691</td>\n",
       "      <td>1095</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>282995</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Three_M_syndrome_2_not_specified_not_provided</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>C2752041_CN169374_CN517202</td>\n",
       "      <td>612921</td>\n",
       "      <td>.</td>\n",
       "      <td>Immunoglobulin-like_domain_Immunoglobulin_subt...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>6</td>\n",
       "      <td>c.581G&gt;A</td>\n",
       "      <td>p.Arg194Lys</td>\n",
       "      <td>691</td>\n",
       "      <td>2513</td>\n",
       "      <td>581</td>\n",
       "      <td>795</td>\n",
       "      <td>194</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>22</td>\n",
       "      <td>c.2182T&gt;C</td>\n",
       "      <td>p.Trp728Arg</td>\n",
       "      <td>2212</td>\n",
       "      <td>2708</td>\n",
       "      <td>2182</td>\n",
       "      <td>2520</td>\n",
       "      <td>728</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Vps16__C-terminal</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9995</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>2</td>\n",
       "      <td>c.185A&gt;G</td>\n",
       "      <td>p.Asn62Ser</td>\n",
       "      <td>220</td>\n",
       "      <td>8432</td>\n",
       "      <td>185</td>\n",
       "      <td>2481</td>\n",
       "      <td>62</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>WD40-repeat-containing_domain</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9996</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>16</td>\n",
       "      <td>c.1742C&gt;T</td>\n",
       "      <td>p.Pro581Leu</td>\n",
       "      <td>1772</td>\n",
       "      <td>2494</td>\n",
       "      <td>1742</td>\n",
       "      <td>2271</td>\n",
       "      <td>581</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>41635</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Lynch_syndrome_Lynch_syndrome_II_Hereditary_ca...</td>\n",
       "      <td>reviewed_by_expert_panel</td>\n",
       "      <td>C4552100_C1333991_C0027672_C0009405_CN169374_C...</td>\n",
       "      <td>PS120435_609310</td>\n",
       "      <td>ORPHA144_ORPHA140162</td>\n",
       "      <td>DNA_mismatch_repair_protein_Mlh1__C-terminal</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9997</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>2</td>\n",
       "      <td>c.197T&gt;C</td>\n",
       "      <td>p.Ile66Thr</td>\n",
       "      <td>328</td>\n",
       "      <td>2067</td>\n",
       "      <td>197</td>\n",
       "      <td>1260</td>\n",
       "      <td>66</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Immunoglobulin_V-set_domain_Immunoglobulin_sub...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9998</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>8</td>\n",
       "      <td>c.1310C&gt;G</td>\n",
       "      <td>p.Ala437Gly</td>\n",
       "      <td>1556</td>\n",
       "      <td>2294</td>\n",
       "      <td>1310</td>\n",
       "      <td>1419</td>\n",
       "      <td>437</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9999</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>23</td>\n",
       "      <td>c.4498G&gt;A</td>\n",
       "      <td>p.Asp1500Asn</td>\n",
       "      <td>4657</td>\n",
       "      <td>21079</td>\n",
       "      <td>4498</td>\n",
       "      <td>15624</td>\n",
       "      <td>1500</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10000 rows × 134 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         ANN[*].EFFECT ANN[*].IMPACT  ANN[*].RANK ANN[*].HGVS_C ANN[*].HGVS_P  \\\n",
       "0     missense_variant      MODERATE            1      c.119T>C    p.Ile40Thr   \n",
       "1     missense_variant      MODERATE            2      c.124C>A    p.Gln42Lys   \n",
       "2     missense_variant      MODERATE           10     c.3283G>A  p.Ala1095Thr   \n",
       "3     missense_variant      MODERATE            6      c.581G>A   p.Arg194Lys   \n",
       "4     missense_variant      MODERATE           22     c.2182T>C   p.Trp728Arg   \n",
       "...                ...           ...          ...           ...           ...   \n",
       "9995  missense_variant      MODERATE            2      c.185A>G    p.Asn62Ser   \n",
       "9996  missense_variant      MODERATE           16     c.1742C>T   p.Pro581Leu   \n",
       "9997  missense_variant      MODERATE            2      c.197T>C    p.Ile66Thr   \n",
       "9998  missense_variant      MODERATE            8     c.1310C>G   p.Ala437Gly   \n",
       "9999  missense_variant      MODERATE           23     c.4498G>A  p.Asp1500Asn   \n",
       "\n",
       "      ANN[*].CDNA_POS  ANN[*].CDNA_LEN  ANN[*].CDS_POS  ANN[*].CDS_LEN  \\\n",
       "0                 367             3144             119            1230   \n",
       "1                 339             5548             124             231   \n",
       "2                3590             6105            3283            5691   \n",
       "3                 691             2513             581             795   \n",
       "4                2212             2708            2182            2520   \n",
       "...               ...              ...             ...             ...   \n",
       "9995              220             8432             185            2481   \n",
       "9996             1772             2494            1742            2271   \n",
       "9997              328             2067             197            1260   \n",
       "9998             1556             2294            1310            1419   \n",
       "9999             4657            21079            4498           15624   \n",
       "\n",
       "      ANN[*].AA_POS  ...  dbNSFP_VEST4_score  dbNSFP_clinvar_id  \\\n",
       "0                40  ...                   .                  .   \n",
       "1                42  ...                   .                  .   \n",
       "2              1095  ...                   .             282995   \n",
       "3               194  ...                   .                  .   \n",
       "4               728  ...                   .                  .   \n",
       "...             ...  ...                 ...                ...   \n",
       "9995             62  ...                   .                  .   \n",
       "9996            581  ...                   .              41635   \n",
       "9997             66  ...                   .                  .   \n",
       "9998            437  ...                   .                  .   \n",
       "9999           1500  ...                   .                  .   \n",
       "\n",
       "     dbNSFP_clinvar_clnsig                               dbNSFP_clinvar_trait  \\\n",
       "0                        .                                                  .   \n",
       "1                        .                                                  .   \n",
       "2                   Benign      Three_M_syndrome_2_not_specified_not_provided   \n",
       "3                        .                                                  .   \n",
       "4                        .                                                  .   \n",
       "...                    ...                                                ...   \n",
       "9995                     .                                                  .   \n",
       "9996         Likely_benign  Lynch_syndrome_Lynch_syndrome_II_Hereditary_ca...   \n",
       "9997                     .                                                  .   \n",
       "9998                     .                                                  .   \n",
       "9999                     .                                                  .   \n",
       "\n",
       "                                  dbNSFP_clinvar_review  \\\n",
       "0                                                     .   \n",
       "1                                                     .   \n",
       "2     criteria_provided,_multiple_submitters,_no_con...   \n",
       "3                                                     .   \n",
       "4                                                     .   \n",
       "...                                                 ...   \n",
       "9995                                                  .   \n",
       "9996                           reviewed_by_expert_panel   \n",
       "9997                                                  .   \n",
       "9998                                                  .   \n",
       "9999                                                  .   \n",
       "\n",
       "                               dbNSFP_clinvar_MedGen_id  \\\n",
       "0                                                     .   \n",
       "1                                                     .   \n",
       "2                            C2752041_CN169374_CN517202   \n",
       "3                                                     .   \n",
       "4                                                     .   \n",
       "...                                                 ...   \n",
       "9995                                                  .   \n",
       "9996  C4552100_C1333991_C0027672_C0009405_CN169374_C...   \n",
       "9997                                                  .   \n",
       "9998                                                  .   \n",
       "9999                                                  .   \n",
       "\n",
       "      dbNSFP_clinvar_OMIM_id  dbNSFP_clinvar_Orphanet_id  \\\n",
       "0                          .                           .   \n",
       "1                          .                           .   \n",
       "2                     612921                           .   \n",
       "3                          .                           .   \n",
       "4                          .                           .   \n",
       "...                      ...                         ...   \n",
       "9995                       .                           .   \n",
       "9996         PS120435_609310        ORPHA144_ORPHA140162   \n",
       "9997                       .                           .   \n",
       "9998                       .                           .   \n",
       "9999                       .                           .   \n",
       "\n",
       "                                                COSV_ID  \\\n",
       "0                                                     .   \n",
       "1                                                     .   \n",
       "2     Immunoglobulin-like_domain_Immunoglobulin_subt...   \n",
       "3                                                     .   \n",
       "4                                     Vps16__C-terminal   \n",
       "...                                                 ...   \n",
       "9995                      WD40-repeat-containing_domain   \n",
       "9996       DNA_mismatch_repair_protein_Mlh1__C-terminal   \n",
       "9997  Immunoglobulin_V-set_domain_Immunoglobulin_sub...   \n",
       "9998                                                  .   \n",
       "9999                                                  .   \n",
       "\n",
       "      dbNSFP_Interpro_domain  \n",
       "0                        NaN  \n",
       "1                        NaN  \n",
       "2                        NaN  \n",
       "3                        NaN  \n",
       "4                        NaN  \n",
       "...                      ...  \n",
       "9995                     NaN  \n",
       "9996                     NaN  \n",
       "9997                     NaN  \n",
       "9998                     NaN  \n",
       "9999                     NaN  \n",
       "\n",
       "[10000 rows x 134 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[columns[10:]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0         POU4F2\n",
       "1       C18orf32\n",
       "2          OBSL1\n",
       "3         COPS7B\n",
       "4          VPS16\n",
       "          ...   \n",
       "9995        ELP2\n",
       "9996        MLH1\n",
       "9997        PSG7\n",
       "9998      MBOAT7\n",
       "9999      RNF213\n",
       "Name: ANN[*].GENE, Length: 10000, dtype: object"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"ANN[*].GENE\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "biogpt",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.10"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
