# Drug of interest
My drug of interest is ethionamide.

# Pre-ranked list of genes
These are the candidate genes for this drugs, tentatively ranked by the likelihood of a pharmacogenetic relationship:
1. NAT2 (z: 3.93)
2. CYP2C19 (z: 3.62)
3. CYP2A6 (z: 2.59)
4. NAT1 (z: 2.04)
5. CYP2B6 (z: 1.95)
6. CYP2C9 (z: 1.46)
7. FMO4 (z: 1.43)
8. ABCB1 (z: 1.08)
9. CYP2A7 (z: 1.06)
10. GSTT1 (z: 0.94)
11. CYP2A13 (z: 0.93)
12. GSTM1 (z: 0.87)
13. HLA-B (z: 0.84)
14. CYP2E1 (z: 0.83)
15. CYP2F1 (z: 0.81)
16. CYP2D7 (z: 0.81)
17. CYP2S1 (z: 0.7)
18. CYP2D6 (z: 0.68)
19. HLA-DRB1 (z: 0.66)
20. CYP2J2 (z: 0.65)
21. GSTA3 (z: 0.62)
22. HLA-C (z: 0.58)
23. GSTM2 (z: 0.58)
24. NR1I2 (z: 0.57)
25. MT-RNR1 (z: 0.56)
26. FMO5 (z: 0.53)
27. HLA-E (z: 0.52)
28. GSTM4 (z: 0.51)
29. HLA-DQB1 (z: 0.49)
30. GSTT2 (z: 0.49)
31. G6PD (z: 0.48)
32. GSTM3 (z: 0.46)
33. HLA-G (z: 0.46)
34. HLA-A (z: 0.45)
35. GSTP1 (z: 0.43)
36. GSTA2 (z: 0.43)
37. GSTM5 (z: 0.43)
38. CYP1A1 (z: 0.42)
39. GSTA5 (z: 0.42)
40. SLCO1B1 (z: 0.42)
41. HLA-DRB3 (z: 0.37)
42. HLA-DRA (z: 0.37)
43. FMO6P (z: 0.37)
44. CYP2C8 (z: 0.35)
45. GSTA1 (z: 0.34)
46. GSTA4 (z: 0.34)
47. C5orf56 (z: 0.33)
48. HLA-DQA1 (z: 0.33)
49. CYP2C18 (z: 0.33)
50. HLA-DRB4 (z: 0.32)

# Auxiliary information about the drug
Below is some auxiliary information about the drug of interest:

**ethionamide**: Ethionamide, a drug used against tuberculosis, is metabolized in the liver with varying efficacy due to genetic variations in the FMO2 enzyme, which can affect its activation and thus its overall effectiveness. Additionally, Ethionamide's action on tuberculosis bacteria involves targeting specific bacterial enzymes like inhA and can be compromised by resistance mutations in katG, affecting its therapeutic efficacy and potential cross-resistance with similar drugs such as isoniazid.

# Auxiliary information about the genes
Below is some auxiliary information about the genes in the pre-ranked list:
**NAT2**: The NAT2 gene encodes an enzyme crucial for the metabolism of various drugs through N-acetylation, influencing their pharmacokinetics and toxicity. Variants in NAT2 affect the metabolism of drugs like isoniazid, hydralazine, procainamide, sulfamethoxazole, and pyrazinamide, leading to differences in acetylator status (rapid, intermediate, slow) among individuals, which impacts drug plasma levels, exposure duration, and risk of adverse effects such as increased toxicity and drug-induced lupus.

**CYP2C19**: The enzyme CYP2C19 significantly influences the metabolism of various drugs including clopidogrel, proton pump inhibitors (omeprazole, esomeprazole), antidepressants (citalopram, escitalopram), antiepileptics (phenytoin), antifungals (voriconazole), and analgesics (methadone, oxycodone), which impacts their therapeutic efficacy and safety. Variations in the CYP2C19 gene affect drug activation and metabolism, leading to different pharmacological outcomes and necessitating genotype-guided dosing to optimize treatment efficacy and minimize adverse effects.

**CYP2A6**: CYP2A6, an enzyme from the cytochrome P450 superfamily, influences the metabolism of various drugs, impacting their efficacy and toxicity due to genetic variations. It plays a critical role in metabolizing smoking cessation aids like nicotine, cancer treatments such as cyclophosphamide and tegafur, HIV treatment efavirenz, and the glaucoma drug pilocarpine, necessitating dosage adjustments or changes in drug choices to optimize outcomes and minimize side effects.

**NAT1**: The NAT1 enzyme, which catalyzes the acetylation of various substrates, affects the metabolism and efficacy of several drugs including cotinine, thioguanine, and mesalazine. Variations in NAT1 activity influence the effectiveness and side effect profiles of these drugs by altering their metabolic pathways, which is critical for determining optimal dosages and therapeutic strategies tailored to individual genetic profiles.

**CYP2B6**: CYP2B6 plays a pivotal role in the metabolism of several drugs, including efavirenz, nevirapine, methadone, bupropion, cyclophosphamide, and ifosfamide, where genetic polymorphisms can significantly influence drug efficacy and safety by altering pharmacokinetic profiles. Genetic variations in CYP2B6 affect the enzyme activity, leading to differences in drug plasma concentrations, therapeutic effects, and potential adverse reactions in individuals, underscoring its importance in personalized medicine.

**CYP2C9**: CYP2C9 gene polymorphisms influence the metabolism of various drugs, affecting their therapeutic effectiveness and risk of side effects. Drugs such as warfarin, phenytoin, celecoxib, losartan, and tolbutamide require dose customization based on CYP2C9 genetic variants to optimize treatment outcomes and minimize complications like bleeding, drug toxicity, and gastrointestinal issues.

**FMO4**: The FMO4 gene encodes for the enzyme flavin-containing monooxygenase 4, which potentially contributes to the metabolism of drugs by affecting their pharmacokinetics, altering plasma drug levels, and impacting therapeutic outcomes or side effects. The exact drugs impacted by FMO4 are still speculative, as FMO4â€™s interactions with specific drugs are currently not well-documented and remain under research.

**ABCB1**: The ABCB1 gene, encoding P-glycoprotein, is crucial in modulating the pharmacokinetics and efficacy of multiple drugs by acting as an efflux transporter that influences their absorption, distribution, and excretion. This gene affects the bioavailability and brain penetration of various drugs such as antivirals like efavirenz and ritonavir, chemotherapeutics like doxorubicin and paclitaxel, cardiovascular drugs like digoxin, immunosuppressants such as tacrolimus and sirolimus, and analgesics like morphine and codeine, necessitating dosage adjustments and careful monitoring in clinical practice for optimized therapeutic outcomes.

**CYP2A7**: CYP2A7, a lesser-known member of the cytochrome P450 family, does not have well-established pharmacogenetic interactions with drugs like mercaptopurine, although it might theoretically affect mercaptopurine's metabolism similarly to its related isoform, CYP2A6. The interaction and impact of CYP2A7 on the metabolism of mercaptopurine are speculative and require further evidence, as mercaptopurine is primarily metabolized through other pathways involving TPMT and XO.

**GSTT1**: The GSTT1 gene influences the pharmacokinetics and detoxification of various drugs, such as isoniazid, rifampin, and multiple chemotherapeutic agents including doxorubicin and oxaliplatin, due to its role in enzymatic detoxification processes. This variation can lead to differences in drug efficacy and an increased risk of adverse effects, particularly in individuals with the null genotype that results in no enzyme expression.

**CYP2A13**: CYP2A13, a cytochrome P450 enzyme primarily found in the respiratory tract, plays a significant role in activating procarcinogens in tobacco smoke, such as nitrosamines, impacting lung cancer risk among smokers. Although it does not metabolize nicotine directly, CYP2A13 affects the activation of other smoke components that interact with nicotine's pathway, potentially influencing nicotine addiction and cancer risks, with genetic variations in CYP2A13 altering susceptibility to these effects.

**GSTM1**: GSTM1 plays a crucial role in the metabolism of several drugs, such as rifampin, busulfan, azathioprine, and isoniazid, with the GSTM1 null genotype leading to reduced enzyme activity that can increase plasma levels, slow elimination rates, and potentially result in higher drug exposure and toxicity. This impaired detoxification can influence both the efficacy and adverse effects of these drugs, underscoring the importance of considering GSTM1 status when managing medications to tailor drug dosages and minimize side effects.

**HLA-B**: The HLA-B gene plays a crucial role in drug hypersensitivity reactions through allele-specific interactions, such as the HLA-B*57:01 allele causing severe reactions to abacavir and the HLA-B*15:02 allele being associated with severe adverse reactions to carbamazepine. These interactions, which involve the immune system's response to drug-modified peptides or the drug itself presented by the HLA-B protein, emphasize the importance of genetic testing in precision medicine to tailor drug therapy and minimize adverse reactions based on individual genetic profiles.

**CYP2E1**: The gene CYP2E1 significantly impacts the metabolism of various drugs; for instance, in the case of acetaminophen and isoniazid, variations in CYP2E1 activity can lead to differences in the rate at which these drugs are metabolized, thereby affecting their efficacy, toxicity, and risk of adverse effects such as hepatotoxicity. Additionally, CYP2E1 metabolizes other substances like caffeine, theophylline, nicotine, and some procarcinogens, indicating that genetic variations could influence both therapeutic outcomes and disease risk through pharmacokinetic and possibly pharmacodynamic interactions.

**CYP2F1**: CYP2F1, involved mainly in metabolizing inhaled environmental pollutants like naphthalene, has limited direct interaction with common drugs but plays a role in the metabolism of certain pharmaceuticals such as imatinib due to shared metabolic pathways or competitive inhibition. Although CYP2F1 does not typically directly influence imatinib's metabolism, its interaction with similar chemical structures could alter the drug's efficacy or toxicity in individuals with specific gene variants.

**CYP2D7**: CYP2D7, a pseudogene, does not encode a functional enzyme and thus does not partake in the metabolism of drugs, unlike its close relative CYP2D6, which metabolizes about 25% of clinically used drugs such as antidepressants, antipsychotics, beta-blockers, and opioids. Although CYP2D7 does not directly interact with these drugs, its similarity to CYP2D6 makes its study important for ensuring accurate pharmacogenetic testing and understanding genetic misannotations which can affect drug dosing and therapy.

**CYP2S1**: CYP2S1, a cytochrome P450 enzyme, influences the metabolism and efficacy of drugs, especially those administered via inhalation or topically, due to its expression in the liver, lungs, and skin. Although specific drugs affected by CYP2S1 are not well-determined, variations in the CYP2S1 gene can alter how these drugs are metabolized, potentially affecting their efficacy and toxicity.

**CYP2D6**: CYP2D6 enzyme is crucial in metabolizing drugs, including codeine, antidepressants like paroxetine and fluoxetine, the antipsychotic risperidone, tamoxifen, and beta-blockers such as metoprolol and timolol, with genetic variations influencing their plasma levels and effects. Differences in CYP2D6 activity can lead to different therapeutic outcomes, with poor metabolizers experiencing potential toxicity and ultrarapid metabolizers having reduced drug efficacy, making pharmacogenetic knowledge essential for adjusting dosages or choosing alternative therapies to optimize treatment efficacy and safety.

**HLA-DRB1**: The HLA-DRB1 gene, through its various alleles, significantly influences the pharmacogenetic interactions that affect the efficacy and safety of medications, causing immune-mediated hypersensitivity reactions. For instance, the HLA-DRB1*15:01 allele is linked to abacavir hypersensitivity in HIV treatment, while HLA-DRB1*07:01 is associated with hepatic injuries from ximelagatran, highlighting the importance of understanding a patient's HLA-DRB1 allele status to optimize and personalize treatment strategies.

**CYP2J2**: CYP2J2, primarily involved in metabolizing polyunsaturated fatty acids affecting cardiovascular physiology, does not have well-documented direct pharmacokinetic interactions with drugs like mycophenolate mofetil and tacrolimus. However, its role in modulating vascular tone and inflammation could potentially influence the pharmacodynamics of these immunosuppressive drugs used in organ transplantation and autoimmune diseases, despite the lack of a direct metabolic pathway involving CYP2J2.

**GSTA3**: GSTA3, part of the glutathione S-transferase family, plays a critical role in detoxifying xenobiotics and drugs, particularly by enhancing the solubility and excretion of agents susceptible to conjugation with glutathione such as certain antineoplastic agents. This enzymatic activity could reduce the effectiveness of drugs like alkylating agents used in chemotherapy by accelerating their detoxification, impacting therapeutic outcomes.

**HLA-C**: HLA-C plays a crucial role in drug hypersensitivity reactions, primarily through immune response mechanisms influenced by its polymorphic nature. The presence of the HLA-C*06:02 allele, for example, is strongly associated with hypersensitivity to abacavir, used in HIV treatment, necessitating genetic screening to prevent severe reactions. Other drugs such as carbamazepine, allopurinol, clozapine, and phenytoin also show interactions with various HLA alleles that might contribute to adverse reactions like Stevens-Johnson syndrome and toxic epidermal necrolysis.

**GSTM2**: The gene GSTM2, instrumental in detoxifying substances, impacts the efficacy and toxicity of chemotherapy drugs like cisplatin and busulfan by influencing their metabolism. Cisplatin's side effects may be moderated by GSTM2, whereas busulfan's metabolism alterations can affect both its effectiveness and toxicity, stressing the importance of considering individual genetic differences in treatment plans.

**NR1I2**: NR1I2, or pregnane X receptor (PXR), regulates genes encoding drug-metabolizing enzymes and transport systems like CYP3A4 and MDR1, influencing the pharmacokinetics of various drugs. Activators of NR1I2 such as rifampin and carbamazepine can increase the metabolism and decrease the efficacy of drugs like sirolimus and rosuvastatin by upregulating CYP3A4, while inhibitors can slow metabolism, increasing toxicity risk.

**MT-RNR1**: The MT-RNR1 gene mutation m.1555A>G heightens susceptibility to ototoxic effects from aminoglycoside antibiotics such as gentamicin and tobramycin due to increased sensitivity of the mitochondrial ribosome, leading to impaired protein synthesis in cochlear cells and increasing the risk of irreversible hearing loss. The interaction between MT-RNR1 and aminoglycosides is pharmacodynamic as it influences the drug's action on the mitochondrial ribosome, rather than affecting the drug's metabolism or disposition.

**FMO5**: The FMO5 gene, unlike its family members, has a unique role in pharmacogenetics, particularly due to its broad substrate specificity and presence in the liver and small intestine. It indirectly influences drug metabolism, such as with metformin, by potentially altering the metabolic environment or modifying intermediate biomolecules, thus affecting the drug's efficacy and safety through non-traditional pathways related to its handling of metabolic byproducts connected to sulfur and nitrogen.

**HLA-E**: The gene HLA-E, which plays a significant role in immune system regulation, may interact pharmacodynamically with methotrexate, a drug used for autoimmune diseases and cancer. This interaction likely occurs as methotrexate modulates immune responses that are under the control of HLA-E, affecting drug efficacy and potential adverse effects rather than affecting the drug's absorption, distribution, metabolism, or excretion.

**GSTM4**: GSTM4 is crucial in detoxifying chemotherapeutic agents and anti-inflammatory drugs by conjugating glutathione to enhance solubility and promote excretion, which impacts drug effectiveness and toxicity. This gene may interact with drugs like cisplatin and busulfan through this pathway, and variations in GSTM4 can influence drug response and the profile of adverse effects by altering the efficiency of detoxification.

**HLA-DQB1**: HLA-DQB1 gene variants significantly impact drug hypersensitivity reactions, particularly for drugs like abacavir, allopurinol, nevirapine, lamotrigine, and carbamazepine, which can cause severe reactions such as Steven-Johnson syndrome and toxic epidermal necrolysis in genetically predisposed individuals. These interactions emphasize the role of HLA-DQB1 in pharmacodynamics, affecting drug safety and efficacy rather than drug metabolism.

**GSTT2**: The GSTT2 gene, involved in cellular detoxification by aiding the conjugation of glutathione to various compounds, influences the pharmacokinetics and detoxification of chemotherapeutic drugs such as busulfan and chlorambucil, which undergo glutathione-related reactions. This suggests that genetic variations in GSTT2 could impact drug clearance and toxicity, potentially affecting treatment efficacy and susceptibility to drug-induced oxidative stress in chemotherapy.

**G6PD**: Certain medications exacerbate hemolysis in individuals with G6PD deficiency due to their effects on oxidative stress and redox balance within cells. Notably, antimalarials like primaquine and chloroquine, antibiotics such as dapsone, sulfasalazine, and sulfamethoxazole, NSAIDs including aspirin and ibuprofen, and potentially diabetic medications like gliclazide and glimepiride, as well as anticoagulants such as warfarin, can precipitate hemolytic reactions, necessitating careful management to avoid these drugs in affected individuals.

**GSTM3**: GSTM3, an enzyme involved in drug metabolism and detoxification, plays a key role in the body's handling of olanzapine and cyclophosphamide by conjugating toxins with glutathione, which affects these drugs' solubility and excretion. For olanzapine, GSTM3 may alter the drug's clearance and efficacy, while for cyclophosphamide, it is crucial in detoxifying metabolites that contribute to the drugâ€™s cytotoxic effects, thus impacting the drug's effectiveness or toxicity.

**HLA-G**: HLA-G's potential pharmacodynamic interactions with drugs like fluorouracil, irinotecan, methotrexate, and capecitabine, primarily used in chemotherapy, relate to its role in modulating immune responses which could influence the immune system's ability to recognize and destroy tumor cells. These interactions likely result from HLA-G enhancing immune evasion capabilities in cancer cells, thereby potentially modifying the therapeutic efficacy of the mentioned drugs.

**HLA-A**: The HLA-A gene is crucial in pharmacogenetic interactions due to its role in immune-mediated hypersensitivity reactions to certain drugs. For instance, the HLA-A*31:01 allele increases the risk of severe skin reactions to carbamazepine, while the HLA-A*02:01 allele is linked to severe adverse reactions to allopurinol, both through mechanisms involving the immune system's response to these drugs or their metabolites. Other drugs such as abacavir, phenytoin, and flucloxacillin among others are also implicated in similar immune responses based on the HLA-A gene variations, affecting drug pharmacodynamics rather than pharmacokinetics.

**GSTP1**: GSTP1 significantly influences the pharmacogenetics and resultant efficacy of various drugs by modifying their metabolism and toxicity, as seen in drugs like oxaliplatin, thiotepa, olanzapine, and cyclophosphamide through enzymatic detoxification processes such as glutathione conjugation. Additionally, GSTP1 impacts the resistance mechanisms to chemotherapy drugs such as doxorubicin, epirubicin, and irinotecan, potentially reducing their effectiveness, while also affecting the efficacy and safety of drugs in other therapeutic areas like anesthesia and immunosuppression.

**GSTA2**: GSTA2 plays a crucial role in the metabolism and detoxification of drugs such as acetaminophen and chemotherapeutic agents by catalyzing their conjugation with glutathione, influencing drug elimination and reducing toxicity. Polymorphisms in GSTA2 can affect the enzyme's efficiency, leading to variations in the detoxification rate, which may alter drug responses and increase toxicity risks in individuals.

**GSTM5**: The GSTM5 gene, which encodes the enzyme Glutathione S-transferase Mu 5, plays a significant role in detoxifying and metabolizing drugs, including the chemotherapy agent busulfan. Variations in GSTM5 can influence the pharmacokinetics of busulfan, affecting its efficacy and toxicity by altering the detoxification process.

**CYP1A1**: CYP1A1, an important enzyme in drug metabolism, exhibits pharmacogenetic interactions that impact the efficacy and toxicity of drugs like theophylline and clozapine through variations in metabolic activity based on genetic differences. Drugs such as deferasirox, valproic acid, and caffeine among others show altered pharmacokinetics when metabolized by CYP1A1, while others like imatinib and nilotinib might have altered pharmacodynamics or metabolism influenced by CYP1A1 activity, affecting treatment outcomes especially in cancer therapy.

**GSTA5**: GSTA5, which is involved in detoxification by catalyzing the conjugation of glutathione to compounds, plays a role in how drugs like busulfan are metabolized. Genetic variations in GSTA5 could influence the pharmacokinetics of busulfan, impacting its efficacy and toxicity, thus potentially optimizing dosing and improving therapeutic outcomes.

**SLCO1B1**: The SLCO1B1 gene, which facilitates the hepatic uptake of drugs, exhibits polymorphisms like c.521T>C that affect the pharmacokinetics of multiple drugs including statins, repaglinide, and rifampin, potentially leading to increased drug plasma concentrations, altered efficacy, and toxicity. This variability in drug transport efficiency necessitates adjustments in dosing to optimize therapeutic outcomes and minimize adverse effects for drugs such as simvastatin, atorvastatin, and others.

**HLA-DRB3**: HLA-DRB3, a gene involved in immune system response, interacts with drugs not by being directly targeted but by influencing how the immune system interacts with these drugs, thus modulating their effects and potentially leading to drug hypersensitivity reactions. For example, variations in HLA-DRB3 are linked to different responses and adverse reactions to drugs like clozapine, where a connection is suggested between HLA alleles and clozapine-induced agranulocytosis, highlighting the gene's role in drug efficacy and safety.

**HLA-DRA**: The HLA-DRA gene, though not directly interacting with drugs through pharmacokinetic pathways, plays an indirect role in pharmacodynamics due to its involvement in the immune system. It is implicated in the immune-mediated drug reactions, such as hypersensitivities to antibiotics like amoxicillin, flucloxacillin, and clavulanate, possibly by affecting how drug-related antigens are presented to the immune system by antigen-presenting cells, thereby influencing the likelihood or severity of allergic reactions.

**FMO6P**: FMO6P, a pseudogene, does not encode a functional protein and hence does not interact in the metabolism of drugs or affect drug actions in the body, unlike its functional counterparts like FMO3 and FMO5. This lack of function is due to evolutionary mutations that inhibit proper gene expression, rendering FMO6P inactive in pharmacokinetic and pharmacodynamic processes.

**CYP2C8**: The CYP2C8 gene plays a crucial role in metabolizing various medications such as paclitaxel, pioglitazone, repaglinide, and amodiaquine, affecting their efficacy and toxicity due to genetic variations that influence drug metabolism and exposure. These interactions, involving pharmacokinetic processes, suggest that CYP2C8 genetic testing could be utilized to optimize drug dosing and reduce toxicity.

**GSTA1**: GSTA1 (Glutathione S-transferase alpha 1) plays a key role in the metabolism and detoxification of drugs like busulfan and cyclophosphamide through its enzyme activity, which involves conjugating reduced glutathione with various electrophilic compounds. Variations in the GSTA1 gene can affect the efficacy and toxicity of these drugs by altering their metabolism, and this gene may also influence the response to other drugs such as prednisone, doxorubicin, and vincristine, possibly through mechanisms related to oxidative stress modulation.

**GSTA4**: The gene GSTA4 plays a significant role in detoxifying various compounds by facilitating the conjugation of glutathione to electrophilic substances, impacting the pharmacokinetics and efficacy of chemotherapy drugs, particularly alkylating agents. Variations in GSTA4 function or expression can alter the metabolism of these drugs, potentially affecting their effectiveness and toxicity by changing how quickly and extensively the drugs are processed.

**C5orf56**: C5orf56, a gene implicated in inflammatory and autoimmune processes, might affect the response to drugs such as hydrochlorothiazide through pharmacodynamic interactions relating to its role in inflammation, rather than through alterations in drug metabolism. These potential interactions, which are yet speculative, suggest that C5orf56 could modulate the effects of hydrochlorothiazide on blood pressure and cardiovascular health via immune modulation or inflammation.

**HLA-DQA1**: Specific HLA-DQA1 alleles are linked to drug-induced hypersensitivity reactions with medications such as abacavir, flucloxacillin, lumiracoxib, and others. These alleles influence the immune system's response to these drugs, leading to heightened or adverse reactions, emphasizing the role of HLA-DQA1 in modulating the immune response to certain medications.

**CYP2C18**: CYP2C18 plays a modest role in the metabolism of drugs like warfarin, escitalopram, and clozapine, mainly affecting their clearance or efficacy to a lesser extent than more prominent CYP enzymes. Genetic variations in CYP2C18 could influence the pharmacodynamics of drugs such as clopidogrel and acenocoumarol, rather than their pharmacokinetic profiles, indicating a potential need for personalized medicine considerations despite the enzyme's relative minor role.

**HLA-DRB4**: HLA-DRB4 gene primarily affects the immune system and influences the treatment response to disease-modifying antirheumatic drugs (DMARDs) in conditions like rheumatoid arthritis by affecting how specific alleles impact disease progression and drug responsiveness. While HLA-DRB4 does not directly interact with drugs, its role may affect the efficacy and adverse effects of treatments such as DMARDs and indirectly influence immune responses related to drugs like clozapine, though robust data on direct interactions is lacking.
