# Drug of interest
My drug of interest is pyrazinamide.

# Pre-ranked list of genes
These are the candidate genes for this drugs, tentatively ranked by the likelihood of a pharmacogenetic relationship:
1. NAT2 (z: 32.54)
2. NAT1 (z: 6.99)
3. CYP2E1 (z: 2.55)
4. CYP2A6 (z: 2.5)
5. GSTT1 (z: 2.43)
6. GSTM1 (z: 2.32)
7. CYP2C19 (z: 2.12)
8. CYP2C9 (z: 2.01)
9. CYP2B6 (z: 1.99)
10. TNF (z: 1.92)
11. GSTA3 (z: 1.11)
12. GSTM2 (z: 1.03)
13. CYP2A7 (z: 0.92)
14. GSTT2 (z: 0.91)
15. GSTM5 (z: 0.89)
16. GSTM4 (z: 0.87)
17. GSTA5 (z: 0.87)
18. GSTM3 (z: 0.86)
19. CYP2F1 (z: 0.86)
20. CYP2J2 (z: 0.83)
21. ATAT1 (z: 0.8)
22. GSTA4 (z: 0.79)
23. ABCB1 (z: 0.75)
24. CYP2A13 (z: 0.72)
25. CYP1A1 (z: 0.67)
26. CERS6 (z: 0.66)
27. CYP2D7 (z: 0.66)
28. MT-RNR1 (z: 0.64)
29. GSTA1 (z: 0.64)
30. CYP2C8 (z: 0.62)
31. MGST2 (z: 0.62)
32. GSTA2 (z: 0.62)
33. CYP2S1 (z: 0.62)
34. ADORA2A-AS1 (z: 0.58)
35. GSTP1 (z: 0.52)
36. HCP5 (z: 0.51)
37. CYP1A2 (z: 0.5)
38. CHAT (z: 0.48)
39. CYP2C18 (z: 0.48)
40. C5orf56 (z: 0.47)
41. MGST3 (z: 0.47)
42. MBOAT1 (z: 0.45)
43. CYP3A4 (z: 0.43)
44. TPMT (z: 0.42)
45. ABCC2 (z: 0.41)
46. GSTK1 (z: 0.39)
47. NR1I2 (z: 0.38)
48. HLA-B (z: 0.37)
49. DAOA (z: 0.37)
50. GSS (z: 0.35)

# Known pharmacogenetic associations
In addition, please consider that the following pharmacogenetic interactions are already known for this drug, according to PharmGKB:
- NAT2

# Auxiliary information about the drug
Below is some auxiliary information about the drug of interest:

**pyrazinamide**: Pyrazinamide, an antitubercular drug, is metabolized in the liver by enzymes such as AOX1 and XDH, with genetic variants in XDH influencing the enzyme activity and potentially affecting the drug's pharmacokinetics and toxicity. Additionally, variations in NAT2 affect the drug's acetylation process, impacting its efficacy and side effect profile, particularly hepatotoxicity, while GSTM1 may also play a role in the drug’s metabolism and disposition. Other genes like CYP2E1, CYP2C19, and TNF could be involved, but their direct impact on pyrazinamide's pharmacokinetic profile is less clearly established.

# Auxiliary information about the genes
Below is some auxiliary information about the genes in the pre-ranked list:
**NAT2**: The NAT2 gene encodes an enzyme crucial for the metabolism of various drugs through N-acetylation, influencing their pharmacokinetics and toxicity. Variants in NAT2 affect the metabolism of drugs like isoniazid, hydralazine, procainamide, sulfamethoxazole, and pyrazinamide, leading to differences in acetylator status (rapid, intermediate, slow) among individuals, which impacts drug plasma levels, exposure duration, and risk of adverse effects such as increased toxicity and drug-induced lupus.

**NAT1**: The NAT1 enzyme, which catalyzes the acetylation of various substrates, affects the metabolism and efficacy of several drugs including cotinine, thioguanine, and mesalazine. Variations in NAT1 activity influence the effectiveness and side effect profiles of these drugs by altering their metabolic pathways, which is critical for determining optimal dosages and therapeutic strategies tailored to individual genetic profiles.

**CYP2E1**: The gene CYP2E1 significantly impacts the metabolism of various drugs; for instance, in the case of acetaminophen and isoniazid, variations in CYP2E1 activity can lead to differences in the rate at which these drugs are metabolized, thereby affecting their efficacy, toxicity, and risk of adverse effects such as hepatotoxicity. Additionally, CYP2E1 metabolizes other substances like caffeine, theophylline, nicotine, and some procarcinogens, indicating that genetic variations could influence both therapeutic outcomes and disease risk through pharmacokinetic and possibly pharmacodynamic interactions.

**CYP2A6**: CYP2A6, an enzyme from the cytochrome P450 superfamily, influences the metabolism of various drugs, impacting their efficacy and toxicity due to genetic variations. It plays a critical role in metabolizing smoking cessation aids like nicotine, cancer treatments such as cyclophosphamide and tegafur, HIV treatment efavirenz, and the glaucoma drug pilocarpine, necessitating dosage adjustments or changes in drug choices to optimize outcomes and minimize side effects.

**GSTT1**: The GSTT1 gene influences the pharmacokinetics and detoxification of various drugs, such as isoniazid, rifampin, and multiple chemotherapeutic agents including doxorubicin and oxaliplatin, due to its role in enzymatic detoxification processes. This variation can lead to differences in drug efficacy and an increased risk of adverse effects, particularly in individuals with the null genotype that results in no enzyme expression.

**GSTM1**: GSTM1 plays a crucial role in the metabolism of several drugs, such as rifampin, busulfan, azathioprine, and isoniazid, with the GSTM1 null genotype leading to reduced enzyme activity that can increase plasma levels, slow elimination rates, and potentially result in higher drug exposure and toxicity. This impaired detoxification can influence both the efficacy and adverse effects of these drugs, underscoring the importance of considering GSTM1 status when managing medications to tailor drug dosages and minimize side effects.

**CYP2C19**: The enzyme CYP2C19 significantly influences the metabolism of various drugs including clopidogrel, proton pump inhibitors (omeprazole, esomeprazole), antidepressants (citalopram, escitalopram), antiepileptics (phenytoin), antifungals (voriconazole), and analgesics (methadone, oxycodone), which impacts their therapeutic efficacy and safety. Variations in the CYP2C19 gene affect drug activation and metabolism, leading to different pharmacological outcomes and necessitating genotype-guided dosing to optimize treatment efficacy and minimize adverse effects.

**CYP2C9**: CYP2C9 gene polymorphisms influence the metabolism of various drugs, affecting their therapeutic effectiveness and risk of side effects. Drugs such as warfarin, phenytoin, celecoxib, losartan, and tolbutamide require dose customization based on CYP2C9 genetic variants to optimize treatment outcomes and minimize complications like bleeding, drug toxicity, and gastrointestinal issues.

**CYP2B6**: CYP2B6 plays a pivotal role in the metabolism of several drugs, including efavirenz, nevirapine, methadone, bupropion, cyclophosphamide, and ifosfamide, where genetic polymorphisms can significantly influence drug efficacy and safety by altering pharmacokinetic profiles. Genetic variations in CYP2B6 affect the enzyme activity, leading to differences in drug plasma concentrations, therapeutic effects, and potential adverse reactions in individuals, underscoring its importance in personalized medicine.

**TNF**: Variations in the TNF gene significantly influence the effectiveness and side effects of TNF inhibitors like infliximab, adalimumab, and etanercept, which are used to treat autoimmune diseases by targeting TNF to reduce inflammation. Additionally, TNF gene variants might affect the metabolism or effects of drugs like atorvastatin, mycophenolate mofetil, gemcitabine, and methotrexate, mainly through impacting immune response and inflammation, thereby modifying pharmacokinetics and pharmacodynamics of these therapies.

**GSTA3**: GSTA3, part of the glutathione S-transferase family, plays a critical role in detoxifying xenobiotics and drugs, particularly by enhancing the solubility and excretion of agents susceptible to conjugation with glutathione such as certain antineoplastic agents. This enzymatic activity could reduce the effectiveness of drugs like alkylating agents used in chemotherapy by accelerating their detoxification, impacting therapeutic outcomes.

**GSTM2**: The gene GSTM2, instrumental in detoxifying substances, impacts the efficacy and toxicity of chemotherapy drugs like cisplatin and busulfan by influencing their metabolism. Cisplatin's side effects may be moderated by GSTM2, whereas busulfan's metabolism alterations can affect both its effectiveness and toxicity, stressing the importance of considering individual genetic differences in treatment plans.

**CYP2A7**: CYP2A7, a lesser-known member of the cytochrome P450 family, does not have well-established pharmacogenetic interactions with drugs like mercaptopurine, although it might theoretically affect mercaptopurine's metabolism similarly to its related isoform, CYP2A6. The interaction and impact of CYP2A7 on the metabolism of mercaptopurine are speculative and require further evidence, as mercaptopurine is primarily metabolized through other pathways involving TPMT and XO.

**GSTT2**: The GSTT2 gene, involved in cellular detoxification by aiding the conjugation of glutathione to various compounds, influences the pharmacokinetics and detoxification of chemotherapeutic drugs such as busulfan and chlorambucil, which undergo glutathione-related reactions. This suggests that genetic variations in GSTT2 could impact drug clearance and toxicity, potentially affecting treatment efficacy and susceptibility to drug-induced oxidative stress in chemotherapy.

**GSTM5**: The GSTM5 gene, which encodes the enzyme Glutathione S-transferase Mu 5, plays a significant role in detoxifying and metabolizing drugs, including the chemotherapy agent busulfan. Variations in GSTM5 can influence the pharmacokinetics of busulfan, affecting its efficacy and toxicity by altering the detoxification process.

**GSTM4**: GSTM4 is crucial in detoxifying chemotherapeutic agents and anti-inflammatory drugs by conjugating glutathione to enhance solubility and promote excretion, which impacts drug effectiveness and toxicity. This gene may interact with drugs like cisplatin and busulfan through this pathway, and variations in GSTM4 can influence drug response and the profile of adverse effects by altering the efficiency of detoxification.

**GSTA5**: GSTA5, which is involved in detoxification by catalyzing the conjugation of glutathione to compounds, plays a role in how drugs like busulfan are metabolized. Genetic variations in GSTA5 could influence the pharmacokinetics of busulfan, impacting its efficacy and toxicity, thus potentially optimizing dosing and improving therapeutic outcomes.

**GSTM3**: GSTM3, an enzyme involved in drug metabolism and detoxification, plays a key role in the body's handling of olanzapine and cyclophosphamide by conjugating toxins with glutathione, which affects these drugs' solubility and excretion. For olanzapine, GSTM3 may alter the drug's clearance and efficacy, while for cyclophosphamide, it is crucial in detoxifying metabolites that contribute to the drug’s cytotoxic effects, thus impacting the drug's effectiveness or toxicity.

**CYP2F1**: CYP2F1, involved mainly in metabolizing inhaled environmental pollutants like naphthalene, has limited direct interaction with common drugs but plays a role in the metabolism of certain pharmaceuticals such as imatinib due to shared metabolic pathways or competitive inhibition. Although CYP2F1 does not typically directly influence imatinib's metabolism, its interaction with similar chemical structures could alter the drug's efficacy or toxicity in individuals with specific gene variants.

**CYP2J2**: CYP2J2, primarily involved in metabolizing polyunsaturated fatty acids affecting cardiovascular physiology, does not have well-documented direct pharmacokinetic interactions with drugs like mycophenolate mofetil and tacrolimus. However, its role in modulating vascular tone and inflammation could potentially influence the pharmacodynamics of these immunosuppressive drugs used in organ transplantation and autoimmune diseases, despite the lack of a direct metabolic pathway involving CYP2J2.

**ATAT1**: Drugs like carboplatin and gemcitabine, used in cancer treatment, indirectly affect the function of the ATAT1 gene, which is critical for the stability of microtubules, by disrupting microtubule dynamics and DNA replication mechanisms. This indirect interaction may impact the efficacy of these drugs and lead to variations in patient responses based on differences in ATAT1 expression levels or genetic variants.

**GSTA4**: The gene GSTA4 plays a significant role in detoxifying various compounds by facilitating the conjugation of glutathione to electrophilic substances, impacting the pharmacokinetics and efficacy of chemotherapy drugs, particularly alkylating agents. Variations in GSTA4 function or expression can alter the metabolism of these drugs, potentially affecting their effectiveness and toxicity by changing how quickly and extensively the drugs are processed.

**ABCB1**: The ABCB1 gene, encoding P-glycoprotein, is crucial in modulating the pharmacokinetics and efficacy of multiple drugs by acting as an efflux transporter that influences their absorption, distribution, and excretion. This gene affects the bioavailability and brain penetration of various drugs such as antivirals like efavirenz and ritonavir, chemotherapeutics like doxorubicin and paclitaxel, cardiovascular drugs like digoxin, immunosuppressants such as tacrolimus and sirolimus, and analgesics like morphine and codeine, necessitating dosage adjustments and careful monitoring in clinical practice for optimized therapeutic outcomes.

**CYP2A13**: CYP2A13, a cytochrome P450 enzyme primarily found in the respiratory tract, plays a significant role in activating procarcinogens in tobacco smoke, such as nitrosamines, impacting lung cancer risk among smokers. Although it does not metabolize nicotine directly, CYP2A13 affects the activation of other smoke components that interact with nicotine's pathway, potentially influencing nicotine addiction and cancer risks, with genetic variations in CYP2A13 altering susceptibility to these effects.

**CYP1A1**: CYP1A1, an important enzyme in drug metabolism, exhibits pharmacogenetic interactions that impact the efficacy and toxicity of drugs like theophylline and clozapine through variations in metabolic activity based on genetic differences. Drugs such as deferasirox, valproic acid, and caffeine among others show altered pharmacokinetics when metabolized by CYP1A1, while others like imatinib and nilotinib might have altered pharmacodynamics or metabolism influenced by CYP1A1 activity, affecting treatment outcomes especially in cancer therapy.

**CERS6**: Drugs that modulate sphingolipid metabolism, specifically ceramides, may have their efficacy and safety profiles altered by the functional status of the CERS6 gene in diseases like cancer, diabetes, and neurodegenerative disorders. This interaction is primarily pharmacodynamic, affecting how drugs influence disease pathways mediated by ceramides, rather than altering the drug metabolism directly.

**CYP2D7**: CYP2D7, a pseudogene, does not encode a functional enzyme and thus does not partake in the metabolism of drugs, unlike its close relative CYP2D6, which metabolizes about 25% of clinically used drugs such as antidepressants, antipsychotics, beta-blockers, and opioids. Although CYP2D7 does not directly interact with these drugs, its similarity to CYP2D6 makes its study important for ensuring accurate pharmacogenetic testing and understanding genetic misannotations which can affect drug dosing and therapy.

**MT-RNR1**: The MT-RNR1 gene mutation m.1555A>G heightens susceptibility to ototoxic effects from aminoglycoside antibiotics such as gentamicin and tobramycin due to increased sensitivity of the mitochondrial ribosome, leading to impaired protein synthesis in cochlear cells and increasing the risk of irreversible hearing loss. The interaction between MT-RNR1 and aminoglycosides is pharmacodynamic as it influences the drug's action on the mitochondrial ribosome, rather than affecting the drug's metabolism or disposition.

**GSTA1**: GSTA1 (Glutathione S-transferase alpha 1) plays a key role in the metabolism and detoxification of drugs like busulfan and cyclophosphamide through its enzyme activity, which involves conjugating reduced glutathione with various electrophilic compounds. Variations in the GSTA1 gene can affect the efficacy and toxicity of these drugs by altering their metabolism, and this gene may also influence the response to other drugs such as prednisone, doxorubicin, and vincristine, possibly through mechanisms related to oxidative stress modulation.

**CYP2C8**: The CYP2C8 gene plays a crucial role in metabolizing various medications such as paclitaxel, pioglitazone, repaglinide, and amodiaquine, affecting their efficacy and toxicity due to genetic variations that influence drug metabolism and exposure. These interactions, involving pharmacokinetic processes, suggest that CYP2C8 genetic testing could be utilized to optimize drug dosing and reduce toxicity.

**MGST2**: Drugs such as chemotherapeutics that require glutathione conjugation, like busulfan and cisplatin, may have their bioavailability and toxicity altered due to variations in MGST2 activity. This enzyme, prominently expressed in the liver, kidneys, and intestines, plays a key role in the detoxification process and thus, influences the pharmacokinetics of these drugs, affecting their efficacy and safety.

**GSTA2**: GSTA2 plays a crucial role in the metabolism and detoxification of drugs such as acetaminophen and chemotherapeutic agents by catalyzing their conjugation with glutathione, influencing drug elimination and reducing toxicity. Polymorphisms in GSTA2 can affect the enzyme's efficiency, leading to variations in the detoxification rate, which may alter drug responses and increase toxicity risks in individuals.

**CYP2S1**: CYP2S1, a cytochrome P450 enzyme, influences the metabolism and efficacy of drugs, especially those administered via inhalation or topically, due to its expression in the liver, lungs, and skin. Although specific drugs affected by CYP2S1 are not well-determined, variations in the CYP2S1 gene can alter how these drugs are metabolized, potentially affecting their efficacy and toxicity.

**ADORA2A-AS1**: The gene ADORA2A-AS1, encoding the antisense RNA of ADORA2A, indirectly influences the effectiveness of drugs targeting the adenosine A2a receptor by modulating the receptor's expression. Among these, caffeine, an antagonist of the A2a receptor, can have altered stimulatory effects due to changes in ADORA2A expression mediated by ADORA2A-AS1, while the connection between ADORA2A-AS1 and methotrexate's effects is less certain and speculative.

**GSTP1**: GSTP1 significantly influences the pharmacogenetics and resultant efficacy of various drugs by modifying their metabolism and toxicity, as seen in drugs like oxaliplatin, thiotepa, olanzapine, and cyclophosphamide through enzymatic detoxification processes such as glutathione conjugation. Additionally, GSTP1 impacts the resistance mechanisms to chemotherapy drugs such as doxorubicin, epirubicin, and irinotecan, potentially reducing their effectiveness, while also affecting the efficacy and safety of drugs in other therapeutic areas like anesthesia and immunosuppression.

**HCP5**: HCP5 plays a crucial role in pharmacogenetics, specifically interacting with abacavir, a drug used in HIV treatment. The hypersensitivity to abacavir is linked to the HCP5-related rs2395029 polymorphism, which affects the expression of the HLA-B*5701 allele, necessitating genetic testing before prescribing the drug to avoid severe adverse reactions. While there are potential interactions with other immune-modulating drugs, such as flucloxacillin and gemcitabine, these are not as clearly defined as the interaction with abacavir.

**CYP1A2**: CYP1A2 plays a pivotal role in the pharmacokinetics of drugs, influencing their metabolism and, consequently, efficacy and safety. This enzyme is responsible for processing medications such as clozapine, theophylline, caffeine, erlotinib, deferasirox, escitalopram, and clopidogrel, where genetic variability can significantly affect therapeutic outcomes and requires careful dosage adjustments to prevent toxicity or enhance drug effectiveness.

**CHAT**: The gene CHAT, critical for synthesizing acetylcholine, interacts with acetylcholinesterase inhibitors such as rivastigmine, donepezil, and galantamine, used in Alzheimer's disease management, by affecting their pharmacodynamics as these drugs aim to increase acetylcholine levels by preventing its breakdown. Variations in CHAT expression among individuals can amplify or modify the effects of these medications, while olanzapine, though primarily an antipsychotic, may also interact indirectly through CHAT due to its broad receptor influence on cognitive and neuromuscular systems.

**CYP2C18**: CYP2C18 plays a modest role in the metabolism of drugs like warfarin, escitalopram, and clozapine, mainly affecting their clearance or efficacy to a lesser extent than more prominent CYP enzymes. Genetic variations in CYP2C18 could influence the pharmacodynamics of drugs such as clopidogrel and acenocoumarol, rather than their pharmacokinetic profiles, indicating a potential need for personalized medicine considerations despite the enzyme's relative minor role.

**C5orf56**: C5orf56, a gene implicated in inflammatory and autoimmune processes, might affect the response to drugs such as hydrochlorothiazide through pharmacodynamic interactions relating to its role in inflammation, rather than through alterations in drug metabolism. These potential interactions, which are yet speculative, suggest that C5orf56 could modulate the effects of hydrochlorothiazide on blood pressure and cardiovascular health via immune modulation or inflammation.

**MGST3**: MGST3, involved in detoxifying substrates via glutathione conjugation, indirectly affects the pharmacokinetics of drugs like acetaminophen, which relies on glutathione for safe metabolism in the liver. Variations in MGST3 could alter glutathione levels, affecting the toxicity and effectiveness of acetaminophen and potentially other drugs that undergo similar metabolic processes.

**MBOAT1**: MBOAT1, crucial in ghrelin acylation and therefore appetite stimulation and energy balance, could be interacted with by drugs targeting metabolic or appetite control pathways, either influencing its expression or enzymatic activity. Particularly, this interaction might involve medications influencing ghrelin or those affecting similar metabolic processes, such as weight management or anti-obesity drugs, and potentially extends to drugs targeting lipid enzymes, affecting MBOAT1's function or efficiency.

**CYP3A4**: CYP3A4, significantly expressed in the liver and small intestine, influences the metabolism and thus the effects and toxicity of many drugs, including tacrolimus, statins like atorvastatin and lovastatin, chemotherapeutics such as docetaxel, and sedatives like midazolam. Interaction with inducers or inhibitors like carbamazepine and erythromycin respectively can either increase or decrease CYP3A4's activity, affecting the metabolism of other drugs such as sildenafil and necessitating dosage adjustments due to changes in pharmacokinetic profiles.

**TPMT**: The TPMT gene, through genetic variations like TPMT*2, TPMT*3A, and TPMT*3C, plays a significant role in influencing the metabolism of thiopurine drugs including azathioprine, mercaptopurine, and thioguanine. These variations can lead to reduced enzyme activity, thereby increasing the risk of severe myelosuppression if standard doses are used; consequently, dose adjustments based on individual TPMT activity are essential to prevent toxicity and optimize treatment efficacy.

**ABCC2**: ABCC2 gene variants influence the function of the MRP2 protein, affecting the pharmacokinetics and therapeutic outcomes of various drugs. For example, genetic variations like c.1249G>A and c.3972C>T can alter the excretion and plasma concentrations of drugs such as methotrexate, mycophenolate mofetil, tacrolimus, rosuvastatin, and others including irinotecan and docetaxel, thereby impacting drug effectiveness and toxicity. Non-pharmacokinetic interactions involve drugs like clopidogrel and sorafenib, which may affect therapeutic outcomes through MRP2-related pathways.

**GSTK1**: GSTK1 influences the metabolism and detoxification of drugs that undergo glutathione conjugation, such as acetaminophen and various chemotherapeutic agents, through its role in glutathione pathways. Variations in the GSTK1 gene due to genetic polymorphisms can affect the efficacy and toxicity of these drugs, potentially altering their therapeutic outcomes and safety, such as modifying the risk of acetaminophen-induced liver toxicity and impacting the efficacy and toxicity of chemotherapeutics in cancer treatment.

**NR1I2**: NR1I2, or pregnane X receptor (PXR), regulates genes encoding drug-metabolizing enzymes and transport systems like CYP3A4 and MDR1, influencing the pharmacokinetics of various drugs. Activators of NR1I2 such as rifampin and carbamazepine can increase the metabolism and decrease the efficacy of drugs like sirolimus and rosuvastatin by upregulating CYP3A4, while inhibitors can slow metabolism, increasing toxicity risk.

**HLA-B**: The HLA-B gene plays a crucial role in drug hypersensitivity reactions through allele-specific interactions, such as the HLA-B*57:01 allele causing severe reactions to abacavir and the HLA-B*15:02 allele being associated with severe adverse reactions to carbamazepine. These interactions, which involve the immune system's response to drug-modified peptides or the drug itself presented by the HLA-B protein, emphasize the importance of genetic testing in precision medicine to tailor drug therapy and minimize adverse reactions based on individual genetic profiles.

**DAOA**: The DAOA gene, which influences neuroactive amino acid modulation and impacts the DAO enzyme linked to D-serine metabolism at NMDA receptors, has pharmacogenetic interactions relevant to drugs like aripiprazole and methamphetamine. These interactions can alter the effectiveness of aripiprazole in treating schizophrenia and bipolar disorder due to its involvement in glutamatergic pathways, and affect the action of methamphetamine through more complex neurotransmitter system modifications, particularly in neuropsychiatric conditions.

**GSS**: Variants in the GSS gene can influence the efficacy and toxicity of drugs like acetaminophen by modifying the availability of glutathione, which detoxifies reactive oxidative intermediates during drug metabolism. Reduced GSS activity is linked to increased risk of drug-induced oxidative stress and liver damage, such as acetaminophen-induced hepatotoxicity, due to diminished detoxification capacity.
