# Drug of interest
My drug of interest is rifampin.

# Pre-ranked list of genes
These are the candidate genes for this drugs, tentatively ranked by the likelihood of a pharmacogenetic relationship:
1. NAT2 (z: 40.77)
2. GSTM1 (z: 34.58)
3. SLCO1B1 (z: 33.61)
4. GSTT1 (z: 30.7)
5. CYP27B1 (z: 30.59)
6. NR1I2 (z: 28.6)
7. VDR (z: 12.38)
8. GSTM2 (z: 6.32)
9. GSTM4 (z: 5.24)
10. GSTM5 (z: 5.04)
11. GSTT2 (z: 4.02)
12. GSTM3 (z: 3.86)
13. GSTA3 (z: 3.62)
14. SLCO1B3 (z: 3.5)
15. GSTA5 (z: 3.06)
16. ABCC2 (z: 3.06)
17. NAT1 (z: 3.04)
18. CYP24A1 (z: 2.56)
19. GSTP1 (z: 2.53)
20. GSTK1 (z: 2.48)
21. CYP2R1 (z: 2.28)
22. DUX1 (z: 2.26)
23. CYP2B6 (z: 2.24)
24. GSTA4 (z: 2.14)
25. CYP2C19 (z: 2.13)
26. GSTA2 (z: 2.05)
27. FMO4 (z: 1.99)
28. MGST1 (z: 1.99)
29. CYP2E1 (z: 1.91)
30. CYP3A4 (z: 1.85)
31. TNF (z: 1.84)
32. CUX2 (z: 1.69)
33. ABCB1 (z: 1.68)
34. CYP2C9 (z: 1.68)
35. MAFK (z: 1.65)
36. SLCO6A1 (z: 1.58)
37. RIPOR2 (z: 1.56)
38. CYP3A5 (z: 1.49)
39. ADORA2A-AS1 (z: 1.45)
40. XPO1 (z: 1.42)
41. BACH1 (z: 1.39)
42. GSTA1 (z: 1.38)
43. AGBL4 (z: 1.28)
44. SLCO1A2 (z: 1.28)
45. GPX5 (z: 1.17)
46. CYP27A1 (z: 1.16)
47. SLCO1C1 (z: 1.13)
48. MGST2 (z: 1.12)
49. GSTO1 (z: 1.07)
50. GSTZ1 (z: 1.03)

# Known pharmacogenetic associations
In addition, please consider that the following pharmacogenetic interactions are already known for this drug, according to PharmGKB:
- NAT2
- GSTM1
- SLCO1B1
- GSTT1
- CYP27B1
- NR1I2
- VDR

# Auxiliary information about the drug
Below is some auxiliary information about the drug of interest:

**rifampin**: Rifampicin's interaction with genes like CYP2C9, CYP2C19, and CYP2E1 affects its metabolism, while polymorphisms in SLCO1B1 influence its hepatic uptake, impacting both the efficacy and safety of the drug. Additionally, rifampicin also influences the function of genes such as CYP2B6, ABCB1, and NR1I2, which affects drug transport and metabolism, along with other enzymes involved in detoxification like GSTT1, GSTM1, and NAT2, thereby modifying drug response or the risk of toxicity.

# Auxiliary information about the genes
Below is some auxiliary information about the genes in the pre-ranked list:
**NAT2**: The NAT2 gene encodes an enzyme crucial for the metabolism of various drugs through N-acetylation, influencing their pharmacokinetics and toxicity. Variants in NAT2 affect the metabolism of drugs like isoniazid, hydralazine, procainamide, sulfamethoxazole, and pyrazinamide, leading to differences in acetylator status (rapid, intermediate, slow) among individuals, which impacts drug plasma levels, exposure duration, and risk of adverse effects such as increased toxicity and drug-induced lupus.

**GSTM1**: GSTM1 plays a crucial role in the metabolism of several drugs, such as rifampin, busulfan, azathioprine, and isoniazid, with the GSTM1 null genotype leading to reduced enzyme activity that can increase plasma levels, slow elimination rates, and potentially result in higher drug exposure and toxicity. This impaired detoxification can influence both the efficacy and adverse effects of these drugs, underscoring the importance of considering GSTM1 status when managing medications to tailor drug dosages and minimize side effects.

**SLCO1B1**: The SLCO1B1 gene, which facilitates the hepatic uptake of drugs, exhibits polymorphisms like c.521T>C that affect the pharmacokinetics of multiple drugs including statins, repaglinide, and rifampin, potentially leading to increased drug plasma concentrations, altered efficacy, and toxicity. This variability in drug transport efficiency necessitates adjustments in dosing to optimize therapeutic outcomes and minimize adverse effects for drugs such as simvastatin, atorvastatin, and others.

**GSTT1**: The GSTT1 gene influences the pharmacokinetics and detoxification of various drugs, such as isoniazid, rifampin, and multiple chemotherapeutic agents including doxorubicin and oxaliplatin, due to its role in enzymatic detoxification processes. This variation can lead to differences in drug efficacy and an increased risk of adverse effects, particularly in individuals with the null genotype that results in no enzyme expression.

**CYP27B1**: The drug rifampin may decrease the efficacy of vitamin D conversion by enhancing the metabolism of its precursors through CYP27B1, a gene responsible for the enzyme converting 25-hydroxyvitamin D3 to its active form. Meanwhile, ribavirin, tenofovir disoproxil fumarate, and deferasirox can modify vitamin D levels and bone health through mechanisms potentially linked to CYP27B1 activities, affecting the pharmacodynamics concerning vitamin D status.

**NR1I2**: NR1I2, or pregnane X receptor (PXR), regulates genes encoding drug-metabolizing enzymes and transport systems like CYP3A4 and MDR1, influencing the pharmacokinetics of various drugs. Activators of NR1I2 such as rifampin and carbamazepine can increase the metabolism and decrease the efficacy of drugs like sirolimus and rosuvastatin by upregulating CYP3A4, while inhibitors can slow metabolism, increasing toxicity risk.

**VDR**: The Vitamin D Receptor (VDR) gene significantly affects the pharmacokinetics and pharmacodynamics of various drugs due to its role in vitamin D metabolism and bone health. On the pharmacokinetic side, drugs like midazolam and rifampin may have altered metabolism or effects based on VDR activity or genotypes, while pharmacodynamically, VDR variants can alter the efficacy of drugs such as calcitriol, raloxifene, alendronate, dexamethasone, and tamoxifen, impacting therapeutic outcomes in conditions like osteoporosis, rickets, and immune or hormone-responsive diseases.

**GSTM2**: The gene GSTM2, instrumental in detoxifying substances, impacts the efficacy and toxicity of chemotherapy drugs like cisplatin and busulfan by influencing their metabolism. Cisplatin's side effects may be moderated by GSTM2, whereas busulfan's metabolism alterations can affect both its effectiveness and toxicity, stressing the importance of considering individual genetic differences in treatment plans.

**GSTM4**: GSTM4 is crucial in detoxifying chemotherapeutic agents and anti-inflammatory drugs by conjugating glutathione to enhance solubility and promote excretion, which impacts drug effectiveness and toxicity. This gene may interact with drugs like cisplatin and busulfan through this pathway, and variations in GSTM4 can influence drug response and the profile of adverse effects by altering the efficiency of detoxification.

**GSTM5**: The GSTM5 gene, which encodes the enzyme Glutathione S-transferase Mu 5, plays a significant role in detoxifying and metabolizing drugs, including the chemotherapy agent busulfan. Variations in GSTM5 can influence the pharmacokinetics of busulfan, affecting its efficacy and toxicity by altering the detoxification process.

**GSTT2**: The GSTT2 gene, involved in cellular detoxification by aiding the conjugation of glutathione to various compounds, influences the pharmacokinetics and detoxification of chemotherapeutic drugs such as busulfan and chlorambucil, which undergo glutathione-related reactions. This suggests that genetic variations in GSTT2 could impact drug clearance and toxicity, potentially affecting treatment efficacy and susceptibility to drug-induced oxidative stress in chemotherapy.

**GSTM3**: GSTM3, an enzyme involved in drug metabolism and detoxification, plays a key role in the body's handling of olanzapine and cyclophosphamide by conjugating toxins with glutathione, which affects these drugs' solubility and excretion. For olanzapine, GSTM3 may alter the drug's clearance and efficacy, while for cyclophosphamide, it is crucial in detoxifying metabolites that contribute to the drug’s cytotoxic effects, thus impacting the drug's effectiveness or toxicity.

**GSTA3**: GSTA3, part of the glutathione S-transferase family, plays a critical role in detoxifying xenobiotics and drugs, particularly by enhancing the solubility and excretion of agents susceptible to conjugation with glutathione such as certain antineoplastic agents. This enzymatic activity could reduce the effectiveness of drugs like alkylating agents used in chemotherapy by accelerating their detoxification, impacting therapeutic outcomes.

**SLCO1B3**: SLCO1B3, which encodes the organic anion transporting polypeptide 1B3 (OATP1B3), plays a crucial role in the hepatic uptake and metabolism of several drugs including docetaxel, mycophenolate mofetil, telmisartan, and voriconazole, affecting their plasma levels, therapeutic effects, and toxicity. Variants in this gene can modify the transport efficiency of these drugs, influencing their absorption, distribution, elimination, and overall pharmacokinetics, thereby necessitating tailored dosing in personalized medicine to enhance efficacy and reduce adverse effects.

**GSTA5**: GSTA5, which is involved in detoxification by catalyzing the conjugation of glutathione to compounds, plays a role in how drugs like busulfan are metabolized. Genetic variations in GSTA5 could influence the pharmacokinetics of busulfan, impacting its efficacy and toxicity, thus potentially optimizing dosing and improving therapeutic outcomes.

**ABCC2**: ABCC2 gene variants influence the function of the MRP2 protein, affecting the pharmacokinetics and therapeutic outcomes of various drugs. For example, genetic variations like c.1249G>A and c.3972C>T can alter the excretion and plasma concentrations of drugs such as methotrexate, mycophenolate mofetil, tacrolimus, rosuvastatin, and others including irinotecan and docetaxel, thereby impacting drug effectiveness and toxicity. Non-pharmacokinetic interactions involve drugs like clopidogrel and sorafenib, which may affect therapeutic outcomes through MRP2-related pathways.

**NAT1**: The NAT1 enzyme, which catalyzes the acetylation of various substrates, affects the metabolism and efficacy of several drugs including cotinine, thioguanine, and mesalazine. Variations in NAT1 activity influence the effectiveness and side effect profiles of these drugs by altering their metabolic pathways, which is critical for determining optimal dosages and therapeutic strategies tailored to individual genetic profiles.

**CYP24A1**: The drugs telaprevir, deferasirox, and daclatasvir may interact with the CYP24A1 enzyme, which is crucial for metabolizing vitamin D, thereby potentially affecting the bioavailability of active vitamin D levels and altering drug metabolism. This understanding is important for predicting drug efficacy and safety, which may necessitate adjustments in dosage and treatment monitoring based on individual genetic variations in CYP24A1 activity.

**GSTP1**: GSTP1 significantly influences the pharmacogenetics and resultant efficacy of various drugs by modifying their metabolism and toxicity, as seen in drugs like oxaliplatin, thiotepa, olanzapine, and cyclophosphamide through enzymatic detoxification processes such as glutathione conjugation. Additionally, GSTP1 impacts the resistance mechanisms to chemotherapy drugs such as doxorubicin, epirubicin, and irinotecan, potentially reducing their effectiveness, while also affecting the efficacy and safety of drugs in other therapeutic areas like anesthesia and immunosuppression.

**GSTK1**: GSTK1 influences the metabolism and detoxification of drugs that undergo glutathione conjugation, such as acetaminophen and various chemotherapeutic agents, through its role in glutathione pathways. Variations in the GSTK1 gene due to genetic polymorphisms can affect the efficacy and toxicity of these drugs, potentially altering their therapeutic outcomes and safety, such as modifying the risk of acetaminophen-induced liver toxicity and impacting the efficacy and toxicity of chemotherapeutics in cancer treatment.

**CYP2R1**: Variations in the CYP2R1 gene can affect the efficiency of vitamin D metabolism by altering the enzyme's ability to convert vitamin D into 25-hydroxyvitamin D, influencing the effectiveness of vitamin D supplementation. Additionally, interactions between CYP2R1 and ribavirin don't involve direct metabolism by the enzyme but may involve indirect effects related to ribavirin’s impact on vitamin D levels or bone health, potentially affecting treatment outcomes in conditions like chronic hepatitis C.

**DUX1**: DUX1, a gene involved in transcription regulation, has suggested non-pharmacokinetic interactions with drugs such as isoniazid, rifampin, and caffeine, possibly through modifications in genetic expression that indirectly affect the metabolism of these drugs. The interaction between DUX1 and these drugs is inferred from the gene’s regulatory roles, rather than direct evidence, suggesting a need for further research to confirm and clarify these effects.

**CYP2B6**: CYP2B6 plays a pivotal role in the metabolism of several drugs, including efavirenz, nevirapine, methadone, bupropion, cyclophosphamide, and ifosfamide, where genetic polymorphisms can significantly influence drug efficacy and safety by altering pharmacokinetic profiles. Genetic variations in CYP2B6 affect the enzyme activity, leading to differences in drug plasma concentrations, therapeutic effects, and potential adverse reactions in individuals, underscoring its importance in personalized medicine.

**GSTA4**: The gene GSTA4 plays a significant role in detoxifying various compounds by facilitating the conjugation of glutathione to electrophilic substances, impacting the pharmacokinetics and efficacy of chemotherapy drugs, particularly alkylating agents. Variations in GSTA4 function or expression can alter the metabolism of these drugs, potentially affecting their effectiveness and toxicity by changing how quickly and extensively the drugs are processed.

**CYP2C19**: The enzyme CYP2C19 significantly influences the metabolism of various drugs including clopidogrel, proton pump inhibitors (omeprazole, esomeprazole), antidepressants (citalopram, escitalopram), antiepileptics (phenytoin), antifungals (voriconazole), and analgesics (methadone, oxycodone), which impacts their therapeutic efficacy and safety. Variations in the CYP2C19 gene affect drug activation and metabolism, leading to different pharmacological outcomes and necessitating genotype-guided dosing to optimize treatment efficacy and minimize adverse effects.

**GSTA2**: GSTA2 plays a crucial role in the metabolism and detoxification of drugs such as acetaminophen and chemotherapeutic agents by catalyzing their conjugation with glutathione, influencing drug elimination and reducing toxicity. Polymorphisms in GSTA2 can affect the enzyme's efficiency, leading to variations in the detoxification rate, which may alter drug responses and increase toxicity risks in individuals.

**FMO4**: The FMO4 gene encodes for the enzyme flavin-containing monooxygenase 4, which potentially contributes to the metabolism of drugs by affecting their pharmacokinetics, altering plasma drug levels, and impacting therapeutic outcomes or side effects. The exact drugs impacted by FMO4 are still speculative, as FMO4’s interactions with specific drugs are currently not well-documented and remain under research.

**MGST1**: Variants in the MGST1 gene can alter the metabolism of drugs, particularly chemotherapeutic and cardiovascular drugs, by affecting glutathione conjugation detoxification rates and response to oxidative stress. This may influence the efficacy and safety of these drugs, as MGST1 variants could lead to variability in patient responses, impacting therapeutic outcomes and the likelihood of adverse effects.

**CYP2E1**: The gene CYP2E1 significantly impacts the metabolism of various drugs; for instance, in the case of acetaminophen and isoniazid, variations in CYP2E1 activity can lead to differences in the rate at which these drugs are metabolized, thereby affecting their efficacy, toxicity, and risk of adverse effects such as hepatotoxicity. Additionally, CYP2E1 metabolizes other substances like caffeine, theophylline, nicotine, and some procarcinogens, indicating that genetic variations could influence both therapeutic outcomes and disease risk through pharmacokinetic and possibly pharmacodynamic interactions.

**CYP3A4**: CYP3A4, significantly expressed in the liver and small intestine, influences the metabolism and thus the effects and toxicity of many drugs, including tacrolimus, statins like atorvastatin and lovastatin, chemotherapeutics such as docetaxel, and sedatives like midazolam. Interaction with inducers or inhibitors like carbamazepine and erythromycin respectively can either increase or decrease CYP3A4's activity, affecting the metabolism of other drugs such as sildenafil and necessitating dosage adjustments due to changes in pharmacokinetic profiles.

**TNF**: Variations in the TNF gene significantly influence the effectiveness and side effects of TNF inhibitors like infliximab, adalimumab, and etanercept, which are used to treat autoimmune diseases by targeting TNF to reduce inflammation. Additionally, TNF gene variants might affect the metabolism or effects of drugs like atorvastatin, mycophenolate mofetil, gemcitabine, and methotrexate, mainly through impacting immune response and inflammation, thereby modifying pharmacokinetics and pharmacodynamics of these therapies.

**CUX2**: The interaction between the gene CUX2 and drugs like rifampin does not involve direct pharmacokinetic mechanisms but may affect pharmacodynamic processes, influencing how the brain responds to medication due to CUX2’s role in neuronal function and development. While rifampin primarily impacts drug-metabolizing enzymes and is not directly linked to neuronal pathways, its indirect effects on neurological conditions could be modified by variations in CUX2, potentially altering the effectiveness or side effects of treatments for conditions such as epilepsy, schizophrenia, and bipolar disorder.

**ABCB1**: The ABCB1 gene, encoding P-glycoprotein, is crucial in modulating the pharmacokinetics and efficacy of multiple drugs by acting as an efflux transporter that influences their absorption, distribution, and excretion. This gene affects the bioavailability and brain penetration of various drugs such as antivirals like efavirenz and ritonavir, chemotherapeutics like doxorubicin and paclitaxel, cardiovascular drugs like digoxin, immunosuppressants such as tacrolimus and sirolimus, and analgesics like morphine and codeine, necessitating dosage adjustments and careful monitoring in clinical practice for optimized therapeutic outcomes.

**CYP2C9**: CYP2C9 gene polymorphisms influence the metabolism of various drugs, affecting their therapeutic effectiveness and risk of side effects. Drugs such as warfarin, phenytoin, celecoxib, losartan, and tolbutamide require dose customization based on CYP2C9 genetic variants to optimize treatment outcomes and minimize complications like bleeding, drug toxicity, and gastrointestinal issues.

**MAFK**: The gene MAFK, which regulates the NRF2 pathway, influences the metabolism of drugs such as isoniazid and rifampin, which are used in treating tuberculosis. This impact is due to their induction of oxidative stress and involvement in hepatic metabolism, where MAFK-mediated changes in enzyme expression levels can alter drug efficacy and toxicity by modifying the rate of drug metabolism.

**SLCO6A1**: SLCO6A1, a gene encoding a protein in the organic anion transporter family, is suspected to play a role in the pharmacokinetics of drugs, particularly those affecting the central nervous system and those requiring passage across the blood-brain barrier. However, specific pharmacogenetic interactions between SLCO6A1 and individual drugs have not been well-documented as of now.

**RIPOR2**: Though RIPOR2 is mainly involved in cell polarization and cytoskeletal organization, it does not have a confirmed direct interaction with drugs affecting pharmacokinetics. However, it might have a non-pharmacokinetic interaction with rifampin, an antibiotic for tuberculosis, potentially influencing immune responses indirectly due to RIPOR2's role in immune cell functioning, which could affect the efficacy or side effect profile of rifampin in conditions with abnormal immune cell activities.

**CYP3A5**: CYP3A5 plays a key role in the metabolism of several drugs, including immunosuppressants like tacrolimus and cyclosporine, where genetic variants impact dosing due to altered metabolism rates. This enzyme also affects the metabolism and effectiveness of other drugs such as itraconazole, statins (simvastatin and atorvastatin), apixaban, vincristine, ondansetron, atazanavir, and fentanyl, which in turn influences their therapeutic outcomes and risk profiles.

**ADORA2A-AS1**: The gene ADORA2A-AS1, encoding the antisense RNA of ADORA2A, indirectly influences the effectiveness of drugs targeting the adenosine A2a receptor by modulating the receptor's expression. Among these, caffeine, an antagonist of the A2a receptor, can have altered stimulatory effects due to changes in ADORA2A expression mediated by ADORA2A-AS1, while the connection between ADORA2A-AS1 and methotrexate's effects is less certain and speculative.

**XPO1**: Selinexor, an inhibitor targeting the XPO1 gene, treats multiple myeloma and DLBCL by blocking the nuclear export of tumor suppressor proteins and promoting apoptosis in cancer cells. In contrast, tuberculosis drugs isoniazid and rifampin affect metabolic pathways in the liver and do not have direct pharmacogenetic interactions with the XPO1 gene, although they might influence drug metabolism and systemic levels of drugs like selinexor indirectly.

**BACH1**: Variations in BACH1, a gene involved in managing cellular responses to oxidative stress, may influence the pharmacodynamics of isoniazid and rifampin, drugs used in tuberculosis treatment that induce oxidative stress as a side effect. Depending on an individual's genetic profile related to BACH1, there could be differences in oxidative stress response, impacting the side effect profile and toxicity of these drugs.

**GSTA1**: GSTA1 (Glutathione S-transferase alpha 1) plays a key role in the metabolism and detoxification of drugs like busulfan and cyclophosphamide through its enzyme activity, which involves conjugating reduced glutathione with various electrophilic compounds. Variations in the GSTA1 gene can affect the efficacy and toxicity of these drugs by altering their metabolism, and this gene may also influence the response to other drugs such as prednisone, doxorubicin, and vincristine, possibly through mechanisms related to oxidative stress modulation.

**AGBL4**: The gene AGBL4, involved in modifying microtubules in the brain, implies potential pharmacogenetic interactions with drugs that affect microtubule stability, such as antimitotic cancer drugs, although no direct evidence of such interactions is currently confirmed. Additionally, AGBL4 may interact with drugs like Rifampin through broader pharmacodynamic effects, potentially affecting neuronal functions or development through mechanisms related to cellular stress or inflammation, rather than through direct enzymatic modifications of microtubules.

**SLCO1A2**: The gene SLCO1A2 significantly affects the pharmacokinetics of drugs like rocuronium, imatinib, and methotrexate by mediating their hepatic uptake, which alters their effectiveness, safety, and systemic clearance. Rocuronium's performance in anesthesia, imatinib's therapeutic outcomes in cancer treatment, and methotrexate's response in treating malignancies and autoimmune disorders are all influenced by variations in SLCO1A2 activity, impacting their hepatic clearance and overall drug dynamics.

**GPX5**: GPX5, mainly involved in reducing oxidative stress in male reproductive tissues, does not directly interact with drugs pharmacokinetically or pharmacodynamically. However, it may indirectly affect the impact of chemotherapeutic agents like carboplatin and gemcitabine, which generate oxidative stress, by potentially mitigating the side effects on non-target tissues, particularly reproductive tissues. This interaction is speculative and based solely on the enzyme's role in protecting against oxidative damage.

**CYP27A1**: CYP27A1, a gene involved in cholesterol and bile acid metabolism, indirectly affects the pharmacogenetics of cerebrotendinous xanthomatosis (CTX) treatment. Mutations in this gene can influence the effectiveness and dosing of chenodeoxycholic acid (CDCA), used to treat CTX by normalizing bile acid levels, and could also interact with other bile acids like cholic acid, suggesting that genetic variations may impact patient responses to these bile acid therapies.

**SLCO1C1**: The SLCO1C1 gene encodes a protein that transports thyroid hormones like thyroxine (T4) across membranes such as the blood-brain barrier, affecting the pharmacokinetics of drugs that rely on similar transport mechanisms. Drugs that inhibit or induce the function of this transporter could potentially alter thyroxine levels, impacting the treatment and outcomes in conditions where thyroid hormone balance is essential, such as hypothyroidism; this interaction may also influence the distribution and efficacy of drugs targeting the central nervous system.

**MGST2**: Drugs such as chemotherapeutics that require glutathione conjugation, like busulfan and cisplatin, may have their bioavailability and toxicity altered due to variations in MGST2 activity. This enzyme, prominently expressed in the liver, kidneys, and intestines, plays a key role in the detoxification process and thus, influences the pharmacokinetics of these drugs, affecting their efficacy and safety.

**GSTO1**: Variants of the GSTO1 gene affect the metabolism of drugs such as cisplatin, carboplatin, and arsenic-based treatments by altering enzyme activity. This impacts the detoxification of drug metabolites and reactive oxygen species, influencing both the effectiveness of the drugs and the severity of side effects associated with their use.

**GSTZ1**: GSTZ1, or Glutathione S-transferase zeta 1, plays a critical role in drug metabolism by detoxifying harmful compounds, including medications, through conjugation with glutathione. Variants in the GSTZ1 gene can affect enzyme activity, thus influencing the metabolism and effectiveness of certain drugs such as dichloroacetic acid used in lactic acidosis treatment, potentially impacting therapeutic outcomes and adverse reactions.
